// lodash-pari/commands/accept.js
module.exports.config = {
  name: 'accept',
  aliases: ['acceptrequest', 'fr','acp'],
  version: '1.0.0',
  permission: 1,
  prefix: true,
  premium: false,
  category: "group",
  description: "Accept friend requests",
  usage: "accept [number/all/uid] or just accept to see list and reply",
  credits: 'CHAND',
  cooldowns: 10,
};

module.exports.run = async function ({ api, event, args, send, client }) {
  const { threadID, senderID, messageID } = event;
  const moment = require("moment-timezone");
  
  const form = {
    av: api.getCurrentUserID(),
    fb_api_req_friendly_name: "FriendingCometFriendRequestsRootQueryRelayPreloader",
    fb_api_caller_class: "RelayModern",
    doc_id: "4499164963466303",
    variables: JSON.stringify({input: {scale: 3}})
  };
  
  const response = await api.httpPost("https://www.facebook.com/api/graphql/", form);
  const jsonRes = JSON.parse(response);
  
  if (!jsonRes.data || !jsonRes.data.viewer || !jsonRes.data.viewer.friending_possibilities) {
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅔𝐫𝐫𝐨𝐫 𝐟𝐞𝐭𝐜𝐡𝐢𝐧𝐠 𝐟𝐫𝐢𝐞𝐧𝐝 𝐫𝐞𝐪𝐮𝐞𝐬𝐭𝐬. 🅟𝐥𝐞𝐚𝐬𝐞 𝐭𝐫𝐲 𝐚𝐠𝐚𝐢𝐧 𝐥𝐚𝐭𝐞𝐫.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
  
  const listRequest = jsonRes.data.viewer.friending_possibilities.edges || [];
  
  if (listRequest.length === 0) {
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅣𝐡𝐞𝐫𝐞 𝐚𝐫𝐞 𝐧𝐨 𝐩𝐞𝐧𝐝𝐢𝐧𝐠 𝐟𝐫𝐢𝐞𝐧𝐝 𝐫𝐞𝐪𝐮𝐞𝐬𝐭𝐬!\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━ ');
  }
  
  let msg = `🖤 ━━▣━━◤ 🪽 ◢━━▣━━━ 🖤\n`;
  msg += `🅟𝐄𝐍𝐃𝐈𝐍𝐆 🅕𝐑𝐈𝐄𝐍𝐃 🅡𝐄𝐐𝐔𝐄𝐒𝐓𝐒\n`;
  msg += `══════════════════════\n\n`;
  
  let i = 0;
  for (const user of listRequest) {
    i++;
    if (i <= 20) {
      msg += `${i}.\nname : ${user.node.name}\n`;
      msg += `id : ${user.node.id}\n`;
      msg += `time : ${moment(user.time*1000).tz("Asia/karachi").format("DD/MM/YYYY HH:mm:ss")}\n\n`;
    }
  }
  
  if (listRequest.length > 20) {
    msg += `... and ${listRequest.length - 20} more\n\n`;
  }
  
  msg += `══════════════════════\n`;
  msg += `📌 Reply: add 1 & add all\n`;
  msg += `📌 Reply: del 1 & del all\n🖤 ━━▣━━◤ 🪽 ◢━━▣━━━ 🖤`;
  
  const pendingList = listRequest;
  
  const info = await send.reply(msg);
  
  if (client.replies && info?.messageID) {
    client.replies.set(info.messageID, {
      commandName: 'accept',
      author: senderID,
      data: { pendingList }
    });
  }
  
  return;
};

module.exports.handleReply = async function ({ api, event, send, data }) {
  const { body, senderID, threadID, messageID } = event;
  
  if (!data || !data.pendingList) {
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅓𝐚𝐭𝐚 𝐧𝐨𝐭 𝐟𝐨𝐮𝐧𝐝. 🅟𝐥𝐞𝐚𝐬𝐞 𝐮𝐬𝐞 𝐚𝐜𝐜𝐞𝐩𝐭 𝐜𝐨𝐦𝐦𝐚𝐧𝐝 𝐚𝐠𝐚𝐢𝐧.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
  
  const { pendingList } = data;
  const input = body.trim().toLowerCase();
  const parts = input.split(" ");
  const action = parts[0];
  const target = parts[1];
  
  if (!action) {
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐬𝐩𝐞𝐜𝐢𝐟𝐲 𝐚𝐜𝐭𝐢𝐨𝐧: 𝐚𝐝𝐝 1 / 𝐚𝐝𝐝 𝐚𝐥𝐥 / 𝐝𝐞𝐥 1 / 𝐝𝐞𝐥 𝐚𝐥𝐥\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
  
  let targetIndices = [];
  
  if (target === "all") {
    targetIndices = pendingList.map((_, idx) => idx);
  } else if (target) {
    const idx = parseInt(target) - 1;
    if (isNaN(idx) || idx < 0 || idx >= pendingList.length) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅘𝐧𝐯𝐚𝐥𝐢𝐝 𝐧𝐮𝐦𝐛𝐞𝐫. 🅤𝐬𝐞 𝐚𝐝𝐝 1 𝐨𝐫 𝐝𝐞𝐥 1\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    targetIndices = [idx];
  } else {
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐬𝐩𝐞𝐜𝐢𝐟𝐲 𝐧𝐮𝐦𝐛𝐞𝐫: 𝐚𝐝𝐝 1 / 𝐝𝐞𝐥 1 / 𝐚𝐝𝐝 𝐚𝐥𝐥 / 𝐝𝐞𝐥 𝐚𝐥𝐥\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
  
  const isAdd = action === "add";
  const isDel = action === "del";
  
  if (!isAdd && !isDel) {
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅘𝐧𝐯𝐚𝐥𝐢𝐝 𝐚𝐜𝐭𝐢𝐨𝐧. 🅤𝐬𝐞: 𝐚𝐝𝐝 1 / 𝐚𝐝𝐝 𝐚𝐥𝐥 / 𝐝𝐞𝐥 1 / 𝐝𝐞𝐥 𝐚𝐥𝐥\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
  
  const docId = isAdd ? "3147613905362928" : "4108254489275063";
  const fbApiName = isAdd ? "FriendingCometFriendRequestConfirmMutation" : "FriendingCometFriendRequestDeleteMutation";
  
  const success = [];
  const failed = [];
  
  for (const idx of targetIndices) {
    const user = pendingList[idx];
    const form = {
      av: api.getCurrentUserID(),
      fb_api_caller_class: "RelayModern",
      fb_api_req_friendly_name: fbApiName,
      doc_id: docId,
      variables: JSON.stringify({
        input: {
          source: "friends_tab",
          actor_id: api.getCurrentUserID(),
          friend_requester_id: user.node.id,
          client_mutation_id: Math.round(Math.random() * 19).toString()
        },
        scale: 3,
        refresh_num: 0
      })
    };
    
    try {
      const res = await api.httpPost("https://www.facebook.com/api/graphql/", form);
      const json = JSON.parse(res);
      
      if (json.errors && json.errors.length > 0) {
        const errorMsg = json.errors[0].message || '';
        if (errorMsg.includes('already') || errorMsg.includes('duplicate') || errorMsg.includes('same')) {
          success.push(user.node.name);
        } else if (isAdd) {
          success.push(user.node.name);
        } else {
          failed.push(user.node.name);
        }
      } else {
        success.push(user.node.name);
      }
    } catch (e) {
      failed.push(user.node.name);
    }
  }
  
  let resultMsg = `${isAdd ? 'Added' : 'Deleted'}: ${success.length}\n`;
  if (success.length > 0) resultMsg += success.join('\n');
  if (failed.length > 0) resultMsg += `\nFailed: ${failed.length}\n` + failed.join('\n');
  
  return send.reply(resultMsg);
};
