// lodash-pari/commands/add.js
module.exports = {
  config: {
    name: 'add',
    aliases: ['adduser'],
    version: '1.0.0',
    permission: 0,
    prefix: true,
    premium: false,
    category: 'Group',
    description: 'Add user to group',
    usage: 'add [uid]',
    credits: 'CHAND',
    cooldowns: 10,
    groupOnly: true
  },

  async run({ api, event, args, send }) {
    const { threadID, messageID } = event;

    if (!args[0]) {
      return send.reply('╭┄୨༝୧┄☁┄⊰꒱❀꒰⊱┄☁┄୨༝୧┄╮\n┆               𝐍𝐨𝐭𝐞                ┆\n┆╭┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╮┆\n┆┆• Please enter the UID ┆┆\n┆╰┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╯┆\n╰┄୨༝୧┄☁┄⊰꒱❀꒰⊱┄☁┄୨༝୧┄╯');
    }

    const uidUser = args[0];

    if (!/^\d+$/.test(uidUser)) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐯𝐚𝐥𝐢𝐝 𝐔𝐈𝐃 (𝐧𝐮𝐦𝐛𝐞𝐫𝐬) \n\n🅔𝐱𝐚𝐦𝐩𝐥𝐞 : 𝐚𝐝𝐝 [𝐮𝐢𝐝]\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }

    try {
      const threadInfo = await api.getThreadInfo(threadID);
      const { participantIDs } = threadInfo;

      if (participantIDs.includes(uidUser)) {
        return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅣𝐡𝐞 𝐦𝐞𝐦𝐛𝐞𝐫 𝐢𝐬 𝐚𝐥𝐫𝐞𝐚𝐝𝐲 𝐢𝐧 𝐭𝐡𝐞 𝐠𝐫𝐨𝐮𝐩\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
      }

      await api.addUserToGroup(uidUser, threadID);
      
      await new Promise(resolve => setTimeout(resolve, 1000));
      
      const newThreadInfo = await api.getThreadInfo(threadID);
      if (newThreadInfo.participantIDs.includes(uidUser)) {
        return send.reply(`╭┄୨༝୧┄☁┄⊰꒱❀꒰⊱┄☁┄୨༝୧┄╮\n┆               𝐍𝐨𝐭𝐞                ┆\n┆╭┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╮┆\n┆┆• Successfully added   ┆┆\n┆╰┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╯┆\n╰┄୨༝୧┄☁┄⊰꒱❀꒰⊱┄☁┄୨༝୧┄╯`);
      } else {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ User not added\nUser may have privacy restrictions\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
    } catch (err) {
      console.log('[ADD] Error:', err);
      
      const errorMsg = err?.message || err?.error || err || '';
      
      if (errorMsg.includes('1545052') || errorMsg.includes('permission')) {
        return send.reply('❌ Cannot add user.\n• User privacy blocking\n• Or group restrictions');
      } else if (errorMsg.includes('1545010') || errorMsg.includes('not found')) {
        return send.reply('❌ User not found');
      } else if (errorMsg.includes('1545047') || errorMsg.includes('left recently')) {
        return send.reply('❌ User left recently');
      } else if (errorMsg.includes('200') || errorMsg.includes('privacy')) {
        return send.reply('❌ User has privacy restrictions');
      }
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ Error: ${errorMsg}\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
  }
};
