// lodash-pari/commands/admin.js
const { GOD_UIDS } = require('../../../Data/system/handle/handleCommand');

module.exports = {
  config: {
    name: 'admin',
    aliases: ['admins', 'botadmin'],
    version: '1.0.0',
    permission: 0,
    prefix: true,
    premium: false,
    category: 'Admin',
    description: 'Manage bot admins',
    usage: 'admin [add/remove/list] [uid]',
    credits: 'CHAND',
    cooldowns: 10,
    adminOnly: true
  },
  
  async run({ api, event, args, send, config: globalConfig }) {
    const { threadID, messageID, senderID } = event || {};
    const action = args[0]?.toLowerCase();
    
    const fs = require('fs-extra');
    const path = require('path');
    const configPath = path.join(__dirname, '../../../config.json');
    let config = fs.readJsonSync(configPath);
    
    const isGod = senderID && GOD_UIDS?.includes(senderID);
    const isBotAdmin = config.ADMINBOT?.includes(senderID);
    
    if (global.config) {
      global.config = config;
    }
    
    if (!action || action === 'list') {
      const admins = config.ADMINBOT || [];
      let msg = `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n    🅑𝐎𝐓 𝐀𝐃𝐌𝐈𝐍𝐒 (${admins.length}) \n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n`;
      
      for (let i = 0; i < admins.length; i++) {
        try {
          const info = await api.getUserInfo(admins[i]);
          let name = info[admins[i]]?.name;
          if (!name || name.toLowerCase() === 'facebook user' || name.toLowerCase() === 'facebook') {
            name = info[admins[i]]?.firstName;
            if (!name || name.toLowerCase() === 'facebook') {
              name = 'Admin';
            }
          }
          msg += `${i + 1}. ${name}\n   UID: ${admins[i]}\n`;
        } catch {
          msg += `${i + 1}. UID: ${admins[i]}\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━ `;
        }
      }
      
      return send.reply(msg);
    }
    
    const uid = args[1];
    
    if (!isBotAdmin && !isGod && (action === 'add' || action === 'remove' || action === 'del' || action === 'rm')) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅞𝐧𝐥𝐲 𝐀𝐃𝐌𝐈𝐍 or 𝐆𝐎𝐃 𝐜𝐚𝐧 𝐚𝐝𝐝/𝐫𝐞𝐦𝐨𝐯𝐞.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (!uid || !/^\d+$/.test(uid)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐯𝐚𝐥𝐢𝐝 𝐔𝐈𝐃.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (action === 'add') {
      if (config.ADMINBOT.includes(uid)) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅣𝐡𝐢𝐬 𝐮𝐬𝐞𝐫 𝐢𝐬 𝐚𝐥𝐫𝐞𝐚𝐝𝐲 𝐚𝐧 𝐚𝐝𝐦𝐢𝐧.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      config.ADMINBOT.push(uid);
      fs.writeFileSync(configPath, JSON.stringify(config, null, 2));
      
      if (global.config) {
        global.config = config;
      }
      
      let name = 'User';
      try {
        const info = await api.getUserInfo(uid);
        const rawName = info[uid]?.name;
        if (rawName && rawName.toLowerCase() !== 'facebook user' && rawName.toLowerCase() !== 'facebook') {
          name = rawName;
        } else if (info[uid]?.firstName && info[uid].firstName.toLowerCase() !== 'facebook') {
          name = info[uid].firstName;
        }
      } catch {}
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅐𝐝𝐝𝐞𝐝 ${name} (${uid}) 𝐚𝐬 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    if (action === 'remove' || action === 'del') {
      if (!config.ADMINBOT.includes(uid)) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅣𝐡𝐢𝐬 𝐮𝐬𝐞𝐫 𝐢𝐬 𝐧𝐨𝐭 𝐚𝐧 𝐚𝐝𝐦𝐢𝐧.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      if (config.ADMINBOT.length === 1) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅒𝐚𝐧𝐧𝐨𝐭 𝐫𝐞𝐦𝐨𝐯𝐞 𝐭𝐡𝐞 𝐥𝐚𝐬𝐭 𝐚𝐝𝐦𝐢𝐧.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      config.ADMINBOT = config.ADMINBOT.filter(id => id !== uid);
      fs.writeFileSync(configPath, JSON.stringify(config, null, 2));
      
      if (global.config) {
        global.config = config;
      }
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅡𝐞𝐦𝐨𝐯𝐞𝐝 ${uid} 𝐟𝐫𝐨𝐦 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅤𝐬𝐚𝐠𝐞: 𝐚𝐝𝐦𝐢𝐧 [𝐚𝐝𝐝/𝐫𝐞𝐦𝐨𝐯𝐞/𝐥𝐢𝐬𝐭] [𝐮𝐢𝐝]\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
};
