// lodash-pari/commands/adminonly.js
module.exports = {
  config: {
    name: 'adminonly',
    aliases: ['onlyadmin', 'adminmode'],
    version: '1.0.0',
    permission: 1,
    prefix: true,
    premium: false,
    category: 'Admin',
    description: 'Toggle admin only mode',
    usage: 'adminonly [on/off]',
    credits: 'CHAND',
    cooldowns: 10,
    adminOnly: true
  },
  
  async run({ api, event, args, send, config }) {
    const fs = require('fs-extra');
    const path = require('path');
    const configPath = path.join(__dirname, '../../../config.json');
    let envConfig = fs.readJsonSync(configPath);
    
    const action = args[0]?.toLowerCase();
    
    if (action === 'on' || action === 'true' || action === 'enable') {
      envConfig.ADMIN_ONLY_MODE = true;
      fs.writeJsonSync(configPath, envConfig, { spaces: 2 });
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅐𝐝𝐦𝐢𝐧 🅞𝐧𝐥𝐲 🅜𝐨𝐝𝐞: 𝐄𝐍𝐀𝐁𝐋𝐄𝐃 ✅\n\n🅞𝐧𝐥𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐜𝐨𝐦𝐦𝐚𝐧𝐝𝐬 𝐧𝐨𝐰.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (action === 'off' || action === 'false' || action === 'disable') {
      envConfig.ADMIN_ONLY_MODE = false;
      fs.writeJsonSync(configPath, envConfig, { spaces: 2 });
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅐𝐝𝐦𝐢𝐧 🅞𝐧𝐥𝐲 🅜𝐨𝐝𝐞: 𝐃𝐈𝐒𝐀𝐁𝐋𝐄𝐃 ❌\n\n🅔𝐯𝐞𝐫𝐲𝐨𝐧𝐞 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐜𝐨𝐦𝐦𝐚𝐧𝐝𝐬 𝐧𝐨𝐰.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const status = envConfig.ADMIN_ONLY_MODE ? 'ENABLED' : 'DISABLED';
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅐𝐝𝐦𝐢𝐧 🅞𝐧𝐥𝐲 🅜𝐨𝐝𝐞: ${status}\n\n🅤𝐬𝐚𝐠𝐞: 𝐚𝐝𝐦𝐢𝐧𝐨𝐧𝐥𝐲 [𝐨𝐧/𝐨𝐟𝐟]\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
