// lodash-pari/commands/age.js

module.exports.config = {
  name: "age",
  version: "1.0.0",
  permission: 0,
  credits: "CHAND",
  description: "Calculate age from birthdate",
  prefix: true,
  premium: false,
  category: "utility",
  usages: "age [DD/MM/YYYY]",
  usage: "age [DD/MM/YYYY]",
  aliases: ["age2", "calculateage"],
  cooldowns: 5,
};

module.exports.run = async function ({ event, args, api, prefix }) {
  const moment = require("moment-timezone");
  const { threadID, messageID } = event;

  const input = args[0];
  if (!input) {
    return api.sendMessage(
      `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐞𝐧𝐭𝐞𝐫 𝐲𝐨𝐮𝐫 𝐛𝐢𝐫𝐭𝐡𝐝𝐚𝐭𝐞 𝐢𝐧 𝐭𝐡𝐢𝐬 𝐟𝐨𝐫𝐦𝐚𝐭:\n\n${prefix}𝐚𝐠𝐞 <𝐝𝐚𝐲/𝐦𝐨𝐧𝐭𝐡/𝐲𝐞𝐚𝐫>\n\n🅔𝐱𝐚𝐦𝐩𝐥𝐞: ${prefix}𝐚𝐠𝐞 01/01/2000\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`,
      threadID,
      messageID
    );
  }

  const [day, month, year] = input.split("/").map(i => parseInt(i));
  if (!day || day < 1 || day > 31) return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅘𝐧𝐯𝐚𝐥𝐢𝐝 𝐝𝐚𝐲.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);
  if (!month || month < 1 || month > 12) return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅘𝐧𝐯𝐚𝐥𝐢𝐝 𝐦𝐨𝐧𝐭𝐡.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);
  if (!year || year > moment().year()) return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅘𝐧𝐯𝐚𝐥𝐢𝐝 𝐲𝐞𝐚𝐫.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);

  const birthDate = moment.tz(`${day}/${month}/${year}`, "DD/MM/YYYY", "Asia/Karachi");
  if (!birthDate.isValid()) return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅓𝐚𝐭𝐞 𝐢𝐬 𝐧𝐨𝐭 𝐯𝐚𝐥𝐢𝐝!\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);

  const now = moment.tz("Asia/Karachi");
  const duration = moment.duration(now.diff(birthDate));

  const years = duration.asYears().toFixed(2);
  const months = duration.asMonths().toFixed(2);
  const weeks = (duration.asDays() / 7).toFixed(2);
  const days = duration.asDays().toFixed(2);
  const hours = duration.asHours().toFixed(2);
  const minutes = duration.asMinutes().toFixed(2);
  const seconds = duration.asSeconds().toFixed(0);

  const msg = `『🪽🅐𝐆𝐄 🅒𝐀𝐋𝐂𝐔𝐋𝐀𝐓𝐎𝐑🪽』

📅 𝐃𝐚𝐭𝐞 𝐨𝐟 𝐁𝐢𝐫𝐭𝐡: ${input}

💖 𝒀𝒆𝒂𝒓𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${years}
💚 𝑴𝒐𝒏𝒕𝒉𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${months}
💜 𝑾𝒆𝒆𝒌𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${weeks}
💙 𝑫𝒂𝒚𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${days}
💛 𝑯𝒐𝒖𝒓𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${hours}
🩷 𝑴𝒊𝒏𝒖𝒕𝒆𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${minutes}
🤍 𝑺𝒆𝒄𝒐𝒏𝒅𝒔 𝒑𝒂𝒔𝒔𝒆𝒅: ${seconds}

❖ ─────────────── ❖`;

  return api.sendMessage(msg, threadID, messageID);
};
