// lodash-pari/commands/ai.js
const axios = require('axios');
const fs = require('fs-extra');
const path = require('path');

module.exports = {
  config: {
    name: 'ai',
    aliases: ["model,","chatgpt"],
    version: '1.0.1',
    permission: 0,
    prefix: true,
    premium: false,
    category: 'ai',
    description: "Chat with AI (multiple models available)",
    usage: "ai <model> <question> | Models: chat, copilot, deepai, dolphin, gemini, gpt, kimi, note, public, soy, talkai, venice",
    credits: 'CHANDU',
    cooldowns: 5
  },

  async run({ api, event, args, send }) {
    const { threadID, messageID, senderID } = event;
    const model = args[0]?.toLowerCase() || 'chat';
    const question = args.slice(1).join(' ').trim();

    if (!question) {
      return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅤𝐬𝐚𝐠𝐞: 𝐚𝐢 <𝐦𝐨𝐝𝐞𝐥> <𝐪𝐮𝐞𝐬𝐭𝐢𝐨𝐧>\n\n🅜𝐨𝐝𝐞𝐥𝐬: 𝐜𝐡𝐚𝐭, 𝐜𝐨𝐩𝐢𝐥𝐨𝐭, 𝐝𝐞𝐞𝐩𝐚𝐢, 𝐝𝐨𝐥𝐩𝐡𝐢𝐧, 𝐠𝐞𝐦𝐢𝐧𝐢, 𝐠𝐩𝐭, 𝐤𝐢𝐦𝐢, 𝐧𝐨𝐭𝐞, 𝐩𝐮𝐛𝐥𝐢𝐜, 𝐬𝐨𝐲, 𝐭𝐚𝐥𝐤𝐚𝐢, 𝐯𝐞𝐧𝐢𝐜𝐞\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);
    }

    try {
      api.sendMessage("🤖 Thinking...", threadID);

      let result;
      switch(model) {
        case 'chat':
        case 'chathub':
          result = await chatHubAI(question);
          break;
        case 'copilot':
          result = await copilotAI(question);
          break;
        case 'deepai':
          result = await deepaiAI(question);
          break;
        case 'dolphin':
          result = await dolphinAI(question);
          break;
        case 'gemini':
          result = await geminiAI(question);
          break;
        case 'gpt':
          result = await gptAI(question);
          break;
        case 'kimi':
          result = await kimiAI(question);
          break;
        case 'note':
        case 'notegpt':
          result = await notegptAI(question);
          break;
        case 'public':
        case 'publicai':
          result = await publicAI(question);
          break;
        case 'soy':
        case 'soymascol':
          result = await soymascolAI(question);
          break;
        case 'talkai':
        case 'talk':
          result = await talkAI(question);
          break;
        case 'venice':
          result = await veniceAI(question);
          break;
        default:
          return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅤𝐧𝐤𝐧𝐨𝐰𝐧 𝐦𝐨𝐝𝐞𝐥. 🅐𝐯𝐚𝐢𝐥𝐚𝐛𝐥𝐞: 𝐜𝐡𝐚𝐭, 𝐜𝐨𝐩𝐢𝐥𝐨𝐭, 𝐝𝐞𝐞𝐩𝐚𝐢, 𝐝𝐨𝐥𝐩𝐡𝐢𝐧, 𝐠𝐞𝐦𝐢𝐧𝐢, 𝐠𝐩𝐭, 𝐤𝐢𝐦𝐢, 𝐧𝐨𝐭𝐞, 𝐩𝐮𝐛𝐥𝐢𝐜, 𝐬𝐨𝐲, 𝐭𝐚𝐥𝐤𝐚𝐢, 𝐯𝐞𝐧𝐢𝐜𝐞\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);
      }

      return api.sendMessage(`🤖 ${model.toUpperCase()}\n\n${result}`, threadID, messageID);
    } catch (error) {
      return api.sendMessage(`❌ Error: ${error.message}`, threadID, messageID);
    }
  }
};

async function chatHubAI(text) {
  const { data } = await axios.post('https://freeaichathub.com/chat', {
    message: text,
    model: "gpt-3.5-turbo",
    language: "en"
  }, {
    headers: {
      'authority': 'freeaichathub.com',
      'accept': '*/*',
      'content-type': 'application/json',
      'origin': 'https://freeaichathub.com',
      'referer': 'https://freeaichathub.com/',
      'user-agent': 'Mozilla/5.0 (Android 10; Mobile; rv:147.0) Gecko/147.0 Firefox/147.0'
    }
  });

  let result = data;
  if (typeof result === 'string') {
    try { result = JSON.parse(result); } catch (e) {}
  }
  return result.choices?.[0]?.message?.content || result.message || result.reply || result.content || "No response";
}

async function copilotAI(prompt) {
  const { data } = await axios.get("https://www.kitsulabs.xyz/api/v1/copilot", {
    params: { model: "gpt-5", prompt }
  });
  return data?.data?.text || "No response";
}

async function deepaiAI(input) {
  const FormData = require('form-data');
  const form = new FormData();
  form.append('chat_style', 'chat');
  form.append('chatHistory', JSON.stringify([{ role: 'user', content: input }]));
  form.append('model', 'standard');
  const res = await axios.post('https://api.deepai.org/hacking_is_a_serious_crime', form, { headers: form.getHeaders() });
  return (res.data?.output || res.data?.text || "No response").substring(0, 2000);
}

async function dolphinAI(message) {
  const { data } = await axios.post('https://chat.dphn.ai/api/chat', {
    messages: [{ role: 'user', content: message }],
    model: 'dolphinserver:24B',
    template: 'logical'
  }, {
    headers: {
      origin: 'https://chat.dphn.ai',
      referer: 'https://chat.dphn.ai/',
      'user-agent': 'Mozilla/5.0 (Linux; Android 15) AppleWebKit/537.36'
    }
  });
  return data.split('\n\n').filter(l => l?.startsWith('data: {')).map(l => JSON.parse(l.substring(6)).choices?.[0]?.delta?.content || '').join('') || "No response";
}

async function geminiAI(message) {
  const { data } = await axios.post('https://gemini.google.com/_/BardChatUi/data/assistant.lamda.BardFrontendService/StreamGenerate?hl=en-US', 
    new URLSearchParams({ 'f.req': JSON.stringify([null, JSON.stringify([[message, 0, null, null, null, null, 0], ["en-US"], ["", "", "", null, null, null, null, null, null, ""]])]) }).toString(),
    { headers: { 'content-type': 'application/x-www-form-urlencoded;charset=UTF-8' } }
  );
  const matches = Array.from(data.matchAll(/^\d+\n(.+?)\n/gm));
  if (matches.length >= 4) {
    const parsed = JSON.parse(JSON.parse(matches[3][1])[0][2]);
    return parsed[4][0][1][0].replace(/\*\*(.+?)\*\*/g, '*$1*') || "No response";
  }
  return "No response";
}

async function gptAI(text) {
  const response = await fetch("https://ai.soymaycol.icu/api/chat", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      modelId: "gpt-4o",
      messages: [{ role: "system", content: "You are a helpful AI assistant" }, { role: "user", content: text }]
    })
  });

  const reader = response.body?.getReader();
  const decoder = new TextDecoder();
  let result = "";
  if (reader) {
    while (true) {
      const { value, done } = await reader.read();
      if (done) break;
      const chunk = decoder.decode(value, { stream: true });
      const lines = chunk.split("\n");
      for (const line of lines) {
        if (line.startsWith("data: ")) {
          try {
            const data = JSON.parse(line.slice(6));
            if (data.type === "text-delta") result += data.delta;
          } catch (e) {}
        }
      }
    }
  }
  return result.trim() || "No response";
}

async function kimiAI(prompt) {
  const nonceRes = await axios.get("https://kimi-ai.chat/chat/");
  const nonce = nonceRes.data.match(/"nonce":"(.*?)"/)?.[1];
  if (!nonce) throw new Error("Failed to get nonce");

  const FormData = require('form-data');
  const form = new FormData();
  form.append('action', 'kimi_send_message');
  form.append('nonce', nonce);
  form.append('message', prompt);
  form.append('model', 'moonshotai/Kimi-K2-Instruct-0905');

  const res = await axios.post("https://kimi-ai.chat/wp-admin/admin-ajax.php", new URLSearchParams(Object.fromEntries(form)).toString(), {
    headers: { 'content-type': 'application/x-www-form-urlencoded' }
  });

  return res.data?.data?.message || "No response";
}

async function notegptAI(message) {
  const response = await axios.post("https://notegpt.io/api/v2/chat/stream", {
    message, language: "ace", model: "deepseek-reasoner", tone: "default", length: "moderate",
    conversation_id: "641eed40-0865-4dcf-9b90-39c868e4b710"
  }, { headers: { "Content-Type": "application/json" }, responseType: "stream" });

  let resultText = "";
  await new Promise((resolve, reject) => {
    response.data.on("data", (chunk) => {
      const lines = chunk.toString().split("\n");
      for (const line of lines) {
        if (line.startsWith("data:")) {
          try {
            const parsed = JSON.parse(line.replace(/^data:\s*/, ""));
            if (parsed.text) resultText += parsed.text;
          } catch {}
        }
      }
    });
    response.data.on("end", resolve);
    response.data.on("error", reject);
  });

  return resultText || "No response";
}

async function publicAI(question) {
  const id = Array.from({length: 16}, () => 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'[Math.floor(Math.random()*62)]).join('');
  const { data } = await axios.post('https://publicai.co/api/chat', {
    tools: {}, id, messages: [{ id, role: 'user', parts: [{ type: 'text', text: question }] }],
    trigger: 'submit-message'
  });

  return data.split('\n\n').filter(l => l && !l.includes('[DONE]')).map(l => JSON.parse(l.substring(6))).filter(l => l.type === 'text-delta').map(l => l.delta).join('') || "No response";
}

async function soymascolAI(question) {
  const { data } = await axios.post("https://ai.soymaycol.icu/api/chat", {
    messages: [{ id: Date.now().toString(), role: "user", content: question, experimental_attachments: [] }],
    modelId: "microsoft/Phi-4"
  }, { headers: { "Content-Type": "application/json" }, responseType: 'text' });

  return data.split("\n").filter((line) => line.startsWith("data:")).map((line) => {
    try {
      const json = JSON.parse(line.replace("data:", "").trim());
      return json.type === "text-delta" && json.delta ? json.delta : '';
    } catch { return ''; }
  }).join('') || "No response";
}

async function talkAI(query) {
  const sessionID = Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
  const sres = await axios.post("https://talkto-api.pfpmaker.workers.dev/api/send-message",
    { data: { botName: "Beebop", role: "AI Companion", message: `B::Hello!\nU::${query}` } },
    { headers: { "Content-Type": "application/json", "User-Agent": "Mozilla/5.0" } }
  );

  if (sres.data.response !== "Message sent") throw new Error("Failed to send message");

  const hash = sres.data.hash;
  const blob = Buffer.from(`B::Hello!\nU::${query}`).toString("base64");
  const getres = await axios.post("https://talkto-api.pfpmaker.workers.dev/api/get-message",
    { data: { hash, role: "AI Companion", sessionID, blob } },
    { headers: { "Content-Type": "application/json", "User-Agent": "Mozilla/5.0" } }
  );

  return getres.data.message || "No response";
}

async function veniceAI(question) {
  const { data } = await axios.post('https://outerface.venice.ai/api/inference/chat', {
    requestId: 'nekorinn',
    modelId: 'dolphin-3.0-mistral-24b',
    prompt: [{ content: question, role: 'user' }],
    systemPrompt: 'Jawablah menggunakan Bahasa Indonesia yang baik dan benar.',
    conversationType: 'text', temperature: 0.8, webEnabled: true, topP: 0.9, isCharacter: false, clientProcessingTime: 15
  }, { headers: { 'content-type': 'application/json', 'origin': 'https://venice.ai', 'referer': 'https://venice.ai/' } });

  const chunks = data.split('\n').filter(c => c.trim()).map(c => JSON.parse(c));
  return chunks.map(c => c.content).join('') || "No response";
}
