// lodash-pari/commands/alltag.js
module.exports = {
  config: {
    name: 'alltag',
    aliases: ['tagall', 'everyone'],
    version: '1.0.0',
    permission: 2,
    prefix: true,
    premium: false,
    category: 'Group',
    description: 'Tag all members in the group with proper mentions',
    usage: 'all [message]',
    credits: 'CHAND',
    cooldowns: 10,
    groupOnly: true
  },
  
  async run({ api, event, args, send, config }) {
    const { threadID, senderID } = event;
    
    let threadInfo;
    try {
      threadInfo = await api.getThreadInfo(threadID);
    } catch (error) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅣𝐡𝐞𝐫𝐞 𝐰𝐚𝐬 𝐚𝐧 𝐞𝐫𝐫𝐨𝐫 𝐠𝐞𝐭𝐭𝐢𝐧𝐠 𝐠𝐫𝐨𝐮𝐩 𝐢𝐧𝐟𝐨.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const adminIDs = threadInfo.adminIDs?.map(a => a.id) || [];
    const isGroupAdmin = adminIDs.includes(senderID);
    
    const members = threadInfo.participantIDs || [];
    const customMessage = args.join(' ') || 'Attention!';
    
    if (members.length === 0) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅝𝐨 𝐦𝐞𝐦𝐛𝐞𝐫𝐬 𝐟𝐨𝐮𝐧𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    let userInfoMap = {};
    try {
      userInfoMap = await api.getUserInfo(members);
    } catch (error) {
    }
    
    const BATCH_SIZE = 10;
    const DELAY_MS = 1500;
    
    await send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n📢 ${customMessage}\n─────────────────\n🅣𝐚𝐠𝐠𝐢𝐧𝐠 ${members.length} 𝐦𝐞𝐦𝐛𝐞𝐫𝐬...\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    
    await new Promise(r => setTimeout(r, 1000));
    
    for (let i = 0; i < members.length; i += BATCH_SIZE) {
      const batch = members.slice(i, i + BATCH_SIZE);
      
      let mentions = [];
      let text = '';
      
      for (const uid of batch) {
        let rawName = userInfoMap[uid]?.name || 'Member';
        if (rawName.toLowerCase() === 'facebook user' || rawName.toLowerCase() === 'facebook') {
          rawName = userInfoMap[uid]?.firstName || 'Member';
          if (rawName.toLowerCase() === 'facebook') {
            rawName = 'Member';
          }
        }
        const firstName = rawName.split(' ')[0];
        const tag = `@${firstName}`;
        
        mentions.push({ 
          id: uid, 
          tag: tag,
          fromIndex: text.length
        });
        
        text += `${tag} `;
      }
      
      text = text.trim();
      
      try {
        await api.sendMessage({ body: text, mentions }, threadID);
      } catch (error) {
      }
      
      if (i + BATCH_SIZE < members.length) {
        await new Promise(r => setTimeout(r, DELAY_MS));
      }
    }
    
    await new Promise(r => setTimeout(r, 500));
    await send.reply(` ━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ ${members.length} 🅣𝐚𝐠𝐠𝐞𝐝 𝐭𝐡𝐞 𝐦𝐞𝐦𝐛𝐞𝐫𝐬!\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
