// lodash-pari/commands/approve.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'approve',
    aliases: ['approvee', 'allow'],
    description: 'Approve or unapprove a group',
    usage: 'approve [threadID] / unapprove [threadID]',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, Threads, commandName }) {
    const { threadID } = event;
    const targetThread = args[0] || threadID;
    
    if (!/^\d+$/.test(targetThread)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐯𝐚𝐥𝐢𝐝 𝐭𝐡𝐫𝐞𝐚𝐝 𝐈𝐃.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const isUnapprove = commandName === 'unapprove' || args.includes('unapprove');
    
    if (isUnapprove) {
      if (!Threads.isApproved(targetThread)) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅣𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩 𝐢𝐬 𝐧𝐨𝐭 𝐚𝐩𝐩𝐫𝐨𝐯𝐞𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      Threads.unapprove(targetThread);
      
      let groupName = 'Unknown';
      try {
        const info = await api.getThreadInfo(targetThread);
        groupName = info.threadName || 'Unknown';
      } catch {}
      
      if (targetThread !== threadID) {
        api.sendMessage(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅣𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩 𝐡𝐚𝐬 𝐛𝐞𝐞𝐧 𝐮𝐧𝐚𝐩𝐩𝐫𝐨𝐯𝐞𝐝 𝐛𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`, targetThread);
      }
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅤𝐧𝐚𝐩𝐩𝐫𝐨𝐯𝐞𝐝 𝐠𝐫𝐨𝐮𝐩:${groupName} (${targetThread})\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━ `);
    } else {
      if (Threads.isApproved(targetThread)) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅣𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩 𝐢𝐬 𝐚𝐥𝐫𝐞𝐚𝐝𝐲 𝐚𝐩𝐩𝐫𝐨𝐯𝐞𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      Threads.approve(targetThread);
      
      let groupName = 'Unknown';
      try {
        const info = await api.getThreadInfo(targetThread);
        groupName = info.threadName || 'Unknown';
      } catch {}
      
      if (targetThread !== threadID) {
        const config = global.config || {};
        api.sendMessage(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅣𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩 𝐡𝐚𝐬 𝐛𝐞𝐞𝐧 𝐚𝐩𝐩𝐫𝐨𝐯𝐞𝐝! 🅨𝐨𝐮 𝐜𝐚𝐧 𝐧𝐨𝐰 𝐮𝐬𝐞 ${config.BOTNAME || 'the bot'}.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`, targetThread);
      }
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅐𝐩𝐩𝐫𝐨𝐯𝐞𝐝 𝐠𝐫𝐨𝐮𝐩: ${groupName} (${targetThread})\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
  }
};
