// lodash-pari/commands/approvallist.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'approvallist',
    aliases: ['aplist', 'aprovallist'],
    description: 'List all groups',
    usage: 'groupslist',
    category: 'Utility',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, send, Threads }) {
    const threads = Threads.getAll();
    
    if (threads.length === 0) {
      return send.reply('No groups in database.');
    }
    
    let msg = `GROUPS LIST (${threads.length})
─────────────────\n`;
    
    for (let i = 0; i < Math.min(threads.length, 20); i++) {
      const thread = threads[i];
      const status = thread.approved === 1 ? '✅' : '❌';
      const banned = thread.banned === 1 ? '🚫' : '';
      
      let groupName = thread.name || 'Unknown';
      try {
        const threadInfo = await api.getThreadInfo(thread.id);
        groupName = threadInfo.threadName || threadInfo.name || groupName;
      } catch (e) {}
      
      msg += `${i + 1}. ${status}${banned} ${groupName}\n   ID: ${thread.id}\n`;
    }
    
    if (threads.length > 20) {
      msg += `\n... and ${threads.length - 20} more groups`;
    }
    
    msg += `\n─────────────────
✅ = Approved | ❌ = Not Approved | 🚫 = Banned`;
    
    return send.reply(msg);
  }
};
