// lodash-pari/commands/autoban.js
const fs = require('fs-extra');
const path = require('path');

const spamTrackerPath = path.join(__dirname, './data/spam_tracker.json');

function getSpamData() {
  try {
    fs.ensureDirSync(path.dirname(spamTrackerPath));
    if (!fs.existsSync(spamTrackerPath)) {
      fs.writeJsonSync(spamTrackerPath, { users: {}, autobanned: {} });
    }
    return fs.readJsonSync(spamTrackerPath);
  } catch {
    return { users: {}, autobanned: {} };
  }
}

function saveSpamData(data) {
  try {
    fs.ensureDirSync(path.dirname(spamTrackerPath));
    fs.writeJsonSync(spamTrackerPath, data, { spaces: 2 });
  } catch (err) {
  }
}

function cleanExpiredBans(data) {
  const now = Date.now();
  let changed = false;
  
  for (const uid in data.autobanned) {
    if (data.autobanned[uid].expiresAt < now) {
      delete data.autobanned[uid];
      changed = true;
    }
  }
  
  if (changed) saveSpamData(data);
  return data;
}

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'autoban',
    aliases: ['spamban'],
    description: 'Toggle autoban for spammers (15 minutes ban)',
    usage: 'autoban [on/off/status/list]',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },
  
  spamCheck: async function(api, event, Threads, config) {
    const { senderID, threadID, body } = event;
    
    if (config.ADMINBOT?.includes(senderID)) return false;
    
    const settings = Threads.getSettings(threadID);
    if (!settings.autoban) return false;
    
    let data = getSpamData();
    data = cleanExpiredBans(data);
    
    if (data.autobanned[senderID]) {
      const remaining = Math.ceil((data.autobanned[senderID].expiresAt - Date.now()) / 60000);
      if (remaining > 0) {
        return { banned: true, remaining };
      }
    }
    
    const now = Date.now();
    const userKey = `${senderID}_${threadID}`;
    
    if (!data.users[userKey]) {
      data.users[userKey] = { messages: [], warns: 0 };
    }
    
    data.users[userKey].messages.push(now);
    data.users[userKey].messages = data.users[userKey].messages.filter(t => now - t < 20000);
    
    if (data.users[userKey].messages.length >= 7) {
      data.autobanned[senderID] = {
        bannedAt: now,
        expiresAt: now + (60 * 60 * 1000),
        reason: 'Spam detected',
        threadID
      };
      
      data.users[userKey] = { messages: [], warns: 0 };
      saveSpamData(data);
      
      return { newBan: true };
    }
    
    saveSpamData(data);
    return false;
  },
  
  isAutoBanned: function(senderID) {
    let data = getSpamData();
    data = cleanExpiredBans(data);
    
    if (data.autobanned[senderID]) {
      const remaining = Math.ceil((data.autobanned[senderID].expiresAt - Date.now()) / 60000);
      if (remaining > 0) return remaining;
    }
    return false;
  },
  
  async run({ api, event, args, send, Threads, config }) {
    const { threadID, senderID } = event;
    
    const threadInfo = await api.getThreadInfo(threadID);
    const adminIDs = threadInfo.adminIDs.map(a => a.id);
    const isGroupAdmin = adminIDs.includes(senderID);
    
    const action = args[0]?.toLowerCase();
    const settings = Threads.getSettings(threadID);
    
    if (!action || action === 'status') {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n  🅐𝐮𝐭𝐨𝐛𝐚𝐧 🅢𝐭𝐚𝐭𝐮𝐬\n─────────────────\n🅢𝐭𝐚𝐭𝐮𝐬: ${settings.autoban ? 'ON' : 'OFF'}\n🅑𝐚𝐧 𝐝𝐮𝐫𝐚𝐭𝐢𝐨𝐧: 60 𝐦𝐢𝐧𝐮𝐭𝐞𝐬\n🅐𝐮𝐭𝐨𝐛𝐚𝐧 𝐭𝐫𝐢𝐠𝐠𝐞𝐫𝐬 𝐰𝐡𝐞𝐧 𝐚 𝐮𝐬𝐞𝐫 𝐬𝐞𝐧𝐝𝐬 7+ 𝐦𝐞𝐬𝐬𝐚𝐠𝐞𝐬 𝐢𝐧 20 𝐬𝐞𝐜𝐨𝐧𝐝𝐬.\n🅤𝐬𝐚𝐠𝐞:\n- 𝐚𝐮𝐭𝐨𝐛𝐚𝐧 𝐨𝐧\n- 𝐚𝐮𝐭𝐨𝐛𝐚𝐧 𝐨𝐟𝐟\n- 𝐚𝐮𝐭𝐨𝐛𝐚𝐧 𝐥𝐢𝐬𝐭\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    if (action === 'on' || action === 'enable') {
      Threads.setSettings(threadID, { autoban: true });
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅐𝐮𝐭𝐨𝐛𝐚𝐧 𝐞𝐧𝐚𝐛𝐥𝐞𝐝.\n\n🅢𝐩𝐚𝐦𝐦𝐞𝐫𝐬 𝐰𝐢𝐥𝐥 𝐛𝐞 𝐚𝐮𝐭𝐨-𝐛𝐚𝐧𝐧𝐞𝐝 𝐟𝐨𝐫 60 𝐦𝐢𝐧𝐮𝐭𝐞𝐬.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (action === 'off' || action === 'disable') {
      Threads.setSettings(threadID, { autoban: false });
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅐𝐮𝐭𝐨𝐛𝐚𝐧 𝐝𝐢𝐬𝐚𝐛𝐥𝐞𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (action === 'list') {
      let data = getSpamData();
      data = cleanExpiredBans(data);
      
      const banned = Object.entries(data.autobanned);
      
      if (banned.length === 0) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅝𝐨 𝐮𝐬𝐞𝐫𝐬 𝐚𝐫𝐞 𝐜𝐮𝐫𝐫𝐞𝐧𝐭𝐥𝐲 𝐚𝐮𝐭𝐨-𝐛𝐚𝐧𝐧𝐞𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      let msg = `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅐𝐮𝐭𝐨-🅑𝐚𝐧𝐧𝐞𝐝 🅤𝐬𝐞𝐫𝐬 (${banned.length})\n─────────────────\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`;
      
      for (const [uid, info] of banned) {
        const remaining = Math.ceil((info.expiresAt - Date.now()) / 60000);
        let name = 'Unknown';
        try {
          const userInfo = await api.getUserInfo(uid);
          name = userInfo[uid]?.name || 'Unknown';
        } catch {}
        
        msg += `${name}\nUID: ${uid}\nRemaining: ${remaining} min\n\n`;
      }
      
      return send.reply(msg);
    }
    
    return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅤𝐬𝐚𝐠𝐞: 𝐚𝐮𝐭𝐨𝐛𝐚𝐧 [𝐨𝐧/𝐨𝐟𝐟/𝐬𝐭𝐚𝐭𝐮𝐬/𝐥𝐢𝐬𝐭]\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
  }
};
