// lodash-pari/commands/balance.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'balance',
    aliases: ['bal', 'money', 'wallet'],
    description: 'Check your balance',
    usage: 'balance [@user]',
    category: 'Economy',
    prefix: true
  },
  
  async run({ api, event, args, send, Currencies, Users }) {
    const { senderID, mentions } = event;
    
    let uid = senderID;
    
    if (Object.keys(mentions).length > 0) {
      uid = Object.keys(mentions)[0];
    } else if (args[0] && /^\d+$/.test(args[0])) {
      uid = args[0];
    } else if (event.messageReply) {
      uid = event.messageReply.senderID;
    }
    
    const name = await Users.getNameUser(uid);
    const wallet = Currencies.getBalance(uid);
    const bank = Currencies.getBank(uid);
    const total = Currencies.getTotal(uid);
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n🅑𝐀𝐋𝐀𝐍𝐂𝐄 𝐂𝐇𝐄𝐂𝐊\n─────────────────\n🅤𝐬𝐞𝐫: ${name}\n─────────────────\n🅦𝐚𝐥𝐥𝐞𝐭: $${wallet.toLocaleString()}\n🅑𝐚𝐧𝐤: $${bank.toLocaleString()}\n─────────────────\n🅣𝐨𝐭𝐚𝐥: $${total.toLocaleString()}\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
