// lodash-pari/commands/ban.js
const fs = require('fs-extra');
const path = require('path');

const spamTrackerPath = path.join(__dirname, './data/spam_tracker.json');

function getSpamData() {
  try {
    if (!fs.existsSync(spamTrackerPath)) {
      fs.writeJsonSync(spamTrackerPath, { users: {}, autobanned: {} });
    }
    return fs.readJsonSync(spamTrackerPath);
  } catch {
    return { users: {}, autobanned: {} };
  }
}

function cleanExpiredBans(data) {
  const now = Date.now();
  let changed = false;
  
  for (const uid in data.autobanned) {
    if (data.autobanned[uid].expiresAt < now) {
      delete data.autobanned[uid];
      changed = true;
    }
  }
  
  if (changed) {
    try { fs.writeJsonSync(spamTrackerPath, data, { spaces: 2 }); } catch {}
  }
  return data;
}

module.exports = {
  config: {
    cooldowns: 10,
    credits: 'CHAND',
    premium: false,
    permission: 1,
    version: '1.0.0',
    name: 'ban',
    aliases: ['banuser', 'spam'],
    description: 'Ban user or show ban list',
    usage: 'ban [list/uid] [reason]',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },

  async run({ api, event, args, send }) {
    const { mentions } = event;
    const action = args[0]?.toLowerCase();
    
    if (action === 'list' || action === 'all') {
      let data = getSpamData();
      data = cleanExpiredBans(data);
      
      const banned = Object.entries(data.autobanned);
      
      if (banned.length === 0) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅝𝐨 𝐮𝐬𝐞𝐫𝐬 𝐚𝐫𝐞 𝐜𝐮𝐫𝐫𝐞𝐧𝐭𝐥𝐲 𝐛𝐚𝐧𝐧𝐞𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      let msg = `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n⛔ 🅢𝐏𝐀𝐌 🅑𝐀𝐍𝐍𝐄𝐃 🅤𝐒𝐄𝐑𝐒 (${banned.length})\n─────────────────\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`;
      
      for (let i = 0; i < Math.min(banned.length, 15); i++) {
        const [uid, info] = banned[i];
        const remaining = Math.ceil((info.expiresAt - Date.now()) / 60000);
        
        let name = 'Unknown';
        try {
          const userInfo = await api.getUserInfo(uid);
          name = userInfo[uid]?.name || 'Unknown';
        } catch {}
        
        msg += `${i + 1}. ${name}\n   UID: ${uid}\n   Remaining: ${remaining} min\n`;
      }
      
      msg += `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n─────────────────\n🅤𝐬𝐞 $𝐮𝐧𝐛𝐚𝐧 [𝐮𝐢𝐝] 𝐭𝐨 𝐮𝐧𝐛𝐚𝐧\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`;
      
      return send.reply(msg);
    }
    
    let uid = '';
    let reason = '';
    
    if (Object.keys(mentions).length > 0) {
      uid = Object.keys(mentions)[0];
      reason = args.slice(1).join(' ') || 'Spam detected';
    } else if (args[0] && /^\d+$/.test(args[0])) {
      uid = args[0];
      reason = args.slice(1).join(' ') || 'Spam detected';
    } else if (event.messageReply) {
      uid = event.messageReply.senderID;
      reason = args.join(' ') || 'Spam detected';
    } else {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐔𝐈𝐃 𝐨𝐫 𝐮𝐬𝐞 $𝐛𝐚𝐧 𝐥𝐢𝐬𝐭\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const data = getSpamData();
    const now = Date.now();
    
    data.autobanned[uid] = {
      bannedAt: now,
      expiresAt: now + (60 * 60 * 1000),
      reason: reason,
      threadID: event.threadID
    };
    
    fs.writeJsonSync(spamTrackerPath, data, { spaces: 2 });
    
    let name = 'User';
    try {
      const userInfo = await api.getUserInfo(uid);
      name = userInfo[uid]?.name || 'User';
    } catch {}
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n⛔ 🅑𝐚𝐧𝐧𝐞𝐝 ${name} (${uid})\n🅡𝐞𝐚𝐬𝐨𝐧: ${reason}\n🅓𝐮𝐫𝐚𝐭𝐢𝐨𝐧: 60 𝐦𝐢𝐧𝐮𝐭𝐞𝐬\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
