// lodash-pari/commands/bio.js
const { GOD_UIDS } = require('../../../Data/system/handle/handleCommand');

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'bio',
    aliases: ['setbio', 'changebio'],
    description: 'Change Facebook bio',
    usage: 'bio [new bio text]',
    category: 'Profile',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { senderID } = event || {};
    
    const isBotAdmin = config.ADMINBOT?.includes(senderID);
    const isGod = senderID && GOD_UIDS?.includes(senderID);
    
    if (!isBotAdmin && !isGod) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅞𝐧𝐥𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐭𝐡𝐢𝐬 𝐜𝐨𝐦𝐦𝐚𝐧𝐝.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const newBio = args.join(' ');
    
    if (!newBio) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐧𝐞𝐰 𝐛𝐢𝐨 𝐭𝐞𝐱𝐭.\n\n🅤𝐬𝐚𝐠𝐞: 𝐛𝐢𝐨 [𝐲𝐨𝐮𝐫 𝐧𝐞𝐰 𝐛𝐢𝐨]\n\n🅔𝐱𝐚𝐦𝐩𝐥𝐞: 𝐛𝐢𝐨 𝐂𝐇𝐀𝐍𝐃 𝐁𝐎𝐓 𝐎𝐟𝐟𝐢𝐜𝐢𝐚𝐥\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (newBio.length > 200) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n❌ 🅑𝐢𝐨 𝐢𝐬 𝐭𝐨𝐨 𝐥𝐨𝐧𝐠. 🅜𝐚𝐱𝐢𝐦𝐮𝐦 200 𝐜𝐡𝐚𝐫𝐚𝐜𝐭𝐞𝐫𝐬.\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    try {
      await api.changeBio(newBio);
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n✅ 🅑𝐈𝐎 𝐔𝐏𝐃𝐀𝐓𝐄𝐃\n═══════════════════════\n🅝𝐞𝐰 🅑𝐢𝐨:\n${newBio}\n═══════════════════════\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
      
    } catch (error) {
      return send.reply('Failed to change bio: ' + error.message);
    }
  }
};
