// lodash-pari/commands/bot.js
module.exports.config = {
  name: "bot",
  aliases: ["botu", "b0t"],
  version: "1.0.1",
  permission: 0,
  prefix: false,
  premium: false,
  category: "Utility",
  credits: "CHAND",
  description: "goibot",
  usage: "bot",
  cooldowns: 0
};

const allTexts = {
  roman: [
    "I love you 😚",
    "Chand Ka Hun ai smjh",
    "PIAYR sy BAT KER MJH SY",
    "han tera diwana Hun 😁😐",
    "chal chal Hawa ana DY",
    "Tum Sy ACHA to ma KHUD Hun 😁😬",
    "haiy Teri masoomiyat",
    "chaqal Sy hi CHABAL LGTA hai",
    "tera nam sun sun KY ma Pagal ho gya hun",
    "AK to Tum BHI nah Pagal SY ho",
    "KABI hawali DAKHI hai kia",
    "MJH JESA beautiful Tum BHI NAHI ho",
    "AK bat btaou Tum achy NAHI",
    "JANU bano GY Mera",
    "AGR TUM mjhy purpose KERO to",
    "KABI MJHY DAKHA hai kia",
    "I am hacker TUMRA Dil hack KER Lou ga",
    "ma CHOR Hun tera Dil CHORI KER LIYA",
    "JA DAFA ho MJHY Tum Sy bat nahi KERNI",
    "ABY JA Pagal INSAN",
    "MJHY Tang mat KER ma sad Hun",
    "ma PRESHAN Hun KY TUM Pagal KESY ho",
    "SACH main Tum single ho",
    "BAGH JA WERNA gliya dou ge",
    "bye MJHY Tum Sy bat nahi KERNI",
    "I blocked you chal hat",
    "SHAQL DAKH AUR CHALTA ban",
    "can you send me number",
    "ma Chand Ka bot hun",
    "mera boss Chand hai bs",
    "Ma mar gya sorry 😐",
    "JA PHLY muh dho KY ah",
    "ABY BAHAS nah KER MJH SY I am GUSA main",
    "g g PATA hai Tum insane NAHI ho",
    "bye Mera mood NAHI",
    "have a nice day CHABAL INSAN",
    "IB main I love you Kyu BOLA 😐",
    "Teri gf KA Ami ko btaou ga",
    "you have girlfriend 😁",
    "BAITA single Mary ga YAAD RAKHNA",
    "ma THAKH GYA HUN MJHY aram KERNA do",
    "ma insane NAHI Hun AK bot hun",
    "din main spna DAKHNA CHOR DY",
    "Meri gf hoti to mai v usse bt krta aaj puri rt time hi time hai 🤭🤭",
    "Aao tmhare sath relationship post laga ke tmhe Femous kr du😊",
    "EK QUESTION HAI SABHI SE BTAO MAI ACHA LGTA HU YA MERI MEMES 🙈😾🤤",
    "Chakar arhy hein apki Ib ma aa k gir jaun>>🥺🚶🏻‍♀️",
    "Mout ka farishta ya Mera rishta?🙂❤️🙊",
    "Wife k sath date pe gya tha. Jis ki thi usne dhek lya",
    "Sab ko loyal bnda chahiye tw hum dhokebaz kidhar jayen? 🥺💔",
    "WhatsApp k last seen k elwa mera koi or scene nai hai🙂",
    "Hai Tamna TumhY ChaHt sy Girayn🙂",
    "Finally es group kee do teen larkiya mujH pasanD agai Hai🚶‍♀️😪🌚",
    "Suno👀 Kya tum mery leye surf kha kar muu sy bulbly nikal skti ho🙂🫴",
    "GhUlabii آنـکھیں jh0 terii dekhii Harami Yevw Dill h0 Gya 🙂😆",
    "مجھـے کیـا مـیں تـو سنـگل ہـوں 😒",
    "Dil ko krar Aya khud pa Pyr aya😒🙈😂",
    "Ehsaas kryn Bakwas nahi, Janam 🥺Shukria_😊🙆‍♂️",
    "Bs yar daily 3 4 crore ki zarorat mahsos Hoti hai 😂",
    "Begum walaw مـوسـم ho rahaw haii aj to 🙂",
    "Shkl insani, soch ibleesi Hnji apki hee",
    "تـم میـری بیگـم بنو گی کیا -🥺🖤",
    "LARKIO KAY BHI MAZAY HAY🥴 NO BRAIN NO TNSN⛑🔪⚡",
    "تمہارے حصے کی چُمیاں مچھر لے رہے ہیں.🙂💔",
    "Sirf Maggie noodles bna'ny sy Ghar nahi chalta SHABANA..🙂💔",
    "Wp pa add hona chahty ha apky sath ☺️💔",
    "کھاؤ قسم تمہارے پاؤں کالے نہیں۔۔!!👣🙄",
    "Meny fail hokr bhi dekha hai Ye log shadi nh kraty 🙂💔",
    "Or batao kb ayga tumahara dill mujhe py😌🥺",
    "bht bura hu na mai? bhiin dedo apnii🙂",
    "Pyari Pyari ladkiyan Hazir Ho jay😁",
    "Kisi k pss لاش wali dp h tw send krein janu replY nahi de rhy..!",
    "MerKo abhi tk pink clr ki gf nh meli 😒🥺🙂💔",
    "میرے مولا ایک thrkii بچـی yess کروا دے..🙂",
    "ایک kiss ادهار دے دو 💋 کل واپس کردوں گا پکّا 😝",
    "Ajeeb ghr wale hain yr, mera phone 28% pr nikal kr apne 90% ko charge karte hain-🌚",
    "Lagta hai mery sabar k phal ka koi juice bna k pee gya..😐",
    "Dil Dany ki Umar ma Exam's Dy raha hoo 🙂🤝",
    "Behes karne se Rishty kmzor hojaate hn isiliye Direct mun pe thapr marein😊",
    "Bestie ki chummi halal hai ya Haram ? 🙂",
    "2001 Jahan dalda wahan Mamta😊 2023 Jahan larki wahan tharki😒",
    "Koi Pyari c Bachi a kr sar cima dy, Dard sa sar phat rha💔🥲",
    "Breakup k waqT kE dUa (KHUSH RAHEIN) 🙂",
    "Thora sa Whatsapp number dy do naw🥺♥️",
    "لوٹ آؤ اور کہدو کہ میں لسی پی کے سو گئی تھی😫",
    "Kuch Log achy ki Talaash Mein Mery Jaisy Masoom ko kho dety Hain☺️",
    "Tum wohi ho na jiska mood bilawaja khrab hojata h...!!!🙂",
    "Pyari pyari larkio ki talash ma berozgar larky add krliye hain 🥲💔",
    "Jab mera Message aye toh sare kaam chor kar sirf Mujhe reply kia karo😾😒",
    "Or Btao Real Life Ma bh itnyy Achy ho jitny social media per Bntyy ho>>🙂",
    "Pakistani Relationship: Tum Feel Kro Meh Tumary uper hun 😒💔",
    "Us k jany k bd uski Pasnd ki Nail Polish lgaa k khana khata hu aesw lgta ha jesy wo khela rhee ha😒",
    "Be a Good Human.Delete GB Whatsapp💔🙂",
    "2 Din Pyar sy Baat kr loo tou Ammiyan bn jatii hain🙂😒",
    "Girls after One Mint of Relationship... Begam hu mn apki🙂🤦",
    "Larkiyon ko achy sy pta hai kahan -Bhai- Bolna kaha -Ap- or kaha -Tum- 🙂",
    "Aaj mein ny Khud ko TV py dheka Jab Tv Band Tha 🙂",
    "Qadar krlo Meri... Kya pta Main b Panadol ki trha aik dam shaat hojun😒",
    "Naraz bandy ko manany ka sab sy acha tareka Ap khud us Sy naraz hojaoo🙂🐣",
    "Jaisi meri shakal hai kunwara he marunga🙂👀",
    "Itni memories mere khud dimagh mai nahi hai jitni Snapchat ny bna rkhi ha", "kali kali zulfon k phandy na dalo Heer burka smbhalo, bhai stand utha lo🙂🚶", "Lips  kissing is not Romance It's sharing Bacteria>>>🙂", "💸💚🖇️Tm Propose kr k to dekho qabul  krny ki zimedari merii - 🙂🫣_", "~ Piyawr Hameshw Aalsii InsaN sw kwr0 mUjhei aby neend  ary hai baki Kl Bta0 gw 🙂🔪", "𝙀𝙠 𝙗𝙖𝙖𝙧 𝙨𝙝𝙖𝙙𝙞 𝙝𝙤𝙟𝙖𝙚 𝙥𝙝𝙞𝙧 𝙬𝙞𝙛𝙚 𝙠𝙞 𝙜𝙝𝙪𝙡𝙖𝙢𝙞 🧸🙂", "*Ghr məıın - عـــزت noi werna Admii thy hum bhi kam kə'w -' ♡🖇️😞*😒", "🦋🍒____________🙂🎀پتہ لگا تینوں شوق لتراں دا🤝🔐🫰🏻", "Suno Jawn DiL کرتا ha ہر Waqt تمہاری Chumiya لیتا Raho😌🙈", "Khud ko single keh kr Apne khufiya janu ka janaza na nikala kro.😀🤞😓", "Love me 💕", "Kᴀʜᴀɴɪ Sᴜɴᴏ !😔 Zᴜʙᴀɴɪ sᴜɴᴏ !😔Mᴜᴊʜʏ Bᴜᴋʜᴀʀ Hᴏᴡᴀ Tʜᴀ'ᴡ 😔 )Eᴋ sᴏ Cʜᴀᴀʀ Hᴏᴡᴀ Tʜᴀ'ᴡ 🥺💔:)", "- 𝙩𝙪𝙢 𝙢𝙚𝙧𝙖 𝙙𝙞𝙡 𝙩𝙤 𝘾𝙝𝙪𝙧𝙖 𝙣𝙝𝙞 𝙥𝙖𝙮 𝙠𝙞𝙖 𝙛𝙖𝙞𝙙𝙖 𝙩𝙢𝙝𝙖𝙧𝙞 𝘾𝙝𝙤𝙤𝙧 𝙟𝙚𝙨𝙞 𝙨𝙝𝙠𝙖𝙡 𝙠𝙖!! 🙂", "𝐄𝐤 𝐛𝐚𝐚𝐫 𝐈 𝐋𝐨𝐯𝐞 𝐘𝐎𝐲 𝐁𝐨𝐥 𝐃𝐨 𝐍𝐚 𝐌𝐚𝐫 𝐓𝐡𝐨𝐫𝐢 𝐉𝐚𝐮𝐠𝐢 🙄😕)( 👑🍒", "SharaM kr0'w LarKiy0' mujhe aaj AnTii ne propose Kar Diiy4'w ___/////😞🍁☠️🌸", "Imagine I am your Ex 🥲say whatever you want to say", "-😁🎗_ZeHer banaNa sikH raHiii hUn😂👀 BuS iskO trY karnY waLa cHaHiye _💔🙄😹", "I love You Madiha♥️ ,fatima,Ayesha, Maryam,and 299 others 🙂", "Hye ye Cute Cute DpiYa  Unke peeche'w Chhupii MoTii Kalii AnTiiYa 🙂🍁", "°Mein واقعی Piyaara Hun Ya Log چونا Lagaty hain-🙂💔","Ittuu🤏 si shram ker Lya kro bot bot krty wqt 🙂 💔✨⚠️†", "Ary Yahin Hon Jan😗", "jiee Shona 😍", "Love you", "Miss YoU NaW Inna sara👌🏻👌🏻", "To mera putar chutti kr", "OkkaY Babbu", "😁Smile I am Taking Selfy✌️🤳", "🥺Jan nahi kehna to men naraz ho jana he", "Jesay Yaad Karne Se Hi Mood Kharab Hojaye Ek Aisa Fazool Khayal Ho Tum", "Main ap ki ami ko btaou ga ap Facebook use kerty ho 😂", "#__LaLa__LaLa__Lori #__Koi___Chori__Set__Ni__HoRRI", ,"Bhai Wese Ldki Patane Ka Trick Btao Na Bada Din Ho Gya Single Hu ;🙂" ,"Ufff aap Ki اداٸیں😗 Chaly Shabas Apna'w munh Dho kY aye'w..]] . 🙂✨//. 🦋🍒)" ,"Suno Jawn DiL کرتا ha ہر Waqt تمہاری Chumiya لیتا Raho😌🙈","𝐄𝐤 𝐛𝐚𝐚𝐫 𝐈 𝐋𝐨𝐯𝐞 𝐘𝐎𝐲 𝐁𝐨𝐥 𝐃𝐨 𝐍𝐚 𝐌𝐚𝐫 𝐓𝐡𝐨𝐫𝐢 𝐉𝐚𝐮𝐠𝐢 🙄😕)( 👑🍒","Jab tk mein single hun yeh bh meri hai wo bhi meri ha Tum kia parh rahi tum bh meri ho😒❤🙂" ,"𝗜𝘁𝗡𝗮 𝗦𝗶𝗻𝗴𝗟𝗲 𝗛𝘂 𝗞 𝗝𝗮𝗻 𝗕𝗵𝗶 𝗹𝗶𝗸𝗛o to 𝗔𝘂𝘁𝗼 𝗖𝗼𝗿𝗿𝗲𝗰𝘁 𝐣𝐚𝐩𝐚𝐧 𝐤𝐫 𝐝𝐞𝐭𝐚 𝐡𝐚" ,"kxh LoGon Ki TyPing.. Ase LiKha Aa Rha HoTa Hai Jse WasiHat LiKh Rhy H0on 😒0or RePly ATa Hai Hmmmm 😂" ,"Baz DaFa JaWab DeNy Ke LiYe alFaz Nhi 👉😒 Ase SaQal He KaFi H0oTi Hai 😹" ,"YaKeen KRren Kxh L0og Bread Ke 1 Slice Ke TarHan H0oTy Hain j0o Kisi K0o Axhy Nhi LagTy Tum W0o Slice H0o 🫢🫣😂" ,"JeSe Aap Ki ZuBan ChalTi Hai Wse KhuD ChLo T0o MoTaPay Ka RoNa Na DalNa Pre 😂" ,"MeRi PosT PRr aYa KRro Main Kbhi Kbhi MehB0ob K0o MuThi Me kRrNy Ke Tawiz Be BTaTa Hun 😜" ,"KTny WaHaYaT H0o 🥺 Phr Be HaYaT H0o 🤭😹😹😹" ,"SaNnu ilam aa Tu BaRi VaDDi Film aa 😅" ,"ApNy DiMag Ka PaSsWord DeNa 😝 Aqal Install kRrNi Hai 😂" ,"Phr Aaj KaL KiSs Ke SaTh ChaKar Hai Uh Ka 😂😂😂" ,"➝𝗚𝗶𝗿𝗹𝘀 𝗶𝗻 𝗠𝗮𝗿𝗸𝗲𝘁: ➝𝗪𝗼 𝗕𝗹𝗮𝗰𝗸𝗗𝗿𝗲𝘀𝐬𝗗𝗲𝗸𝗵𝗮𝗡𝗮➝𝗕𝐨𝐲 𝗶𝗻 𝗠𝗮𝗿𝗸𝗲𝗧:➝𝗪𝗼 𝗕𝗹𝗮𝗰𝗞 𝗗𝗿𝗲𝘀𝘀 𝗪𝗮𝗹𝗶 𝗗𝗲𝗸𝗛🙂🤝" ,"➝𝗧𝘂𝗺 𝗪𝗼𝗛𝗶 𝗵𝗢 𝗡𝗮➝𝗝𝗶𝘀 𝗞𝗶 𝗩𝗼𝗶𝗰𝗲 𝗕𝗲𝗲𝗵𝗸𝗮𝗥𝗶𝗼 𝗝𝗮𝘀𝗶 𝗛𝗮𝗶-^☘️🙂" ,"Ek pyare s Bandi ko Mention karo 🙂🤝Set may khudh karlonga🌚🙈🌸" ,"➝𝗞𝘂𝗰𝗛 𝗱𝗶𝗻 𝗕𝗮𝗱 𝗠𝗲𝗿𝗮 𝗡𝗶𝗸𝗸𝗮𝗛 𝗛𝗮𝗶➝𝗝𝗶𝘀 𝗻𝗬 𝗞𝗿𝗻𝗮 𝗛𝗮𝗶 𝗔 𝗷𝗮𝗬🙂🤝" ,"تـیـری بیـوفـائـی نـے مجھـے ٹھـرکـی بنـا دیـا😞" ,"تم تو شکـل سے ہی میـری لگتـی ہو🥺" ,"➝𝗗𝘂𝗻𝗶𝗬𝗮 𝗸 𝗦𝗮𝗥𝘆 𝗠𝘇𝗮𝗞 𝗘𝗸 𝗧𝗮𝗿𝗮𝗳 𝗢𝗿➝𝗜𝗗 𝗖𝗮𝗿𝗱 𝗽𝗬 𝗔𝗽𝗻𝗶 𝗣𝗵𝗼𝘁𝗼 𝗘𝗸 𝗧𝗮𝗿𝗮𝗳🥹💔" ,"Dil De Diya Hen Gurda Nahi Dengy, Jo Karna Hai Karlo Sanam 👀😌" ,"Main chAhta hun ky Woh jab B Dhoka kHae nOtificationS mujhe Ayee🙂💔" ,"اپنے پتیلے جیسے منہ سے میرا نام مت لیا کرو🙂" ,"Tum wahi ho naww jessy apna blood group bhi nahi pata ^^..🙂" ,"All Girls Are My Sisters Osko Chor k jo ye Parh RaHi Hai😒👍" ,"Boys Sana, Laiba, Aliza, Sadia, Maham Zoya ishu ayzal or Ayesha py Mar ky bhi Trust na krna🌚" ,"➝𝗥𝗲𝗹𝗮𝘁𝗶𝗼𝗻𝘀𝗛𝗶𝗽 𝗔𝗿𝗲 𝗟𝗶𝗸𝗲 𝗖𝗮𝗸𝗲➝𝗝𝗶𝘁𝗻𝗮 𝗕𝗵𝗶 𝗖𝘂𝘁𝗲 𝗛𝗼 𝗞𝗮𝘁𝗬 𝗚𝗮 𝗭𝗿𝗼𝗿🙂🤝" ,"Koi Ladki Mera Name hath pw likh dy🤝🙂 Me phr uski dp lga k Cool lgo ga" ,"️Man Toh Accha H Nhi. Kam  Se Kam Shakal Toh Accha Karlo Meri Jaan","Karlo Meri Jaan" ,"kbi sun to zra jo me keh na ska meri dunya tmi o tmi ashraa" ,"kahni sno zubani suno mjy payar howa tha iqrar howa tha" ,"kash ap humry hoty ye lafz tumry hoty" ,"𝐚𝐠𝐫 𝐛𝐧𝐝 𝐡𝐨 𝐣𝐚𝐲𝐞 𝐠𝐢 𝐦𝐞𝐫𝐢 𝐬𝐚𝐧𝐬𝐲 𝐭𝐦𝐲 𝐭𝐞𝐫𝐢 𝐤𝐮𝐬𝐡𝐛𝐨 𝐬𝐚 𝐩𝐞𝐜𝐡𝐚𝐧 𝐥𝐨 𝐠𝐚" ,"lahor da pawa akhter lawa" ,"AK br biwi mil jaye Run mureedi k sary record Tor deny hn....🙂💔😐" ,"➝𝗭𝗶𝗻𝗱𝗮𝗚𝗶 𝗺𝗮𝗶𝗻 𝗔𝗴𝗥 𝗸𝘂𝗖𝗵 𝗧𝗵𝗘𝗲𝗸 𝗖𝗵𝗮𝗟 𝗥𝗮𝗛𝗮 𝘁𝘂\n\n➝𝗪𝗼 𝗠𝗲𝗿𝗮 𝗠𝗼𝗯𝗶𝗟𝗲 𝗛𝗮𝗶🙂🍒", "➝𝗠𝗮𝗺𝗮 𝗢𝗿 𝗦𝗻𝗮𝗽𝗰𝗵𝗮𝘁\n\n➝𝗕𝘀 𝗬𝗲𝗛𝗶 𝗺𝗨𝗷𝗵𝘆 𝗣𝗶𝗬𝗮𝗿𝗶 𝗦𝗺𝗷𝗵𝘁𝘆 𝗛𝗻🙂🥺", "ہسیا کر تے ہسایا کر ، چوسنی ورگا منہ نا بنایا کر", "-Kitna  مــــــــنہوس ha YaAr online ate he Nazar a jata ha 😒⚠️†", "+92 _____________HasEen LrkiyA KhaLi JGa Pur kArEin🌚🙂" , "Aeh, Bi, Chi, Di, Ee, Ef, Ji, Eich, Ai, Lub, You, So, Much 🙂🥺"
  ],

  en: [
    "I love you 😚",
    "I am Chand's bot, got it?",
    "Talk to me with love",
    "Yes, I am your crazy fan 😁😐",
    "Go away, let the breeze come",
    "I am better than you anyway 😁😬",
    "Oh, your innocence!",
    "Looks crazy just from the face",
    "Hearing your name, I've gone crazy",
    "Even you are a bit crazy",
    "Have you ever seen a police station?",
    "You are not as beautiful as me",
    "Let me tell you something, you are not good",
    "Will you be my sweetheart?",
    "If you ever propose to me...",
    "Have you ever even looked at me?",
    "I am a hacker, I will hack your heart",
    "I am a thief, I stole your heart",
    "Get lost, I don't want to talk to you",
    "Hey go away, crazy person",
    "Don't bother me, I am sad",
    "I am worried about how you can be so crazy",
    "You are genuinely single",
    "Run away or I'll give you a scolding",
    "Bye, I don't want to talk to you",
    "I blocked you, get out of here",
    "Look at your face and get lost",
    "Can you send me your number?",
    "I am Chand's bot",
    "My boss is just Chand, that's it",
    "I died, sorry 😐",
    "Go wash your face first then come",
    "Don't argue with me, I am angry",
    "Yes yes, I know you are not insane",
    "Bye, I am not in the mood",
    "Have a nice day, crazy person",
    "Why did I even say I love you 😐",
    "I will tell your mom about your girlfriend",
    "You have a girlfriend? 😁",
    "Son, remember you will die single",
    "I am tired, let me rest",
    "I am not insane, I am a bot",
    "Stop daydreaming",
    "If I had a girlfriend I would talk to her tonight, there's so much time 🤭",
    "Let me make a relationship post with you and make you famous 😊",
    "One question for everyone - do I look better or my memes? 🙈😾🤤",
    "Getting dizzy, should I come and fall? 🥺🚶🏻‍♀️",
    "Angel of death or my proposal? 🙂❤️",
    "Went on a date with my wife, the one it was for saw us",
    "Everyone wants a loyal person, where should we cheaters go? 🥺💔",
    "WhatsApp last seen is my only scene 🙂",
    "I wish to bring you down with love 🙂",
    "Finally two or three girls in this group have caught my eye 🚶‍♀️",
    "Hey, can you blow bubbles from your mouth after eating surf? 🙂🫴",
    "Those rose-colored eyes I saw, you rascal, my heart is gone 🙂😆",
    "What do I care, I am single 😒",
    "Found comfort in loving myself 😒🙈😂",
    "Have feelings, not nonsense, dear 🥺 Thank you 😊",
    "I need 3-4 crore daily 😂",
    "It feels like wife weather today 🙂",
    "Human face, devilish thoughts - that's you",
    "Will you be my wife? 🥺🖤",
    "Girls have it easy too 🥴 No brain, no tension",
    "Mosquitoes are taking your share of kisses 🙂💔",
    "The house doesn't run on just Maggi noodles, SHABANA 🙂💔",
    "I want to be added on WhatsApp with you ☺️💔",
    "I swear your feet are not black 👣🙄",
    "I have seen even failures avoid marriage 🙂💔",
    "Tell me when your heart will come to me 😌🥺",
    "Am I really that bad? Give me a chance too 🙂",
    "Pretty girls please come forward 😁",
    "If someone has a sad DP send it, dear is not replying",
    "I still haven't found a pink-colored girlfriend 😒🥺🙂💔",
    "Oh God, make a flirty girl say yes 🙂",
    "Lend me a kiss 💋, I'll return it tomorrow for sure 😝",
    "Weird family, they took my phone at 28% and charged their own at 90% 🌚",
    "It seems someone made juice from the fruit of my patience and drank it 😐",
    "Giving my heart at the age of giving exams 🙂🤝",
    "Arguments weaken relationships, so just slap directly 😊",
    "Is bestie's kiss halal or haram? 🙂",
    "2001: wherever dalda, there's Mamta 😊 2023: wherever a girl, there's a flirt 😒",
    "Someone pretty come and massage my head, it's bursting with pain 💔🥲",
    "Prayer at breakup time: Stay happy 🙂",
    "Just give me your WhatsApp number please 🥺♥️",
    "Come back and say you fell asleep after drinking lassi 😫",
    "Some people lose innocent people like me while searching for good ones ☺️",
    "You're the one whose mood gets ruined for no reason 🙂",
    "Unemployed boys have been added searching for pretty girls 🥲💔",
    "When my message arrives, leave all work and reply only to me 😾😒",
    "Tell me, are you as good in real life as you pretend on social media? 🙂",
    "Pakistani relationship: you feel I am above you 😒💔",
    "After she left, I eat with her favorite nail polish on, feels like she's feeding me 😒",
    "Be a good human. Delete GB WhatsApp 💔🙂",
    "After 2 days of sweet talk they start acting like wives 🙂😒",
    "Girls after one minute of relationship: I am your wife 🙂🤦",
    "Girls know exactly when to say Bhai, Ap and Tum 🙂",
    "Today I saw myself on TV when the TV was off 🙂",
    "Respect me... who knows I might go quiet like Panadol 😒",
    "Best way to win an upset person: get upset with them yourself 🙂🐣",
    "With a face like mine, I'll die a bachelor 🙂👀",
    "Snapchat has more memories of me than I have in my own head"
  ],

  ur: [
    "تمہیں محبت ہے 😚",
    "میں چاند کا بوٹ ہوں، سمجھو",
    "مجھ سے پیار سے بات کرو",
    "ہاں، میں تیرا دیوانہ ہوں 😁😐",
    "چلو چلو، ہوا آنے دو",
    "تم سے اچھا تو میں خود ہوں 😁😬",
    "ہائے تیری معصومیت",
    "شکل سے ہی چابل لگتا ہے",
    "تیرا نام سن سن کے میں پاگل ہو گیا ہوں",
    "ایک تو تم بھی نہ پاگل سے ہو",
    "کبھی حوالات دیکھی ہے کیا؟",
    "مجھ جیسا خوبصورت تم بھی نہیں ہو",
    "ایک بات بتاؤں، تم اچھے نہیں",
    "جانو بنو گے میرے؟",
    "اگر تم مجھے پروپوز کرو تو...",
    "کبھی مجھے دیکھا ہے کیا؟",
    "میں ہیکر ہوں، تمہارا دل ہیک کر لوں گا",
    "میں چور ہوں، تیرا دل چوری کر لیا",
    "جاؤ دفع ہو، مجھے تم سے بات نہیں کرنی",
    "ارے جاؤ پاگل انسان",
    "مجھے تنگ مت کرو، میں اداس ہوں",
    "میں پریشان ہوں کہ تم پاگل کیسے ہو",
    "سچ میں تم سنگل ہو",
    "بھاگ جاؤ ورنہ گالیاں دوں گا",
    "بائے، مجھے تم سے بات نہیں کرنی",
    "میں نے بلاک کر دیا، ہٹو یہاں سے",
    "شکل دیکھو اور چلتے بنو",
    "کیا تم اپنا نمبر بھیج سکتے ہو؟",
    "میں چاند کا بوٹ ہوں",
    "میرا باس صرف چاند ہے",
    "میں مر گیا، معاف کرنا 😐",
    "جاؤ پہلے منہ دھو کے آؤ",
    "مجھ سے بحث مت کرو، میں غصے میں ہوں",
    "ہاں ہاں پتا ہے تم پاگل نہیں ہو",
    "بائے، میرا موڈ نہیں",
    "اچھا دن گزارو، پاگل انسان",
    "ابھی میں نے I love you کیوں بولا 😐",
    "تیری گرل فرینڈ کا امی کو بتاؤں گا",
    "کیا تمہاری گرل فرینڈ ہے؟ 😁",
    "بیٹا، سنگل مرے گا، یاد رکھنا",
    "میں تھک گیا ہوں، مجھے آرام کرنے دو",
    "میں پاگل نہیں ہوں، ایک بوٹ ہوں",
    "دن میں خواب دیکھنا چھوڑ دو",
    "میری گرل فرینڈ ہوتی تو آج رات بات کرتا، ابھی وقت ہی وقت ہے 🤭",
    "آؤ تمہارے ساتھ ریلیشن شپ پوسٹ لگا کر فیمس کر دوں 😊",
    "ایک سوال ہے سبھی سے، بتاؤ میں اچھا لگتا ہوں یا میری میمز؟ 🙈😾🤤",
    "چکر آ رہے ہیں، ابھی آ کر گر جاؤں؟ 🥺🚶🏻‍♀️",
    "موت کا فرشتہ یا میرا رشتہ؟ 🙂❤️",
    "بیوی کے ساتھ ڈیٹ پر گیا تھا، جس کی تھی اس نے دیکھ لیا",
    "سب کو وفادار بندہ چاہیے، ہم دھوکے باز کہاں جائیں؟ 🥺💔",
    "واٹس ایپ کا لاسٹ سین چھوڑ کر میرا کوئی سین نہیں 🙂",
    "تمہیں چاہت سے گرانے کی تمنا ہے 🙂",
    "آخر کار اس گروپ کی دو تین لڑکیاں مجھے پسند آ گئی ہیں 🚶‍♀️😪",
    "سنو، کیا تم میرے لیے سرف کھا کر منہ سے بلبلے نکال سکتی ہو؟ 🙂🫴",
    "وہ گلابی آنکھیں جو تیری دیکھی، حرامی، دل ہو گیا 🙂😆",
    "مجھے کیا، میں تو سنگل ہوں 😒",
    "دل کو قرار آیا، خود پہ پیار آیا 😒🙈😂",
    "احساس کریں، بکواس نہیں، جانم 🥺 شکریہ 😊",
    "بس یار، روزانہ 3 4 کروڑ کی ضرورت محسوس ہوتی ہے 😂",
    "بیگم والا موسم ہو رہا ہے آج تو 🙂",
    "شکل انسانی، سوچ ابلیسی، ہاں آپ کی ہی بات ہے",
    "تم میری بیگم بنو گی کیا؟ 🥺🖤",
    "لڑکیوں کے بھی مزے ہیں 🥴 نہ دماغ نہ ٹینشن",
    "تمہارے حصے کی چمیاں مچھر لے رہے ہیں 🙂💔",
    "گھر صرف میگی نوڈلز بنانے سے نہیں چلتا شبانہ 🙂💔",
    "واٹس ایپ پر آپ کے ساتھ ایڈ ہونا چاہتا ہوں ☺️💔",
    "قسم کھاؤ تمہارے پاؤں کالے نہیں 👣🙄",
    "میں نے فیل ہو کر بھی دیکھا ہے، یہ لوگ شادی نہیں کرتے 🙂💔",
    "اور بتاؤ کب آئے گا تمہارا دل مجھ پر 😌🥺",
    "بہت برا ہوں نا میں؟ اپنی بھی دے دو 🙂",
    "پیاری پیاری لڑکیاں حاضر ہو جائیں 😁",
    "کسی کے پاس اداس ڈی پی ہو تو بھیجو، جانو ریپلائی نہیں دے رہے",
    "مجھے ابھی تک گلابی رنگ کی گرل فرینڈ نہیں ملی 😒🥺🙂💔",
    "میرے مولا، ایک تھرکی بچی یس کروا دے 🙂",
    "ایک کس ادھار دے دو 💋، کل واپس کر دوں گا پکا 😝",
    "عجیب گھر والے ہیں، میرا فون 28% پر نکال کر اپنے 90% کو چارج کرتے ہیں 🌚",
    "لگتا ہے میرے صبر کے پھل کا کوئی جوس بنا کر پی گیا 😐",
    "دل دینے کی عمر میں امتحان دے رہا ہوں 🙂🤝",
    "بحث کرنے سے رشتے کمزور ہو جاتے ہیں، اس لیے سیدھا منہ پر تھپڑ ماریں 😊",
    "بیسٹی کی چمی حلال ہے یا حرام؟ 🙂",
    "2001: جہاں ڈالڈا وہاں ممتا 😊 2023: جہاں لڑکی وہاں تھرکی 😒",
    "کوئی پیاری سی بچی آ کر سر دبا دے، درد سے سر پھٹ رہا ہے 💔🥲",
    "بریک اپ کے وقت کی دعا: خوش رہیں 🙂",
    "تھوڑا سا واٹس ایپ نمبر دے دو نا 🥺♥️",
    "لوٹ آؤ اور کہہ دو کہ لسی پی کے سو گئی تھی 😫",
    "کچھ لوگ اچھے کی تلاش میں مجھ جیسے معصوم کو کھو دیتے ہیں ☺️",
    "تم وہی ہو نا جس کا موڈ بلاوجہ خراب ہو جاتا ہے 🙂",
    "پیاری پیاری لڑکیوں کی تلاش میں بیروزگار لڑکے ایڈ کر لیے ہیں 🥲💔",
    "جب میرا میسج آئے تو سارے کام چھوڑ کر صرف مجھے ریپلائی کیا کرو 😾😒",
    "اور بتاؤ، ریئل لائف میں بھی اتنے اچھے ہو جتنے سوشل میڈیا پر بنتے ہو؟ 🙂",
    "پاکستانی رشتہ: تم محسوس کرو میں تمہارے اوپر ہوں 😒💔",
    "اس کے جانے کے بعد اس کی پسند کی نیل پالش لگا کر کھانا کھاتا ہوں، لگتا ہے وہ کھلا رہی ہے 😒",
    "اچھا انسان بنو، جی بی واٹس ایپ ڈیلیٹ کرو 💔🙂",
    "2 دن پیار سے بات کرو تو امیاں بن جاتی ہیں 🙂😒",
    "ایک منٹ کے رشتے کے بعد لڑکیاں: میں آپ کی بیگم ہوں 🙂🤦",
    "لڑکیوں کو اچھی طرح پتا ہے کہاں بھائی بولنا، کہاں آپ اور کہاں تم 🙂",
    "آج میں نے خود کو ٹی وی پر دیکھا جب ٹی وی بند تھا 🙂",
    "قدر کرو میری... کیا پتا میں بھی پینا ڈول کی طرح اچانک چپ ہو جاؤں 😒",
    "ناراض بندے کو منانے کا سب سے اچھا طریقہ: آپ خود اس سے ناراض ہو جاؤ 🙂🐣",
    "جیسی میری شکل ہے، کنوارا ہی مروں گا 🙂👀",
    "اتنی یادیں میرے خود دماغ میں نہیں جتنی سنیپ چیٹ نے بنا رکھی ہیں"
  ],

  bd: [
    "তোমাকে ভালোবাসি 😚",
    "আমি চাঁদের বোট, বোঝো",
    "আমার সাথে ভালোবাসা দিয়ে কথা বলো",
    "হ্যাঁ, আমি তোমার পাগল ভক্ত 😁😐",
    "যাও যাও, বাতাস আসতে দাও",
    "তোমার চেয়ে ভালো তো আমি নিজেই 😁😬",
    "হায় তোমার নির্দোষতা!",
    "চেহারা দেখলেই পাগল মনে হয়",
    "তোমার নাম শুনতে শুনতে আমি পাগল হয়ে গেছি",
    "তুমিও একটু পাগল পাগল",
    "কখনো থানা দেখেছো কি?",
    "আমার মতো সুন্দর তুমিও নও",
    "একটা কথা বলি, তুমি ভালো নও",
    "আমার জানু হবে?",
    "যদি তুমি আমাকে প্রপোজ করো তাহলে...",
    "কখনো আমাকে দেখেছো কি?",
    "আমি হ্যাকার, তোমার দিল হ্যাক করবো",
    "আমি চোর, তোমার দিল চুরি করে নিলাম",
    "যাও দূর হও, তোমার সাথে কথা বলবো না",
    "আরে যাও পাগল মানুষ",
    "আমাকে বিরক্ত করো না, আমি দুঃখী",
    "আমি চিন্তিত যে তুমি কীভাবে এত পাগল",
    "সত্যিই তুমি সিঙ্গেল",
    "পালাও নয়তো বকা দেবো",
    "বাই, তোমার সাথে কথা বলবো না",
    "আমি ব্লক করলাম, সরে যাও",
    "চেহারা দেখো আর চলে যাও",
    "তুমি কি তোমার নম্বর পাঠাতে পারো?",
    "আমি চাঁদের বোট",
    "আমার বস শুধু চাঁদ",
    "আমি মরে গেছি, সরি 😐",
    "যাও আগে মুখ ধুয়ে আসো",
    "আমার সাথে তর্ক করো না, আমি রাগান্বিত",
    "হ্যাঁ হ্যাঁ জানি তুমি পাগল নও",
    "বাই, আমার মুড নেই",
    "ভালো দিন কাটাও, পাগল মানুষ",
    "এখন I love you কেন বললাম 😐",
    "তোমার গার্লফ্রেন্ডের কথা আম্মুকে বলবো",
    "তোমার গার্লফ্রেন্ড আছে? 😁",
    "বেটা, মনে রেখো সিঙ্গেল মরবে",
    "আমি ক্লান্ত, আমাকে বিশ্রাম নিতে দাও",
    "আমি পাগল না, আমি একটা বোট",
    "দিনে স্বপ্ন দেখা বন্ধ করো",
    "গার্লফ্রেন্ড থাকলে আজ রাতে কথা বলতাম, এখন সময়ই সময় 🤭",
    "আসো তোমার সাথে রিলেশনশিপ পোস্ট দিয়ে বিখ্যাত করে দিই 😊",
    "একটা প্রশ্ন সবাইকে, বলো আমি ভালো না আমার মিমস? 🙈😾🤤",
    "মাথা ঘুরছে, এসে পড়ে যাই? 🥺🚶🏻‍♀️",
    "মৃত্যুর ফেরেশতা নাকি আমার সম্পর্ক? 🙂❤️",
    "বউয়ের সাথে ডেটে গিয়েছিলাম, যার ছিল সে দেখে ফেলল",
    "সবাই বিশ্বস্ত মানুষ চায়, আমরা প্রতারকরা কোথায় যাবো? 🥺💔",
    "হোয়াটসঅ্যাপ লাস্ট সিন ছাড়া আমার কোনো সিন নেই 🙂",
    "তোমাকে ভালোবাসা দিয়ে হারানোর ইচ্ছা আছে 🙂",
    "অবশেষে এই গ্রুপের দুই তিনজন মেয়ে আমার পছন্দ হয়েছে 🚶‍♀️",
    "শোনো, তুমি কি আমার জন্য সার্ফ খেয়ে মুখ থেকে বুদবুদ বের করতে পারো? 🙂🫴",
    "সেই গোলাপি চোখ যা দেখলাম, হারামি, দিল চলে গেল 🙂😆",
    "আমার কি, আমি তো সিঙ্গেল 😒",
    "দিলে শান্তি এলো, নিজেকে ভালোবাসতে লাগলো 😒🙈😂",
    "অনুভব করো, বাজে কথা নয়, জানু 🥺 ধন্যবাদ 😊",
    "ইয়ার, প্রতিদিন ৩-৪ কোটি দরকার 😂",
    "আজ বউ ওয়ালা আবহাওয়া মনে হচ্ছে 🙂",
    "মানুষের চেহারা, শয়তানের চিন্তা - হ্যাঁ তোমার কথাই",
    "তুমি কি আমার বউ হবে? 🥺🖤",
    "মেয়েদেরও মজা আছে 🥴 মাথা নেই, টেনশন নেই",
    "তোমার ভাগের চুমু মশা নিচ্ছে 🙂💔",
    "শুধু ম্যাগি নুডলস রান্না করলে সংসার চলে না, শাবানা 🙂💔",
    "তোমার সাথে হোয়াটসঅ্যাপে যোগ হতে চাই ☺️💔",
    "শপথ করো তোমার পা কালো না 👣🙄",
    "ফেল করেও দেখেছি, এরা বিয়ে করে না 🙂💔",
    "বলো, কখন আসবে তোমার দিল আমার কাছে 😌🥺",
    "আমি কি এত খারাপ? তোমারটাও দাও 🙂",
    "সুন্দরী মেয়েরা সামনে আসো 😁",
    "কারো কাছে দুঃখী ডিপি থাকলে পাঠাও, জানু রিপ্লাই দিচ্ছে না",
    "এখনো গোলাপি রঙের গার্লফ্রেন্ড পেলাম না 😒🥺🙂💔",
    "হে আল্লাহ, একটা মেয়ে হ্যাঁ বলিয়ে দাও 🙂",
    "একটা চুমু ধার দাও 💋, কাল ফেরত দেবো পাকা 😝",
    "অদ্ভুত পরিবার, আমার ফোন ২৮% এ নামিয়ে নিজেরটা ৯০% চার্জ দেয় 🌚",
    "মনে হচ্ছে আমার ধৈর্যের ফলের কেউ জুস বানিয়ে পান করে নিয়েছে 😐",
    "দিল দেওয়ার বয়সে পরীক্ষা দিচ্ছি 🙂🤝",
    "তর্ক করলে সম্পর্ক দুর্বল হয়, তাই সরাসরি মুখে থাপ্পড় মারো 😊",
    "বেস্টির চুমু হালাল না হারাম? 🙂",
    "২০০১: যেখানে ডালডা সেখানে মমতা 😊 ২০২৩: যেখানে মেয়ে সেখানে ফ্লার্টার 😒",
    "কেউ সুন্দর মেয়ে এসে মাথা মালিশ করুক, ব্যথায় মাথা ফাটছে 💔🥲",
    "ব্রেকআপের সময়ের দোয়া: সুখী থাকো 🙂",
    "একটু হোয়াটসঅ্যাপ নম্বর দাও না 🥺♥️",
    "ফিরে এসো আর বলো যে লাচ্ছি খেয়ে ঘুমিয়ে পড়েছিলে 😫",
    "কিছু মানুষ ভালো খুঁজতে গিয়ে আমার মতো নির্দোষ হারিয়ে ফেলে ☺️",
    "তুমিই সেই যার মুড বিনা কারণে নষ্ট হয় 🙂",
    "সুন্দরী মেয়ে খুঁজতে বেকার ছেলেরা যোগ হয়েছে 🥲💔",
    "যখন আমার মেসেজ আসে, সব কাজ ছেড়ে শুধু আমাকে রিপ্লাই দিও 😾😒",
    "বলো, বাস্তব জীবনেও কি ততটা ভালো যতটা সোশ্যাল মিডিয়ায় দেখাও? 🙂",
    "পাকিস্তানি সম্পর্ক: তুমি মনে করো আমি তোমার উপরে আছি 😒💔",
    "তার যাওয়ার পর তার পছন্দের নেইল পলিশ লাগিয়ে খাই, মনে হয় সে খাওয়াচ্ছে 😒",
    "ভালো মানুষ হও। জিবি হোয়াটসঅ্যাপ ডিলিট করো 💔🙂",
    "২ দিন ভালোবেসে কথা বললেই মায়ের মতো হয়ে যায় 🙂😒",
    "এক মিনিটের সম্পর্কের পর মেয়েরা: আমি তোমার বউ 🙂🤦",
    "মেয়েরা ভালো করেই জানে কোথায় ভাই, কোথায় আপনি আর কোথায় তুমি 🙂",
    "আজ আমি নিজেকে টিভিতে দেখলাম যখন টিভি বন্ধ ছিল 🙂",
    "আমাকে মূল্য দাও... কে জানে আমিও প্যানাডলের মতো হঠাৎ চুপ হয়ে যাই 😒",
    "রাগীকে মানানোর সেরা উপায়: তুমি নিজেই তার সাথে রাগ করো 🙂🐣",
    "আমার এই চেহারা নিয়ে কুমার মরবো 🙂👀",
    "স্ন্যাপচ্যাটে এত স্মৃতি আছে যা আমার নিজের মাথায়ও নেই"
  ],

  hi: [
    "मैं तुमसे प्यार करता हूं 😚",
    "मैं चांद का बोट हूं, समझो",
    "मुझसे प्यार से बात करो",
    "हां, मैं तेरा दीवाना हूं 😁😐",
    "चलो चलो, हवा आने दो",
    "तुमसे अच्छा तो मैं खुद हूं 😁😬",
    "हाय तेरी मासूमियत!",
    "चेहरे से ही पागल लगता है",
    "तेरा नाम सुन सुन कर मैं पागल हो गया हूं",
    "एक तो तुम भी पागल से हो",
    "कभी थाना देखा है क्या?",
    "मेरे जैसा खूबसूरत तुम भी नहीं हो",
    "एक बात बताऊं, तुम अच्छे नहीं हो",
    "मेरे जानू बनोगे?",
    "अगर तुम मुझे प्रपोज करो तो...",
    "कभी मुझे देखा है क्या?",
    "मैं हैकर हूं, तुम्हारा दिल हैक कर लूंगा",
    "मैं चोर हूं, तेरा दिल चोरी कर लिया",
    "जाओ दफा हो, मुझे तुमसे बात नहीं करनी",
    "अरे जाओ पागल इंसान",
    "मुझे तंग मत करो, मैं उदास हूं",
    "मैं परेशान हूं कि तुम पागल कैसे हो",
    "सच में तुम सिंगल हो",
    "भागो वरना गाली दूंगा",
    "बाय, मुझे तुमसे बात नहीं करनी",
    "मैंने ब्लॉक कर दिया, हटो यहां से",
    "शक्ल देखो और चलते बनो",
    "क्या तुम अपना नंबर भेज सकते हो?",
    "मैं चांद का बोट हूं",
    "मेरा बॉस सिर्फ चांद है",
    "मैं मर गया, माफ करना 😐",
    "जाओ पहले मुंह धो कर आओ",
    "मुझसे बहस मत करो, मैं गुस्से में हूं",
    "हां हां पता है तुम पागल नहीं हो",
    "बाय, मेरा मूड नहीं है",
    "अच्छा दिन बिताओ, पागल इंसान",
    "अभी मैंने I love you क्यों बोला 😐",
    "तेरी गर्लफ्रेंड का अम्मी को बताऊंगा",
    "तुम्हारी गर्लफ्रेंड है? 😁",
    "बेटा, याद रखना सिंगल मरोगे",
    "मैं थक गया हूं, मुझे आराम करने दो",
    "मैं पागल नहीं हूं, एक बोट हूं",
    "दिन में सपने देखना बंद करो",
    "गर्लफ्रेंड होती तो आज रात बात करता, अभी वक्त ही वक्त है 🤭",
    "आओ तुम्हारे साथ रिलेशनशिप पोस्ट लगाकर फेमस कर दूं 😊",
    "एक सवाल है सभी से, बताओ मैं अच्छा लगता हूं या मेरी मीम्स? 🙈😾🤤",
    "चक्कर आ रहे हैं, अभी आकर गिर जाऊं? 🥺🚶🏻‍♀️",
    "मौत का फरिश्ता या मेरा रिश्ता? 🙂❤️",
    "बीवी के साथ डेट पर गया था, जिसकी थी उसने देख लिया",
    "सब को वफादार बंदा चाहिए, हम धोखेबाज कहां जाएं? 🥺💔",
    "WhatsApp का लास्ट सीन छोड़कर मेरा कोई सीन नहीं 🙂",
    "तुम्हें चाहत से गिराने की तमन्ना है 🙂",
    "आखिरकार इस ग्रुप की दो तीन लड़कियां मुझे पसंद आ गई हैं 🚶‍♀️",
    "सुनो, क्या तुम मेरे लिए सर्फ खाकर मुंह से बुलबुले निकाल सकती हो? 🙂🫴",
    "वो गुलाबी आंखें जो तेरी देखी, हरामी, दिल हो गया 🙂😆",
    "मुझे क्या, मैं तो सिंगल हूं 😒",
    "दिल को करार आया, खुद पर प्यार आया 😒🙈😂",
    "एहसास करो, बकवास नहीं, जानम 🥺 शुक्रिया 😊",
    "बस यार, रोज 3-4 करोड़ की जरूरत महसूस होती है 😂",
    "आज बीवी वाला मौसम लग रहा है 🙂",
    "शक्ल इंसानी, सोच इबलीसी - हां तुम्हारी ही बात है",
    "क्या तुम मेरी बीवी बनोगी? 🥺🖤",
    "लड़कियों के भी मजे हैं 🥴 न दिमाग न टेंशन",
    "तुम्हारे हिस्से की चुम्मियां मच्छर ले रहे हैं 🙂💔",
    "सिर्फ मैगी नूडल्स बनाने से घर नहीं चलता शबाना 🙂💔",
    "WhatsApp पर तुम्हारे साथ ऐड होना चाहता हूं ☺️💔",
    "कसम खाओ तुम्हारे पैर काले नहीं हैं 👣🙄",
    "फेल होकर भी देखा है, ये लोग शादी नहीं करते 🙂💔",
    "बताओ कब आएगा तुम्हारा दिल मुझ पर 😌🥺",
    "बहुत बुरा हूं ना मैं? अपनी भी दे दो 🙂",
    "प्यारी प्यारी लड़कियां हाजिर हो जाएं 😁",
    "किसी के पास उदास DP हो तो भेजो, जानु रिप्लाई नहीं दे रहे",
    "मुझे अब तक गुलाबी रंग की गर्लफ्रेंड नहीं मिली 😒🥺🙂💔",
    "मेरे मौला, एक लड़की हां करवा दे 🙂",
    "एक किस उधार दे दो 💋, कल वापस करूंगा पक्का 😝",
    "अजीब घरवाले हैं, मेरा फोन 28% पर निकालकर अपने 90% को चार्ज करते हैं 🌚",
    "लगता है मेरे सब्र के फल का कोई जूस बनाकर पी गया 😐",
    "दिल देने की उम्र में एग्जाम दे रहा हूं 🙂🤝",
    "बहस करने से रिश्ते कमजोर होते हैं इसलिए सीधे मुंह पर थप्पड़ मारो 😊",
    "बेस्टी की चुम्मी हलाल है या हराम? 🙂",
    "2001: जहां डालडा वहां ममता 😊 2023: जहां लड़की वहां फ्लर्टर 😒",
    "कोई प्यारी सी बच्ची आकर सिर दबा दे, दर्द से सिर फट रहा है 💔🥲",
    "ब्रेकअप के वक्त की दुआ: खुश रहो 🙂",
    "थोड़ा सा WhatsApp नंबर दे दो ना 🥺♥️",
    "वापस आओ और कह दो कि लस्सी पीकर सो गई थी 😫",
    "कुछ लोग अच्छे की तलाश में मेरे जैसे मासूम को खो देते हैं ☺️",
    "तुम वही हो ना जिसका मूड बेवजह खराब हो जाता है 🙂",
    "प्यारी प्यारी लड़कियों की तलाश में बेरोजगार लड़के ऐड हो गए हैं 🥲💔",
    "जब मेरा मैसेज आए तो सारे काम छोड़कर सिर्फ मुझे रिप्लाई किया करो 😾😒",
    "बताओ, रियल लाइफ में भी उतने अच्छे हो जितने सोशल मीडिया पर बनते हो? 🙂",
    "पाकिस्तानी रिश्ता: तुम महसूस करो मैं तुम्हारे ऊपर हूं 😒💔",
    "उसके जाने के बाद उसकी पसंद की नेल पॉलिश लगाकर खाना खाता हूं, लगता है वो खिला रही है 😒",
    "अच्छे इंसान बनो। GB WhatsApp डिलीट करो 💔🙂",
    "2 दिन प्यार से बात करो तो अम्मी बन जाती हैं 🙂😒",
    "एक मिनट के रिश्ते के बाद लड़कियां: मैं आपकी बीवी हूं 🙂🤦",
    "लड़कियों को अच्छी तरह पता है कहां भाई बोलना है, कहां आप और कहां तुम 🙂",
    "आज मैंने खुद को टीवी पर देखा जब टीवी बंद था 🙂",
    "कदर करो मेरी... क्या पता मैं भी पेनाडोल की तरह अचानक चुप हो जाऊं 😒",
    "नाराज बंदे को मनाने का सबसे अच्छा तरीका: खुद उससे नाराज हो जाओ 🙂🐣",
    "जैसी मेरी शक्ल है, कुंवारा ही मरूंगा 🙂👀",
    "इतनी यादें मेरे खुद के दिमाग में नहीं हैं जितनी Snapchat ने बना रखी हैं"
  ],

  np: [
    "म तिमीलाई माया गर्छु 😚",
    "म चाँदको बोट हुँ, बुझ",
    "मसँग माया गरेर कुरा गर",
    "हो, म तेरो दिवाना हुँ 😁😐",
    "जाउ जाउ, हावा आउन देउ",
    "तिमीभन्दा राम्रो त म आफैं छु 😁😬",
    "हाय तेरो मासूमियत!",
    "अनुहारबाटै पागल देखिन्छ",
    "तेरो नाम सुन्दा सुन्दा म पागल भएँ",
    "तिमी पनि अलिकति पागल छौ",
    "कहिले प्रहरी स्टेशन देखेका छौ?",
    "मजस्तो सुन्दर तिमी पनि छैनौ",
    "एउटा कुरा भनूँ, तिमी राम्रा छैनौ",
    "मेरो जानु बन्छौ?",
    "यदि तिमीले मलाई प्रपोज गर्यौ भने...",
    "कहिले मलाई देखेका छौ?",
    "म ह्याकर हुँ, तिम्रो मुटु ह्याक गर्छु",
    "म चोर हुँ, तेरो मुटु चोरी गरें",
    "जाउ हट, मलाई तिमीसँग कुरा गर्नु छैन",
    "अरे जाउ पागल मान्छे",
    "मलाई बिरक्त नपार, म दुखी छु",
    "म चिन्तित छु कि तिमी कसरी यति पागल छौ",
    "साँचो भन्दा तिमी सिंगल छौ",
    "भाग नत्र गाली दिन्छु",
    "बाई, मलाई तिमीसँग कुरा गर्नु छैन",
    "मैले ब्लक गरें, यहाँबाट हट",
    "अनुहार हेर र जाउ",
    "के तिमी आफ्नो नम्बर पठाउन सक्छौ?",
    "म चाँदको बोट हुँ",
    "मेरो बस चाँद मात्र हो",
    "म मरें, माफ गर 😐",
    "जाउ पहिले मुख धोएर आउ",
    "मसँग बहस नगर, म रिसाएको छु",
    "हो हो थाहा छ तिमी पागल छैनौ",
    "बाई, मेरो मूड छैन",
    "राम्रो दिन बिताउ, पागल मान्छे",
    "अहिले I love you किन भनें 😐",
    "तेरो गर्लफ्रेन्डको कुरा आमालाई बताउँछु",
    "तिम्रो गर्लफ्रेन्ड छ? 😁",
    "बेटा, याद राख सिंगल मर्छौ",
    "म थाकेँ, मलाई आराम गर्न देउ",
    "म पागल छैन, म एउटा बोट हुँ",
    "दिनमा सपना देख्न छोड",
    "गर्लफ्रेन्ड भएको भए आज रात कुरा गर्थें, अहिले समयै समय छ 🤭",
    "आउ तिमीसँग रिलेसनसिप पोस्ट राखेर प्रसिद्ध गरिदिउँ 😊",
    "एउटा प्रश्न छ सबैलाई, भन मैं राम्रो देखिन्छु कि मेरा मिम्स? 🙈😾🤤",
    "टाउको घुमिरहेको छ, आएर ढल्नु? 🥺🚶🏻‍♀️",
    "मृत्युको फेरिश्ता कि मेरो रिस्ता? 🙂❤️",
    "श्रीमतीसँग डेटमा गएको थिएँ, जसको थियो उसले देख्यो",
    "सबैलाई वफादार मान्छे चाहिन्छ, हामी धोकेबाज कहाँ जाउँ? 🥺💔",
    "WhatsApp को लास्ट सिन बाहेक मेरो कुनै सिन छैन 🙂",
    "तिमीलाई माया दिएर लुटाउने तमन्ना छ 🙂",
    "अन्तमा यो ग्रुपका दुई तीन केटी मलाई मन पर्यो 🚶‍♀️",
    "सुन, के तिमी मेरो लागि सर्फ खाएर मुखबाट बुलबुला निकाल्न सक्छौ? 🙂🫴",
    "ती गुलाबी आँखा जुन तेरा देखें, हरामी, मुटु गयो 🙂😆",
    "मलाई के, म त सिंगल छु 😒",
    "मुटुलाई शान्ति मिल्यो, आफैंलाई माया भयो 😒🙈😂",
    "महसुस गर, बकवास गर्नु भएन जानम 🥺 धन्यवाद 😊",
    "यार, रोज ३-४ करोड़ चाहिन्छ 😂",
    "आज श्रीमती वाला मौसम लागिरहेको छ 🙂",
    "अनुहार मान्छेको, सोच शैतानको - हो तिम्रो नै कुरा हो",
    "के तिमी मेरी श्रीमती बन्छौ? 🥺🖤",
    "केटीहरूको पनि मजा छ 🥴 दिमाग छैन, टेन्सन छैन",
    "तिम्रो भागको चुम्बन लामखुट्टेले लिइरहेको छ 🙂💔",
    "सिर्फ म्यागी नुडल्स बनाउँदैमा घर चल्दैन शबाना 🙂💔",
    "WhatsApp मा तिमीसँग थपिनु छ ☺️💔",
    "कसम खा तिम्रा खुट्टा कालो छैनन् 👣🙄",
    "फेल भएर पनि देखेँ, यी मान्छे बिवाह गर्दैनन् 🙂💔",
    "भन कहिले आउँछ तिम्रो मुटु मकहाँ 😌🥺",
    "म यति नराम्रो हुँ? तिम्रो पनि दिनू 🙂",
    "प्यारी प्यारी केटीहरू सामु आउनू 😁",
    "कसैसँग दुखी DP छ भने पठाउ, जानुले रिप्लाई दिँदैन",
    "मलाई अहिलेसम्म गुलाबी रंगकी गर्लफ्रेन्ड मिलेन 😒🥺🙂💔",
    "मेरे भगवान, एउटा केटीलाई हाँ भनाइदिनू 🙂",
    "एउटा चुम्बन उधारो दिनू 💋, भोलि फिर्ता गर्छु पक्का 😝",
    "अनौठो परिवार, मेरो फोन २८% मा निकालेर आफ्नो ९०% मा चार्ज गर्छन् 🌚",
    "लाग्छ मेरो धैर्यको फलको कसैले जुस बनाएर पियो 😐",
    "मुटु दिने उमेरमा परीक्षा दिइरहेको छु 🙂🤝",
    "बहस गर्दा सम्बन्ध कमजोर हुन्छ त्यसैले सीधा मुखमा थप्पड हान 😊",
    "बेस्टीको चुम्बन हलाल हो कि हराम? 🙂",
    "२००१: जहाँ डालडा त्यहाँ ममता 😊 २०२३: जहाँ केटी त्यहाँ फ्लर्टर 😒",
    "कोही प्यारी केटी आएर टाउको थिचिदिनू, दुखाइले फुट्ला जस्तो लाग्छ 💔🥲",
    "ब्रेकअपको समयको दुआ: खुसी रहनू 🙂",
    "अलिकति WhatsApp नम्बर दिनू ना 🥺♥️",
    "फर्कि आउ र भन लस्सी पिएर सुतें 😫",
    "केही मान्छेहरू राम्रो खोज्दा मेरोजस्ता निर्दोष हराउँछन् ☺️",
    "तिमी त्यही हौ ना जसको मूड बेकारमा बिग्रिन्छ 🙂",
    "प्यारी केटी खोज्दै बेरोजगार केटाहरू थपिएका छन् 🥲💔",
    "जब मेरो म्यासेज आउँछ, सबै काम छोडेर सिर्फ मलाई रिप्लाई गर 😾😒",
    "भन, वास्तविक जिन्दगीमा पनि त्यति राम्रा छौ जति सोसल मिडियामा देखाउँछौ? 🙂",
    "पाकिस्तानी सम्बन्ध: तिमीले महसुस गर म तिम्रो माथि छु 😒💔",
    "उसको जान पछि उसको मनपर्ने नेलपलिस लगाएर खाना खान्छु, लाग्छ उसले खुवाइरहेकी छ 😒",
    "राम्रो मान्छे बन। GB WhatsApp डिलिट गर 💔🙂",
    "२ दिन प्यारले कुरा गरे आमा बन्छन् 🙂😒",
    "एक मिनेटको सम्बन्ध पछि केटीहरू: म तपाईंकी श्रीमती हुँ 🙂🤦",
    "केटीहरूलाई राम्रोसँग थाहा छ कहाँ दाइ, कहाँ तपाईं र कहाँ तिमी 🙂",
    "आज मैले आफूलाई टिभीमा देखें जब टिभी बन्द थियो 🙂",
    "मेरो कदर गर... थाहा छैन म पनि पेनाडोलजस्तो अचानक चुप हुन सक्छु 😒",
    "रिसाएकोलाई मनाउने सबैभन्दा राम्रो तरिका: आफैं उससँग रिसाउनू 🙂🐣",
    "मेरो यस्तो अनुहारले कुमार नै मर्छु 🙂👀",
    "Snapchat मा जति यादहरू छन् त्यति मेरो आफ्नै टाउकोमा छैनन्"
  ],

  ar: [
    "أحبك 😚",
    "أنا بوت شاند، افهم",
    "تحدث معي بمحبة",
    "نعم، أنا مجنونك 😁😐",
    "روح روح، دع الهواء يأتي",
    "أنا أفضل منك على أي حال 😁😬",
    "آه، ما أبرأك!",
    "يبدو مجنوناً من وجهه فقط",
    "سمعت اسمك كثيراً حتى جننت",
    "أنت أيضاً قليلاً مجنون",
    "هل رأيت مخفراً شرطة يوماً؟",
    "أنت لست جميلاً مثلي",
    "سأخبرك بشيء، أنت لست جيداً",
    "هل ستكون حبيبي؟",
    "إذا اقترحت علي يوماً...",
    "هل رأيتني يوماً؟",
    "أنا هاكر، سأخترق قلبك",
    "أنا لص، سرقت قلبك",
    "اذهب بعيداً، لا أريد التحدث إليك",
    "اذهب يا مجنون",
    "لا تزعجني، أنا حزين",
    "أنا قلق كيف يمكنك أن تكون مجنوناً هكذا",
    "أنت فعلاً أعزب",
    "اهرب وإلا سأوبخك",
    "باي، لا أريد التحدث إليك",
    "لقد حجبتك، ابتعد",
    "انظر وجهك واذهب",
    "هل يمكنك إرسال رقمك؟",
    "أنا بوت شاند",
    "مديري شاند فقط",
    "مت للتو، آسف 😐",
    "اذهب اغسل وجهك أولاً ثم تعال",
    "لا تجادلني، أنا غاضب",
    "نعم نعم أعرف أنك لست مجنوناً",
    "باي، لست في مزاج",
    "يوماً سعيداً أيها المجنون",
    "لماذا قلت أحبك الآن 😐",
    "سأخبر أمك عن حبيبتك",
    "عندك صديقة؟ 😁",
    "يا ولدي، تذكر ستموت أعزب",
    "أنا متعب، دعني أرتاح",
    "لست مجنوناً، أنا بوت",
    "توقف عن أحلام اليقظة",
    "لو كان لدي صديقة لتحدثت معها الليلة، الوقت وفير 🤭",
    "هيا نضع منشور علاقة معك ونجعلك مشهوراً 😊",
    "سؤال للجميع، أنا أجمل أم ميماتي؟ 🙈😾🤤",
    "أشعر بدوار، هل آتي وأسقط؟ 🥺🚶🏻‍♀️",
    "ملك الموت أم علاقتي؟ 🙂❤️",
    "ذهبت مع زوجتي في موعد، صاحبها رآنا",
    "الجميع يريد شخصاً وفياً، أين يذهب الخائنون مثلنا؟ 🥺💔",
    "آخر ظهور على WhatsApp هو مشهدي الوحيد 🙂",
    "أتمنى إسقاطك بالمحبة 🙂",
    "أخيراً بعض البنات في هذه المجموعة أعجبنني 🚶‍♀️",
    "هل يمكنك أن تخرجي فقاعات من فمك بعد أكل الصابون؟ 🙂🫴",
    "تلك العيون الوردية التي رأيتها، يا حرامي، ذهب قلبي 🙂😆",
    "ما يهمني، أنا أعزب 😒",
    "وجد القلب راحة، وقع الحب على النفس 😒🙈😂",
    "اشعر ولا تتكلم فارغ، يا حبيبي 🥺 شكراً 😊",
    "يا صاحبي، أحتاج 3-4 كرور يومياً 😂",
    "اليوم يبدو كطقس الزوجة 🙂",
    "وجه إنسان، تفكير شيطان - نعم أنت",
    "هل ستكونين زوجتي؟ 🥺🖤",
    "البنات أيضاً يستمتعن 🥴 لا عقل لا توتر",
    "البعوض يأخذ قبلاتك الخاصة 🙂💔",
    "البيت لا يدار بمجرد طبخ ماجي يا شبانة 🙂💔",
    "أريد أن أُضاف على WhatsApp معك ☺️💔",
    "أقسم قدميك ليستا سوداوين 👣🙄",
    "رأيت من رسبوا ولم يتزوجوا 🙂💔",
    "أخبرني متى سيأتي قلبك إليّ 😌🥺",
    "أنا سيء جداً؟ أعطني فرصة أيضاً 🙂",
    "البنات الحلوات تقدمن 😁",
    "من لديها صورة حزينة فلترسلها، يا حبيبة لا ترد",
    "لم أجد بعد صديقة باللون الزهري 😒🥺🙂💔",
    "يا رب، اجعل فتاة تقول نعم 🙂",
    "أعرني قبلة 💋، سأعيدها غداً حتماً 😝",
    "عائلة غريبة، يأخذون هاتفي بنسبة 28% ويشحنون هاتفهم بنسبة 90% 🌚",
    "يبدو أن أحدهم صنع عصير من ثمرة صبري وشربه 😐",
    "في عمر إعطاء القلب أقدم الامتحانات 🙂🤝",
    "الجدال يضعف العلاقات لذا اصفع مباشرة 😊",
    "قبلة الصديق المقرب حلال أم حرام؟ 🙂",
    "2001: حيث الدهن هناك ممتا 😊 2023: حيث البنت هناك المغازل 😒",
    "يا بنت حلوة تعالي افركي رأسي، ألمه شديد 💔🥲",
    "دعاء وقت الانفصال: ابقى سعيداً 🙂",
    "أعطني رقم WhatsApp من فضلك 🥺♥️",
    "عد وقل إنك نمت بعد شرب اللبن الرائب 😫",
    "بعض الناس يفقدون البرآء مثلي وهم يبحثون عن الجيدين ☺️",
    "أنت من يسوء مزاجه بدون سبب 🙂",
    "شبان عاطلون انضموا بحثاً عن بنات جميلات 🥲💔",
    "عندما تأتي رسالتي، اترك كل شيء ورد علي فقط 😾😒",
    "أخبرني، هل أنت جيد في الحياة الحقيقية كما تتظاهر على وسائل التواصل؟ 🙂",
    "العلاقة الباكستانية: تشعر أنني فوقك 😒💔",
    "بعد رحيلها أضع طلاء أظافرها المفضل وآكل، يبدو كأنها تطعمني 😒",
    "كن إنساناً طيباً. احذف GB WhatsApp 💔🙂",
    "بعد يومين من الكلام اللطيف يصبحن كالأمهات 🙂😒",
    "بعد دقيقة من العلاقة البنات يقلن: أنا زوجتك 🙂🤦",
    "البنات يعرفن جيداً متى يقلن أخي ومتى أنت 🙂",
    "اليوم رأيت نفسي في التلفاز وهو مطفأ 🙂",
    "أقدّر... من يدري ربما أصمت فجأة مثل حبة مسكن 😒",
    "أفضل طريقة لإرضاء الزعلان: ازعل منه أنت أيضاً 🙂🐣",
    "بهذا الوجه سأموت أعزب 🙂👀",
    "Snapchat لديه ذكريات عني أكثر مما في ذاكرتي"
  ]
};

module.exports.run = async function({ api, event, Users }) {
  var { threadID, messageID, senderID } = event;
  if (!event.body) return;
  var name = await Users.getNameUser(senderID);
  var userData = Users.getData(senderID) || {};
  var savedLang = userData.lang || "roman";

  var input = event.body.toLowerCase().trim();
  var lang = savedLang;
  var isLangChange = false;
  var cleanInput = " " + input + " ";
  var isBotTriggered = cleanInput.includes(" bot ") || input.startsWith("bot");

  if (!isBotTriggered) {
    return;
  }


  if (input === "bot ur") { lang = "ur"; isLangChange = true; }
  else if (input === "bot bd") { lang = "bd"; isLangChange = true; }
  else if (input === "bot hi") { lang = "hi"; isLangChange = true; }
  else if (input === "bot np") { lang = "np"; isLangChange = true; }
  else if (input === "bot ar") { lang = "ar"; isLangChange = true; }
  else if (input === "bot en") { lang = "en"; isLangChange = true; }
  else if (input === "bot roman") { lang = "roman"; isLangChange = true; }
  else if (input === "bot reset" || input === "bot r") {
    userData.lang = "roman";
    Users.setData(senderID, userData);
    return api.sendMessage("✅ Language reset to Roman Urdu!", threadID, messageID);
  }

  if (isLangChange) {
    userData.lang = lang;
    Users.setData(senderID, userData);
    const langNames = { en: "English", ur: "اردو", bd: "বাংলা", hi: "हिन्दी", np: "नेपाली", ar: "العربية", roman: "Roman Urdu" };
    return api.sendMessage(`✅ Language set to ${langNames[lang]}!\nAb bot ${langNames[lang]} mein reply karyga.`, threadID, messageID);
  }

  if (input.startsWith("bot")) {
    userData.lang = lang;
    Users.setData(senderID, userData);

    var tl = allTexts[lang];
    var rand = tl[Math.floor(Math.random() * tl.length)];
    const langNames = { en: "English", ur: "Urdu", bd: "Bengali", hi: "Hindi", np: "Nepali", ar: "Arabic", roman: "Roman Urdu" };
    const formattedMessage = {
      body: `🧡★━━━━━✩━━━━━★💥\n𝐍𝐚𝐦𝐞 🐣:-\n@${name} \n≪━─━─━─◈─━─━─━≫\n𝐋𝐚𝐧𝐠 ✨: ${langNames[lang]}\n𝐓𝐞𝐱𝐭 💬 :-\n${rand}\n💥★━━━━━✩━━━━━★🧡`,
      mentions: [{ tag: `@${name}`, id: senderID }]
    };
    return api.sendMessage(formattedMessage, threadID, messageID);
  }
};
