// lodash-pari/commands/catbox.js
const axios = require("axios");
const fs = require("fs-extra");
const path = require("path");
const FormData = require("form-data");

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,
    name: "catbox",
    aliases: ["cb"],
    version: "1.1.0",
    credits: "CHAND",
    description: "Upload files to Catbox",
    category: "utility",
    usage: "reply to video/audio/image with: catbox",
    prefix: true
  },

  async run({ api, event }) {
    const { threadID, messageID, type, messageReply } = event;

    if (type !== "message_reply" || !messageReply.attachments || messageReply.attachments.length === 0) {
      return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅡𝐞𝐩𝐥𝐲 𝐭𝐨 𝐚 𝐯𝐢𝐝𝐞𝐨, 𝐚𝐮𝐝𝐢𝐨, 𝐨𝐫 𝐢𝐦𝐚𝐠𝐞 𝐰𝐢𝐭𝐡: catbox\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);
    }

    try {
      const attachment = messageReply.attachments[0];
      const fileUrl = attachment.url;

      if (!fileUrl) {
        return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅕𝐢𝐥𝐞 𝐔𝐑𝐋 𝐧𝐨𝐭 𝐟𝐨𝐮𝐧𝐝\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);
      }

      await api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n⏳ 🅤𝐩𝐥𝐨𝐚𝐝𝐢𝐧𝐠 𝐭𝐨 𝐂𝐚𝐭𝐛𝐨𝐱...\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", threadID, messageID);

      // download file
      const fileRes = await axios.get(fileUrl, {
        responseType: "arraybuffer"
      });

      const ext = attachment.type || "file";
      const fileName = `upload_${Date.now()}.${ext}`;

      const form = new FormData();
      form.append("reqtype", "fileupload");
      form.append("fileToUpload", Buffer.from(fileRes.data), {
        filename: fileName,
        contentType: attachment.mimeType || "application/octet-stream"
      });

      const uploadRes = await axios.post(
        "https://catbox.moe/user/api.php",
        form,
        {
          headers: {
            ...form.getHeaders()
          },
          maxBodyLength: Infinity,
          maxContentLength: Infinity
        }
      );

      const uploadedUrl = uploadRes.data.trim();

      if (!uploadedUrl || uploadedUrl.includes("error")) {
        return api.sendMessage(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅤𝐩𝐥𝐨𝐚𝐝 𝐟𝐚𝐢𝐥𝐞𝐝\n${uploadRes.data}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`, threadID, messageID);
      }

      return api.sendMessage(
        `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅤𝐩𝐥𝐨𝐚𝐝𝐞𝐝 𝐒𝐮𝐜𝐜𝐞𝐬𝐬𝐟𝐮𝐥𝐥𝐲\n\n📁 ${uploadedUrl}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`,
        threadID,
        messageID
      );

    } catch (err) {
      return api.sendMessage(`❌ Upload Error: ${err.message}`, threadID, messageID);
    }
  }
};
