// lodash-pari/commands/clearcache.js
const fs = require('fs-extra');
const path = require('path');
const { GOD_UIDS } = require('../../../Data/system/handle/handleCommand');

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'clearcache',
    aliases: ['cc', 'cache'],
    description: 'Clear cache folder (images, audio, video files)',
    usage: 'clearcache',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { senderID } = event || {};
    
    const isBotAdmin = config.ADMINBOT?.includes(senderID);
    const isGod = senderID && GOD_UIDS?.includes(senderID);
    
    if (!isBotAdmin && !isGod) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅞𝐧𝐥𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐭𝐡𝐢𝐬 𝐜𝐨𝐦𝐦𝐚𝐧𝐝.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const cacheDir = path.join(__dirname, 'cache');
    
    if (!fs.existsSync(cacheDir)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅒𝐚𝐜𝐡𝐞 𝐟𝐨𝐥𝐝𝐞𝐫 𝐝𝐨𝐞𝐬 𝐧𝐨𝐭 𝐞𝐱𝐢𝐬𝐭 𝐨𝐫 𝐢𝐬 𝐚𝐥𝐫𝐞𝐚𝐝𝐲 𝐞𝐦𝐩𝐭𝐲.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    try {
      const files = fs.readdirSync(cacheDir);
      
      if (files.length === 0) {
        return send.reply('Cache folder is already empty.');
      }
      
      const mediaExtensions = ['.jpg', '.jpeg', '.png', '.gif', '.mp3', '.mp4', '.mpeg', '.webp', '.wav', '.ogg'];
      
      let deleted = 0;
      let totalSize = 0;
      let skipped = 0;
      
      for (const file of files) {
        const filePath = path.join(cacheDir, file);
        const ext = path.extname(file).toLowerCase();
        
        if (mediaExtensions.includes(ext)) {
          try {
            const stats = fs.statSync(filePath);
            totalSize += stats.size;
            fs.unlinkSync(filePath);
            deleted++;
          } catch {
            skipped++;
          }
        } else {
          skipped++;
        }
      }
      
      const sizeMB = (totalSize / (1024 * 1024)).toFixed(2);
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅒𝐚𝐜𝐡𝐞 𝐂𝐋𝐄𝐀𝐑𝐄𝐃\n═══════════════════════\n🅓𝐞𝐥𝐞𝐭𝐞𝐝: ${deleted} files\n🅢𝐤𝐢𝐩𝐩𝐞𝐝: ${skipped} files\n🅕𝐫𝐞𝐞𝐝: ${sizeMB} MB\n═══════════════════════\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
      
    } catch (error) {
      return send.reply('Failed to clear cache: ' + error.message);
    }
  }
};
