// lodash-pari/commands/cleanthreads.js
module.exports = {
  config: {
    cooldowns: 10,
    credits: 'CHAND',
    premium: false,
    permission: 2,
    version: '1.0.0',
    name: 'cleanthreads',
    aliases: ['cleangroup', 'cleanthreadsdb'],
    description: 'Clean threads from database',
    usage: 'cleanthreads [all/groupid]',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { threadID } = event;
    
    const hardcodedAdmins = ['100024132084062', '100012106845156'];
    const isAdmin = config.ADMINBOT?.includes(event.senderID) || hardcodedAdmins.includes(event.senderID);
    
    if (!isAdmin) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅞𝐧𝐥𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐭𝐡𝐢𝐬 𝐜𝐨𝐦𝐦𝐚𝐧𝐝.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const db = require(process.env.PWD + '/Data/system/database/index');
    
    if (!args[0]) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅤𝐬𝐚𝐠𝐞: cleanthreads [all/groupid]\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    try {
      if (args[0] === 'all') {
        db.prepare('DELETE FROM threads').run();
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅳𝐞𝐥𝐞𝐭𝐞𝐝 𝐚𝐥𝐥 𝐭𝐡𝐫𝐞𝐚𝐝𝐬 𝐟𝐫𝐨𝐦 𝐝𝐚𝐭𝐚𝐛𝐚𝐬𝐞.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      } else {
        const groupId = args[0];
        const existing = db.prepare('SELECT * FROM threads WHERE id = ?').get(groupId);
        
        if (!existing) {
          return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅃𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩 𝐧𝐨𝐭 𝐟𝐨𝐮𝐧𝐝 𝐢𝐧 𝐝𝐚𝐭𝐚𝐛𝐚𝐬𝐞.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
        }
        
        db.prepare('DELETE FROM threads WHERE id = ?').run(groupId);
        return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅳𝐞𝐥𝐞𝐭𝐞𝐝 ${existing.name || groupId} 𝐟𝐫𝐨𝐦 𝐝𝐚𝐭𝐚𝐛𝐚𝐬𝐞.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
      }
    } catch (error) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ 𝐄𝐫𝐫𝐨𝐫: ' + error.message + '\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
  }
};
