// lodash-pari/commands/color.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'setcolor',
    aliases: ['setcolor', 'groupcolor', 'gpcolor'],
    description: 'Change group chat color/theme',
    usage: 'color [color name]',
    category: 'Group Management',
    adminOnly: true,
    groupOnly: true,
    prefix: true
  },

  async run({ api, event, args, send }) {
    const { threadID } = event;

    if (!args.length) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🎨 🅐𝐯𝐚𝐢𝐥𝐚𝐛𝐥𝐞 🅣𝐡𝐞𝐦𝐞 🅒𝐨𝐥𝐨𝐫𝐬:\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n🅑𝐚𝐬𝐢𝐜 🅒𝐨𝐥𝐨𝐫𝐬:\nDefaultBlue, Red, TealBlue, Yellow, Orange, Green,\nLavenderPurple, HotPink, AquaBlue, BrightPurple,\nCoralPink\n🅢𝐩𝐞𝐜𝐢𝐚𝐥 🅣𝐡𝐞𝐦𝐞𝐬:\nAqua, Mango, Berry, Citrus, Candy, Earth, Support,\nMusic, Pride, DoctorStrange, LoFi, Sky,\nLunarNewYear, Celebration, Chill, StrangerThings,\nDune, Care, Astrology, Birthday, Cottagecore, Ocean,\nLove, TieDye, Monochrome, Rocket, Unicorn, Tropical,\nMaple, Sushi, Shadow, Rose, Lavender, Tulip,\nClassic, Peach, Honey, Kiwi, Grape, NonBinary,\nTransgender, TaylorSwift, Autumn, Cyberpunk2077,\n& many more!\n📌 🅤𝐬𝐚𝐠𝐞: .color [color_name]\n🅔𝐱𝐚𝐦𝐩𝐥𝐞: .color blue\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }

    const colorName = args.join(' ').trim();
    const colorMap = getColorMap();
    
    // Find color (case-insensitive)
    let colorID = null;
    for (const [key, value] of Object.entries(colorMap)) {
      if (key.toLowerCase() === colorName.toLowerCase()) {
        colorID = value;
        break;
      }
    }

    if (!colorID) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅒𝐨𝐥𝐨𝐫 "${colorName}" 𝐧𝐨𝐭 𝐟𝐨𝐮𝐧𝐝.\n\n🅣𝐫𝐲 .color 𝐟𝐨𝐫 𝐚 𝐥𝐢𝐬𝐭 𝐨𝐟 𝐚𝐯𝐚𝐢𝐥𝐚𝐛𝐥𝐞 𝐜𝐨𝐥𝐨𝐫𝐬.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }

    try {
      await api.changeThreadColor(colorID, threadID);
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅖𝐫𝐨𝐮𝐩 𝐭𝐡𝐞𝐦𝐞 𝐜𝐡𝐚𝐧𝐠𝐞𝐝 𝐭𝐨 ${colorName}! 🎨\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    } catch (error) {
      return send.reply(`❌ Failed to change color: ${error.message}`);
    }
  }
};

function getColorMap() {
  return {
    MessengerBlue: "196241301102133",
    Viking: "1928399724138152",
    GoldenPoppy: "174636906462322",
    RadicalRed: "2129984390566328",
    Shocking: "2058653964378557",
    FreeSpeechGreen: "2136751179887052",
    Pumpkin: "175615189761153",
    LightCoral: "980963458735625",
    MediumSlateBlue: "234137870477637",
    DeepSkyBlue: "2442142322678320",
    BrilliantRose: "169463077092846",
    DefaultBlue: "196241301102133",
    HotPink: "169463077092846",
    AquaBlue: "2442142322678320",
    BrightPurple: "234137870477637",
    CoralPink: "980963458735625",
    Orange: "175615189761153",
    Green: "2136751179887052",
    LavenderPurple: "2058653964378557",
    Red: "2129984390566328",
    Yellow: "174636906462322",
    TealBlue: "1928399724138152",
    Aqua: "417639218648241",
    Mango: "930060997172551",
    Berry: "164535220883264",
    Citrus: "370940413392601",
    Candy: "205488546921017",
    Earth: "1833559466821043",
    Support: "365557122117011",
    Music: "339021464972092",
    Pride: "1652456634878319",
    DoctorStrange: "538280997628317",
    LoFi: "1060619084701625",
    Sky: "3190514984517598",
    LunarNewYear: "357833546030778",
    Celebration: "627144732056021",
    Chill: "390127158985345",
    StrangerThings: "1059859811490132",
    Dune: "1455149831518874",
    Care: "275041734441112",
    Astrology: "3082966625307060",
    JBalvin: "184305226956268",
    Birthday: "621630955405500",
    Cottagecore: "539927563794799",
    Ocean: "736591620215564",
    Love: "741311439775765",
    TieDye: "230032715012014",
    Monochrome: "788274591712841",
    Default: "3259963564026002",
    Rocket: "582065306070020",
    Berry2: "724096885023603",
    Candy2: "624266884847972",
    Unicorn: "273728810607574",
    Tropical: "262191918210707",
    Maple: "2533652183614000",
    Sushi: "909695489504566",
    Citrus2: "557344741607350",
    Lollipop: "280333826736184",
    Shadow: "271607034185782",
    Rose: "1257453361255152",
    Lavender: "571193503540759",
    Tulip: "2873642949430623",
    Classic: "3273938616164733",
    Peach: "3022526817824329",
    Honey: "672058580051520",
    Kiwi: "3151463484918004",
    Grape: "193497045377796",
    NonBinary: "737761000603635",
    ThankfulForFriends: "1318983195536293",
    Transgender: "504518465021637",
    TaylorSwift: "769129927636836",
    NationalComingOutDay: "788102625833584",
    Autumn: "822549609168155",
    Cyberpunk2077: "780962576430091",
    MothersDay: "1288506208402340",
    APAHM: "121771470870245",
    Parenthood: "810978360551741",
    StarWars: "1438011086532622",
    GuardianOfTheGalaxy: "101275642962533",
    Bloom: "158263147151440",
    BubbleTea: "195296273246380",
    Basketball: "6026716157422736",
    ElephantsAndFlowers: "693996545771691"
  };
}
