// lodash-pari/commands/convert.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: "convert",
    aliases: [],
    description: "Convert video (mp4) to audio (mp3)",
    usage: "[reply to a video or provide URL]",
    category: "Utility",
    prefix: true
  },

  async run({ api, args, event, send }) {
    const axios = require("axios");
    const fs = require("fs-extra");
    const path = __dirname + "/cache/video_to_audio.mp3";

    try {
      let url;

      if (args[0]) {
        url = args[0];
      } else if (event.messageReply?.attachments[0]?.type === "video") {
        url = event.messageReply.attachments[0].url;
      } else {
        return send.reply("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅟𝐥𝐞𝐚𝐬𝐞 𝐫𝐞𝐩𝐥𝐲 𝐭𝐨 𝐚 𝐯𝐢𝐝𝐞𝐨.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━");
      }

      const response = await axios.get(url, { responseType: "arraybuffer" });
      fs.writeFileSync(path, Buffer.from(response.data, "utf-8"));

      const msg = {
        body: "━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅢𝐮𝐜𝐜𝐞𝐬𝐬𝐟𝐮𝐥𝐥𝐲 𝐜𝐨𝐧𝐯𝐞𝐫𝐭𝐞𝐝 𝐯𝐢𝐝𝐞𝐨 𝐭𝐨 𝐚𝐮𝐝𝐢𝐨 (MP3)!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━",
        attachment: fs.createReadStream(path)
      };

      api.sendMessage(msg, event.threadID, () => fs.unlinkSync(path), event.messageID);

    } catch (error) {
      return send.reply("❌ An error occurred during conversion. Please try again.");
    }
  }
};
