// lodash-pari/commands/copy.js
const fs = require("fs");
const path = require("path");
const axios = require("axios");

const dataFolder = path.join(__dirname, "cache", "data");

const getUserJsonPath = (userID) => {
  return path.join(dataFolder, `copiedMessage_${userID}.json`);
};

const scheduleCacheCleanup = (userID, filePaths) => {
  setTimeout(() => {
    try {
      const userJsonPath = getUserJsonPath(userID);
      
      if (fs.existsSync(userJsonPath)) {
        const data = JSON.parse(fs.readFileSync(userJsonPath, 'utf-8'));
        
        data.attachments.forEach(attPath => {
          if (fs.existsSync(attPath)) {
            fs.unlinkSync(attPath);
          }
        });
        
        fs.unlinkSync(userJsonPath);
      }
      
    } catch (error) {
    }
  }, 3600000);
  
};

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,
    name: "copy",
    aliases: [],
    version: "1.0.1",
    credits: "RAZA",
    description: "📋 Copy a message (text + media) and save it",
    category: "Utility",
    usage: "Reply to a message to copy it",
    prefix: true
  },

  async run({ api, event, send }) {
    const { messageReply, threadID, senderID } = event;
    const userJsonPath = getUserJsonPath(senderID);

    if (!messageReply) {
      return send.reply("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n⚠️ | 🅟𝐥𝐞𝐚𝐬𝐞 𝐫𝐞𝐩𝐥𝐲 𝐭𝐨 𝐚 𝐦𝐞𝐬𝐬𝐚𝐠𝐞 𝐭𝐨 𝐜𝐨𝐩𝐲 𝐢𝐭.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━");
    }

    const { body, attachments } = messageReply;

    if (!fs.existsSync(dataFolder)) fs.mkdirSync(dataFolder, { recursive: true });

    let savedData = {
      body: body || '',
      attachments: []
    };

    const downloadPromises = attachments.map(async (att, index) => {
      const ext = att.type === "photo" ? ".jpg" :
                  att.type === "video" ? ".mp4" :
                  att.type === "audio" ? ".mp3" : "";
      const filename = `copied_${senderID}_${Date.now()}_${index}${ext}`;
      const filePath = path.join(dataFolder, filename);

      const response = await axios.get(att.url, { responseType: "stream" });
      const writer = fs.createWriteStream(filePath);
      response.data.pipe(writer);

      await new Promise((resolve, reject) => {
        writer.on("finish", resolve);
        writer.on("error", reject);
      });

      savedData.attachments.push(filePath);
    });

    await Promise.all(downloadPromises);

    savedData.timestamp = Date.now();
    savedData.userID = senderID;
    
    fs.writeFileSync(userJsonPath, JSON.stringify(savedData, null, 2));
    
    scheduleCacheCleanup(senderID, savedData.attachments);

    return send.reply("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ | 🅜𝐞𝐬𝐬𝐚𝐠𝐞 𝐜𝐨𝐩𝐢𝐞𝐝 𝐚𝐧𝐝 𝐬𝐚𝐯𝐞𝐝 𝐬𝐮𝐜𝐜𝐞𝐬𝐬𝐟𝐮𝐥𝐥𝐲! (Auto-delete 𝐢𝐧 1 𝐡𝐨𝐮𝐫)\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━");
  }
};
