// lodash-pari/commands/creategroup.js
const { GOD_UIDS } = require('../../../Data/system/handle/handleCommand');

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'creategroup',
    aliases: ['newgroup', 'makegroup'],
    description: 'Create a new group with specified members',
    usage: 'creategroup [name] | @mention1 @mention2...',
    category: 'Utility',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { senderID, mentions } = event || {};
    
    const mentionIDs = Object.keys(mentions);
    
    if (mentionIDs.length < 1) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅟𝐥𝐞𝐚𝐬𝐞 𝐦𝐞𝐧𝐭𝐢𝐨𝐧 𝐚𝐭 𝐥𝐞𝐚𝐬𝐭 1 𝐩𝐞𝐫𝐬𝐨𝐧 𝐭𝐨 𝐜𝐫𝐞𝐚𝐭𝐞 𝐚 𝐠𝐫𝐨𝐮𝐩 𝐰𝐢𝐭𝐡.\n\n🅤𝐬𝐚𝐠𝐞: .creategroup Group Name | @user1 @user2\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const input = args.join(' ');
    const parts = input.split('|');
    let groupName = 'New Group';
    
    if (parts.length > 1) {
      groupName = parts[0].trim() || 'New Group';
    }
    
    mentionIDs.push(senderID);
    
    try {
      if (typeof api.createNewGroup !== 'function') {
        return send.reply('This feature is not supported by the current API.');
      }
      
      const threadID = await api.createNewGroup(mentionIDs, groupName);
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅖𝐫𝐨𝐮𝐩 "${groupName}" 𝐜𝐫𝐞𝐚𝐭𝐞𝐝 𝐬𝐮𝐜𝐜𝐞𝐬𝐬𝐟𝐮𝐥𝐥𝐲!\n\n🅣𝐡𝐫𝐞𝐚𝐝 𝐈𝐃: ${threadID}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    } catch (error) {
      return send.reply(`Failed to create group: ${error.message || 'Unknown error'}`);
    }
  }
};
