// lodash-pari/commands/daily.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'daily',
    aliases: ['claim', 'reward'],
    description: 'Claim daily reward with streak bonus',
    usage: 'daily',
    category: 'Economy',
    prefix: true
  },
  
  async run({ api, event, send, Currencies, Users }) {
    const { senderID } = event;
    
    const result = Currencies.claimDaily(senderID);
    const name = await Users.getNameUser(senderID);
    
    if (!result.success) {
      if (result.reason === 'already_claimed') {
        return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n${name}, 𝐲𝐨𝐮 𝐡𝐚𝐯𝐞 𝐚𝐥𝐫𝐞𝐚𝐝𝐲 𝐜𝐥𝐚𝐢𝐦𝐞𝐝 𝐲𝐨𝐮𝐫 𝐝𝐚𝐢𝐥𝐲 𝐫𝐞𝐰𝐚𝐫𝐝 𝐭𝐨𝐝𝐚𝐲!\n\n🅒𝐨𝐦𝐞 𝐛𝐚𝐜𝐤 𝐭𝐨𝐦𝐨𝐫𝐫𝐨𝐰.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
      }
    }
    
    const balance = Currencies.getBalance(senderID);
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅓𝐚𝐢𝐥𝐲 🅡𝐞𝐰𝐚𝐫𝐝 🅒𝐋𝐀𝐈𝐌𝐄𝐃!\n─────────────────\n🅤𝐬𝐞𝐫: ${name}\n🅡𝐞𝐰𝐚𝐫𝐝: $${result.reward.toLocaleString()}\n🅢𝐭𝐫𝐞𝐚𝐤: ${result.streak} days\n─────────────────\n🅝𝐞𝐰 🅑𝐚𝐥𝐚𝐧𝐜𝐞: $${balance.toLocaleString()}\n🅚𝐞𝐞𝐩 𝐜𝐥𝐚𝐢𝐦𝐢𝐧𝐠 𝐝𝐚𝐢𝐥𝐲 𝐭𝐨 𝐢𝐧𝐜𝐫𝐞𝐚𝐬𝐞 𝐲𝐨𝐮𝐫 𝐬𝐭𝐫𝐞𝐚𝐤 𝐛𝐨𝐧𝐮𝐬!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
