// lodash-pari/commands/deposit.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'deposit',
    aliases: ['dep', 'save'],
    description: 'Deposit money to bank',
    usage: 'deposit [amount/all]',
    category: 'Economy',
    prefix: true
  },
  
  async run({ api, event, args, send, Currencies, Users }) {
    const { senderID } = event;
    
    const name = await Users.getNameUser(senderID);
    const wallet = Currencies.getBalance(senderID);
    
    if (wallet <= 0) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n${name}, 𝐲𝐨𝐮 𝐡𝐚𝐯𝐞 𝐧𝐨 𝐦𝐨𝐧𝐞𝐲 𝐢𝐧 𝐲𝐨𝐮𝐫 𝐰𝐚𝐥𝐥𝐞𝐭 𝐭𝐨 𝐝𝐞𝐩𝐨𝐬𝐢𝐭!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    let amount;
    
    if (args[0]?.toLowerCase() === 'all') {
      amount = wallet;
    } else {
      amount = parseInt(args[0]);
    }
    
    if (!amount || amount <= 0 || isNaN(amount)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐯𝐚𝐥𝐢𝐝 𝐚𝐦𝐨𝐮𝐧𝐭 𝐭𝐨 𝐝𝐞𝐩𝐨𝐬𝐢𝐭.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (amount > wallet) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n${name}, 𝐲𝐨𝐮 𝐝𝐨𝐧'𝐭 𝐡𝐚𝐯𝐞 𝐞𝐧𝐨𝐮𝐠𝐡 𝐦𝐨𝐧𝐞𝐲!\n\n🅦𝐚𝐥𝐥𝐞𝐭: $${wallet.toLocaleString()}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    const success = Currencies.deposit(senderID, amount);
    
    if (!success) {
      return send.reply('Failed to deposit money.');
    }
    
    const newWallet = Currencies.getBalance(senderID);
    const newBank = Currencies.getBank(senderID);
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅓𝐞𝐩𝐨𝐬𝐢𝐭 🅢𝐔𝐂𝐂𝐄𝐒𝐒𝐅𝐔𝐋!\n─────────────────\n🅤𝐬𝐞𝐫: ${name}\n🅓𝐞𝐩𝐨𝐬𝐢𝐭𝐞𝐝: $${amount.toLocaleString()}\n─────────────────\n🅦𝐚𝐥𝐥𝐞𝐭: $${newWallet.toLocaleString()}\n🅑𝐚𝐧𝐤: $${newBank.toLocaleString()}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
