// lodash-pari/commands/filter.js
module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 2,
    name: "filter",
    aliases: [""],
    version: "2.0.0",
    credits: "CHAND",
    description: "Remove fbuser",
    category: "moderation",
    prefix: true,
    usage: ""
  },

  async run({ api, event, send }) {
    const { userInfo, adminIDs } = await api.getThreadInfo(event.threadID);
    let successCount = 0;
    let failCount = 0;
    let fbUsers = [];
    
    for (const user of userInfo) {
      if (user.gender === undefined) {
        fbUsers.push(user.id);
      }
    }
    
    const isBotAdmin = adminIDs.map(admin => admin.id).some(id => id == api.getCurrentUserID());
    
    if (fbUsers.length === 0) {
      return send.reply("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅘𝐧 𝐲𝐨𝐮𝐫 𝐠𝐫𝐨𝐮𝐩 𝐝𝐨𝐞𝐬 𝐧𝐨𝐭 𝐞𝐱𝐢𝐬𝐭 'Facebook User'.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━");
    }
    
    api.sendMessage(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅔𝐱𝐢𝐬𝐭𝐢𝐧𝐠 𝐠𝐫𝐨𝐮𝐩 𝐨𝐟 𝐟𝐫𝐢𝐞𝐧𝐝𝐬 ${fbUsers.length} 'Facebook users'.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`, event.threadID, function() {
      if (isBotAdmin) {
        return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🚀 🅡𝐨𝐜𝐊𝐞𝐭 🅛𝐚𝐮𝐧𝐜𝐡\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", event.threadID, async function() {
          for (const userId of fbUsers) {
            try {
              await new Promise(resolve => setTimeout(resolve, 1000));
              await api.removeUserFromGroup(parseInt(userId), event.threadID);
              successCount++;
            } catch (error) {
              failCount++;
            }
          }
          
          api.sendMessage(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅝𝐲𝐜 🅜𝐨𝐕𝐞\n${successCount} ✈️\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`, event.threadID, function() {
            if (failCount !== 0) {
              return api.sendMessage(`😜${failCount} 🤳`, event.threadID);
            }
          });
        });
      } else {
        return api.sendMessage("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅜𝐚𝐊𝐞 🅜𝐞 🅐𝐃𝐦𝐢𝐧 🅑0𝐒𝐬 🤧\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━", event.threadID);
      }
    });
  }
};
