// lodash-pari/commands/friendlist.js
const { GOD_UIDS } = require('../../../Data/system/handle/handleCommand');

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'friendlist',
    aliases: ['friends', 'myfriends'],
    description: 'Show friends list',
    usage: 'friendlist [page]',
    category: 'Friend',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { senderID } = event || {};
    
    const isBotAdmin = config.ADMINBOT?.includes(senderID);
    const isGod = senderID && GOD_UIDS?.includes(senderID);
    
    if (!isBotAdmin && !isGod) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅞𝐧𝐥𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐭𝐡𝐢𝐬 𝐜𝐨𝐦𝐦𝐚𝐧𝐝.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    await send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅕𝐞𝐭𝐜𝐡𝐢𝐧𝐠 𝐟𝐫𝐢𝐞𝐧𝐝𝐬 𝐥𝐢𝐬𝐭...\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    
    try {
      const friends = await api.getFriendsList();
      
      if (!friends || friends.length === 0) {
        return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅝𝐨 𝐟𝐫𝐢𝐞𝐧𝐝𝐬 𝐟𝐨𝐮𝐧𝐝.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
      }
      
      const page = parseInt(args[0]) || 1;
      const perPage = 20;
      const totalPages = Math.ceil(friends.length / perPage);
      const startIdx = (page - 1) * perPage;
      const endIdx = Math.min(startIdx + perPage, friends.length);
      const pageFriends = friends.slice(startIdx, endIdx);
      
      let msg = `━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅕𝐫𝐢𝐞𝐧𝐝𝐬 🅛𝐈𝐒𝐓 (${friends.length})\n═══════════════════════\n🅟𝐚𝐠𝐞 ${page}/${totalPages}\n─────────────────\n\n\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`;
      
      for (let i = 0; i < pageFriends.length; i++) {
        const friend = pageFriends[i];
        const name = friend.fullName || friend.name || 'Unknown';
        const gender = friend.gender === 1 ? '♀' : friend.gender === 2 ? '♂' : '';
        
        msg += `${startIdx + i + 1}. ${name} ${gender}
   UID: ${friend.userID}\n`;
      }
      
      if (totalPages > 1) {
        msg += `\n─────────────────
Use: friendlist [page] for more`;
      }
      
      return send.reply(msg);
    } catch (error) {
      return send.reply('Failed to get friends list: ' + error.message);
    }
  }
};
