// lodash-pari/commands/help2.js
module.exports.config = {
  name: "help2",
  aliases: ['menu2', 'helplist', "help all", "helpall"],
  version: "1.0.3",
  permission: 0,
  credits: "CHAND",
  description: "Show command list or command details",
  prefix: true,
  premium: false,
  category: "Guide",
  usage: "[page number] or [command name]",
  cooldowns: 5,
};

module.exports.run = async function ({ api, event, args, getText, botname, prefix }) {
  const { threadID, messageID } = event;
  const seenNames = new Set();
  const allCommands = Array.from(global.client.commands.values()).filter(cmd => {
    if (seenNames.has(cmd.config.name)) return false;
    seenNames.add(cmd.config.name);
    if (cmd.config.aliases) {
      cmd.config.aliases.forEach(alias => seenNames.add(alias));
    }
    return true;
  });
  const commandsPerPage = 1000;
  const autoUnsend = true;
  const delayUnsend = 6000;

  if (!args[0] || !isNaN(args[0])) {
    const page = parseInt(args[0]) || 1;
    const totalPages = Math.ceil(allCommands.length / commandsPerPage);

    if (page < 1 || page > totalPages)
      return api.sendMessage(`Invalid page number. Choose between 1 and ${totalPages}.`, threadID, messageID);

    const start = (page - 1) * commandsPerPage;
    const end = start + commandsPerPage;
    const commandsToShow = allCommands.slice(start, end);

    let msg = `╔═════•| 💜 |•═════╗\n         𝐀𝐓𝐅 𝐏𝐑𝐎𝐉𝐄𝐂𝐓\n╚═════•| 💜 |•═════╝\n\n━❮●❯━━━❪💝❫━━━❮●❯━\n`;

    for (let i = 0; i < commandsToShow.length; i++) {
      msg += `\n│ ${start + i + 1} ➥ ${commandsToShow[i].config.name}`;
    }

    msg += `\n\nPAGE ${page}/${totalPages}\n\n━❮●❯━━━❪💝❫━━━❮●❯━\nHow To Make Free This Bot\nhttps://youtube.com/@chandtricker\n\n━❮●❯━━━❪💝❫━━━❮●❯━`;

    const sent = await api.sendMessage(msg, threadID, async (err, info) => {
      if (autoUnsend) {
        await new Promise(r => setTimeout(r, delayUnsend * 10000));
        return api.unsendMessage(info.messageID);
      }
    }, messageID);
  } else {
    const name = args[0].toLowerCase();
    const command = global.client.commands.get(name) || 
                    Array.from(global.client.commands.values()).find(cmd => cmd.config.aliases?.includes(name));

    if (!command)
      return api.sendMessage(`Command "${name}" not found.`, threadID, messageID);

    const {
      name: cmdName,
      version,
      permission,
      description,
      usages,
      category,
      credits,
      cooldowns
    } = command.config;

    const permText = permission === 0 ? "User" : permission === 1 ? "Admin" : permission === 2 ? "Group Admin" : "Bot Admin";

    const detail = `COMMAND DETAILS\n\nName: ${cmdName}\nVersion: ${version}\nPermission: ${permText}\nDescription: ${description}\nUsage: ${prefix}${cmdName} ${usages || ''}\nCategory: ${category}\nCooldown: ${cooldowns}s\nCredits: ${credits}`;

    const sent = await api.sendMessage(detail, threadID, async (err, info) => {
      if (autoUnsend) {
        await new Promise(r => setTimeout(r, delayUnsend * 30000));
        return api.unsendMessage(info.messageID);
      }
    }, messageID);
  }
};
