// lodash-pari/commands/hug.js
const request = require("request");
const fs = require("fs");
const axios = require("axios");

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,

    name: "hug",
    aliases: [],
    version: "1.0.0",
    credits: "KASHIF RAZA",
    description: "Hug the tagged friend",
    category: "Love",
    usage: "[tag]",
    prefix: true
  },

  async run({ api, event, send }) {
    var link = [    
      "https://i.postimg.cc/CLVYFKnW/anime-hug-38.gif",
      "https://i.postimg.cc/rFCTzLSz/anime-hug-cry-gif-4.gif",
      "https://i.postimg.cc/ZnzkKfnr/g-DEE1-QGHMm-MAOJRb4-Q-ehq-F7ckhc-VAUyzog-C6-VP5v-LTa-MUavlk-FTEj-Yp-SFl-BPX1c-SJXC7qzk-D4s-Huogbit.gif",
      "https://i.postimg.cc/sDyFk0tz/r9aU2xv.gif",
    ];

    var mention = Object.keys(event.mentions);
    
    if (!mention || mention.length === 0) 
      return send.reply("⚝──⭒─⭑─⭒──⚝\n\nPlease tag one person to hug 💓\n\n⚝──⭒─⭑─⭒──⚝");

    let tag = event.mentions[mention[0]].replace("@", "");

    var callback = () => api.sendMessage(
      {
        body: `≿━━━━༺❀༻━━━━≾\n\n${tag} here's a special hug 🤗💓\n\n≿━━━━༺❀༻━━━━≾`,
        mentions: [{ tag: tag, id: Object.keys(event.mentions)[0] }],
        attachment: fs.createReadStream(__dirname + "/cache/hug.gif")
      },
      event.threadID,
      () => fs.unlinkSync(__dirname + "/cache/hug.gif")
    );

    return request(encodeURI(link[Math.floor(Math.random() * link.length)]))
      .pipe(fs.createWriteStream(__dirname + "/cache/hug.gif"))
      .on("close", () => callback());
  }
};
