// lodash-pari/commands/kick.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'kick',
    aliases: ['remove'],
    description: 'Kick a member from the group',
    usage: 'kick @user/uid',
    category: 'Group',
    groupOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, Users, config }) {
    const { threadID, senderID, mentions } = event;
    
    const threadInfo = await api.getThreadInfo(threadID);
    const adminIDs = threadInfo.adminIDs.map(a => a.id);
    const botID = api.getCurrentUserID();
    
    if (!adminIDs.includes(botID)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅑𝐨𝐭 𝐦𝐮𝐬𝐭 𝐛𝐞 𝐚 𝐠𝐫𝐨𝐮𝐩 𝐚𝐝𝐦𝐢𝐧 𝐭𝐨 𝐤𝐢𝐜𝐤 𝐦𝐞𝐦𝐛𝐞𝐫𝐬.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const isGroupAdmin = adminIDs.includes(senderID);
    
    let uid = '';
    
    if (Object.keys(mentions).length > 0) {
      uid = Object.keys(mentions)[0];
    } else if (args[0] && /^\d+$/.test(args[0])) {
      uid = args[0];
    } else if (event.messageReply) {
      uid = event.messageReply.senderID;
    } else {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅟𝐥𝐞𝐚𝐬𝐞 𝐦𝐞𝐧𝐭𝐢𝐨𝐧 𝐚 𝐮𝐬𝐞𝐫, 𝐫𝐞𝐩𝐥𝐲 𝐭𝐨 𝐭𝐡𝐞𝐢𝐫 𝐦𝐞𝐬𝐬𝐚𝐠𝐞, 𝐨𝐫 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐭𝐡𝐞𝐢𝐫 𝐔𝐈𝐃.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (uid === botID) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅒𝐚𝐧𝐧𝐨𝐭 𝐤𝐢𝐜𝐤 𝐭𝐡𝐞 𝐛𝐨𝐭 𝐢𝐭𝐬𝐞𝐥𝐟.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (adminIDs.includes(uid)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅒𝐚𝐧𝐧𝐨𝐭 𝐤𝐢𝐜𝐤 𝐚 𝐠𝐫𝐨𝐮𝐩 𝐚𝐝𝐦𝐢𝐧.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    try {
      await api.removeUserFromGroup(uid, threadID);
      
      let name = 'User';
      try {
        name = await Users.getValidName(uid, 'User');
      } catch {
        try {
          const info = await api.getUserInfo(uid);
          const rawName = info[uid]?.name;
          if (rawName && rawName.toLowerCase() !== 'facebook user' && rawName.toLowerCase() !== 'facebook') {
            name = rawName;
          }
        } catch {}
      }
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅚𝐢𝐜𝐤𝐞𝐝 ${name} 𝐟𝐫𝐨𝐦 𝐭𝐡𝐞 𝐠𝐫𝐨𝐮𝐩.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    } catch (error) {
      return send.reply('Failed to kick user: ' + error.message);
    }
  }
};
