// lodash-pari/commands/kickall.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 2,
version: '1.0.0',
    name: 'kickall',
    aliases: ['removeall', 'cleargroup'],
    description: 'Kick all members except admins',
    usage: 'kickall [confirm]',
    category: 'Group',
    groupOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { threadID, senderID } = event;
    
    const threadInfo = await api.getThreadInfo(threadID);
    const adminIDs = threadInfo.adminIDs.map(a => a.id);
    const botID = api.getCurrentUserID();
    
    if (!adminIDs.includes(botID)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅑𝐨𝐭 𝐦𝐮𝐬𝐭 𝐛𝐞 𝐚 𝐠𝐫𝐨𝐮𝐩 𝐚𝐝𝐦𝐢𝐧 𝐭𝐨 𝐤𝐢𝐜𝐤 𝐦𝐞𝐦𝐛𝐞𝐫𝐬.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const participants = threadInfo.participantIDs || [];
    const membersToKick = participants.filter(id => 
      id !== botID && !adminIDs.includes(id)
    );
    
    if (membersToKick.length === 0) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅝𝐨 𝐦𝐞𝐦𝐛𝐞𝐫𝐬 𝐭𝐨 𝐤𝐢𝐜𝐤 (𝐨𝐧𝐥𝐲 𝐚𝐝𝐦𝐢𝐧𝐬 𝐫𝐞𝐦𝐚𝐢𝐧𝐢𝐧𝐠).\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (args[0]?.toLowerCase() !== 'confirm') {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅐𝐫𝐞 𝐲𝐨𝐮 𝐬𝐮𝐫𝐞 𝐲𝐨𝐮 𝐰𝐚𝐧𝐭 𝐭𝐨 𝐤𝐢𝐜𝐤 ${membersToKick.length} 𝐦𝐞𝐦𝐛𝐞𝐫𝐬?\n🅣𝐡𝐢𝐬 𝐰𝐢𝐥𝐥 𝐫𝐞𝐦𝐨𝐯𝐞 🅐𝐋𝐋 𝐧𝐨𝐧-𝐚𝐝𝐦𝐢𝐧 𝐦𝐞𝐦𝐛𝐞𝐫𝐬 𝐟𝐫𝐨𝐦 𝐭𝐡𝐞 𝐠𝐫𝐨𝐮𝐩.\n🅣𝐲𝐩𝐞: kickall confirm\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    await send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅚𝐢𝐜𝐤𝐢𝐧𝐠 ${membersToKick.length} 𝐦𝐞𝐦𝐛𝐞𝐫𝐬...\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    
    let kicked = 0;
    let failed = 0;
    
    for (const uid of membersToKick) {
      try {
        await api.removeUserFromGroup(uid, threadID);
        kicked++;
        await new Promise(r => setTimeout(r, 500));
      } catch {
        failed++;
      }
    }
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅚𝐢𝐜𝐤𝐚𝐥𝐥 🅒𝐨𝐦𝐩𝐥𝐞𝐭𝐞\n─────────────────\n🅚𝐢𝐜𝐤𝐞𝐝: ${kicked}\n🅕𝐚𝐢𝐥𝐞𝐝: ${failed}\n🅡𝐞𝐦𝐚𝐢𝐧𝐢𝐧𝐠: ${adminIDs.length} admins\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
