// lodash-pari/commands/news.js
const { Jimp, JimpMime } = require('jimp');
const axios = require('axios');
const fs = require('fs-extra');
const path = require('path');

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,
    name: "news",
    aliases: ["londybaz"],
    version: "1.0.1",
    credits: "CHAND",
    description: "Funny News",
    category: "group",
    usage: "news",
    prefix: true
  },

  async run({ event, api, args, Users, send }) {
    try {
      const id = Object.keys(event.mentions)[0] || event.senderID;
      const cacheDir = path.join(__dirname, 'cache');
      if (!fs.existsSync(cacheDir)) fs.mkdirSync(cacheDir, { recursive: true });
      const path_toilet = path.join(cacheDir, `toilet_${Date.now()}.png`);

      const bg = await Jimp.read('https://i.ibb.co/xsRjhY1/20250113-025520.png');

      const avatarRes = await axios.get(`https://graph.facebook.com/${id}/picture?width=512&height=512&access_token=6628568379%7Cc1e620fa708a1d5696fb991c1bde5662`, { responseType: 'arraybuffer' });
      let avatar = await Jimp.read(Buffer.from(avatarRes.data));
      avatar.resize({ w: 470, h: 477 });

      bg.composite(avatar, 0, 107);

      const buffer = await bg.getBuffer(JimpMime.png);
      fs.writeFileSync(path_toilet, buffer);
      api.sendMessage({ attachment: fs.createReadStream(path_toilet), body: "📰 Breaking News!" }, event.threadID, () => {
        if (fs.existsSync(path_toilet)) fs.unlinkSync(path_toilet);
      }, event.messageID);
    } catch (e) {
      console.error("News command error:", e);
      api.sendMessage("❌ Error creating news image.", event.threadID, event.messageID);
    }
  }
};
