// lodash-pari/commands/outall.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'outall',
    aliases: ['leaveall', 'exitall'],
    description: 'Leave all groups except current',
    usage: 'outall [confirm]',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, Threads, config }) {
    const { threadID, senderID } = event;
    
    const allThreads = Threads.getAll();
    const groupThreads = allThreads.filter(t => t.id !== threadID);
    
    if (groupThreads.length === 0) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅝𝐨 𝐨𝐭𝐡𝐞𝐫 𝐠𝐫𝐨𝐮𝐩𝐬 𝐭𝐨 𝐥𝐞𝐚𝐯𝐞.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (args[0]?.toLowerCase() !== 'confirm') {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅐𝐫𝐞 𝐲𝐨𝐮 𝐬𝐮𝐫𝐞 𝐲𝐨𝐮 𝐰𝐚𝐧𝐭 𝐭𝐨 𝐥𝐞𝐚𝐯𝐞 𝐠𝐫𝐨𝐮𝐩𝐬?\n🅣𝐡𝐢𝐬 𝐰𝐢𝐥𝐥 𝐦𝐚𝐤𝐞 𝐛𝐨𝐭 𝐥𝐞𝐚𝐯𝐞 🅐𝐋𝐋 𝐠𝐫𝐨𝐮𝐩𝐬 𝐞𝐱𝐜𝐞𝐩𝐭 𝐭𝐡𝐢𝐬 𝐨𝐧𝐞.\n🅣𝐲𝐩𝐞: outall confirm\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    await send.reply(`Checking and leaving active groups...`);
    
    let left = 0;
    let alreadyLeft = 0;
    let failed = 0;
    const botID = api.getCurrentUserID();
    
    for (const thread of groupThreads) {
      try {
        const info = await api.getThreadInfo(thread.id);
        
        if (!info || !info.participantIDs || !info.participantIDs.includes(botID)) {
          alreadyLeft++;
          continue;
        }
        
        await api.removeUserFromGroup(botID, thread.id);
        left++;
        await new Promise(r => setTimeout(r, 1000));
      } catch (err) {
        if (err.message && (err.message.includes('not in') || err.message.includes('already left') || err.message.includes('not a participant'))) {
          alreadyLeft++;
        } else {
          failed++;
        }
      }
    }
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅞𝐮𝐭𝐚𝐥𝐥 🅒𝐨𝐦𝐩𝐥𝐞𝐭𝐞\n─────────────────\n🅛𝐞𝐟𝐭: ${left} groups\n🅐𝐥𝐫𝐞𝐚𝐝𝐲 🅛𝐞𝐟𝐭: ${alreadyLeft}\n🅕𝐚𝐢𝐥𝐞𝐝: ${failed}\n🅡𝐞𝐦𝐚𝐢𝐧𝐢𝐧𝐠 𝐢𝐧 𝐭𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩 𝐨𝐧𝐥𝐲.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
