// lodash-pari/commands/owner.js
const axios = require('axios');
const fs = require('fs-extra');
const path = require('path');

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'owner',
    aliases: ['dev', 'creator', 'developer', 'bot owner', 'coder'],
    description: 'Show bot owner information',
    usage: 'owner',
    category: 'Info',
    prefix: false
  },

  async run({ api, event, send, config }) {
    const { threadID, messageID } = event;

    const ownerPics = [
      'http://i.ibb.co/spvvhygp/cdaab6b4d9c0.jpg',
      'https://i.ibb.co/4RnD4dF4/46e884db844b.jpg',
      'https://i.ibb.co/995z6Rhx/aeba26df1084.jpg',
      'https://i.ibb.co/LhS6Zc0X/a3f0c935f1cf.jpg',
      'https://i.ibb.co/vvHNX4Nr/d264bf533482.jpg',
      'https://i.ibb.co/XZp2tBL0/19ba2831d511.jpg'
    ];

    const randomPic = ownerPics[Math.floor(Math.random() * ownerPics.length)];

    const ownerInfo = `
╔═══════════════════╗
║   ✨ 𝐁𝐎𝐓 𝐎𝐖𝐍𝐄𝐑 𝐈𝐍𝐅𝐎 ✨   
╠═══════════════════╣
║                           
║  👤 𝐍𝐚𝐦𝐞: CHAND TRICKER
║                           
╠═══════════════════╣
║  📱 𝐂𝐨𝐧𝐭𝐚𝐜𝐭 𝐈𝐧𝐟𝐨:          
║                           
║  🌐 𝐅𝐚𝐜𝐞𝐛𝐨𝐨𝐤:              
║  fb.com/CH4ND.TRICKER 
║                           
║  🌐 𝐖𝐞𝐛𝐬𝐢𝐭𝐞 𝐛𝐨𝐭:             
║  https://chandtricker.com/       
║                           
╠═══════════════════╣
║  🤖 𝐁𝐨𝐭 𝐃𝐞𝐭𝐚𝐢𝐥𝐬:           
║                           
║  📛 Name: ${config.BOTNAME || 'CHAND BOT'}
║  ⚡ Prefix: ${config.PREFIX || '.'}
║  💻 Version: 1.0.0        
║                           
╠═══════════════════╣
║  💝 𝙏𝙝𝙖𝙣𝙠 𝙮𝙤𝙪 𝙛𝙤𝙧 𝙪𝙨𝙞𝙣𝙜!    ║
╚═══════════════════╝
    `.trim();

    try {
      const cacheDir = path.join(__dirname, 'cache');
      fs.ensureDirSync(cacheDir);
      const imgPath = path.join(cacheDir, `owner_${Date.now()}.jpg`);
      
      const response = await axios.get(randomPic, { responseType: 'arraybuffer' });
      fs.writeFileSync(imgPath, Buffer.from(response.data));
      
      api.sendMessage(
        {
          body: ownerInfo,
          attachment: fs.createReadStream(imgPath)
        },
        threadID,
        () => {
          try { fs.unlinkSync(imgPath); } catch {}
        },
        messageID
      );
    } catch (error) {
      return send.reply(ownerInfo);
    }
  }
};
