// lodash-pari/commands/pair.js
const axios = require("axios");
const fs = require("fs-extra");
const path = require("path");
const { Jimp } = require("jimp");

module.exports.config = {
  name: "pair",
  aliases: ["couple", "pairing"],
  version: "1.0.0",
  permission: 0,
  prefix: true,
  premium: false,
  category: "Love",
  credits: "KASHIF RAZA",
  description: "Create a romantic pair edit with profile pics",
  usage: "[@mention optional]",
  cooldowns: 5,
};

const cacheDir = path.join(__dirname, "cache", "canvas");
const templateUrl = "https://i.ibb.co/nMcPznDm/d57df01d663b.jpg";
const templatePath = path.join(cacheDir, "pair_new_template.png");

const maleNames = ["ali", "ahmed", "muhammad", "hassan", "hussain", "kashif", "raza", "usman", "bilal", "hamza", "asad", "zain", "fahad", "faisal", "imran", "kamran", "adnan", "arslan", "waqas", "waseem", "irfan", "junaid", "khalid", "nadeem", "naveed", "omer", "qasim", "rizwan", "sajid", "salman", "shahid", "tariq", "umar", "yasir", "zahid"];
const femaleNames = ["fatima", "ayesha", "maria", "sana", "hira", "zara", "maryam", "khadija", "sara", "amina", "bushra", "farah", "iqra", "javeria", "kinza", "laiba", "maham", "nadia", "rabia", "saima", "tahira", "uzma", "zainab", "anam", "asma", "dua", "esha", "fiza", "huma", "iram"];

const romanticMessages = [
  "𝐘𝐨𝐮 𝐚𝐫𝐞 𝐭𝐡𝐞 𝐫𝐞𝐚𝐬𝐨𝐧 𝐈 𝐬𝐦𝐢𝐥𝐞 💖",
  "𝐈𝐧 𝐲𝐨𝐮𝐫 𝐞𝐲𝐞𝐬, 𝐈 𝐟𝐨𝐮𝐧𝐝 𝐡𝐨𝐦𝐞 🏡✨",
  "𝐄𝐯𝐞𝐫𝐲 𝐥𝐨𝐯𝐞 𝐬𝐭𝐨𝐫𝐲 𝐢𝐬 𝐛𝐞𝐚𝐮𝐭𝐢𝐟𝐮𝐥, 𝐛𝐮𝐭 𝐨𝐮𝐫𝐬 𝐢𝐬 𝐦𝐲 𝐟𝐚𝐯𝐨𝐫𝐢𝐭𝐞 💕",
  "𝐘𝐨𝐮 𝐚𝐫𝐞 𝐦𝐲 𝐭𝐨𝐝𝐚𝐲 𝐚𝐧𝐝 𝐚𝐥𝐥 𝐨𝐟 𝐦𝐲 𝐭𝐨𝐦𝐨𝐫𝐫𝐨𝐰𝐬 💝",
  "𝐖𝐢𝐭𝐡 𝐲𝐨𝐮, 𝐈 𝐚𝐦 𝐡𝐨𝐦𝐞 🌹",
  "𝐋𝐨𝐯𝐞 𝐢𝐬 𝐧𝐨𝐭 𝐣𝐮𝐬𝐭 𝐚 𝐟𝐞𝐞𝐥𝐢𝐧𝐠, 𝐢𝐭'𝐬 𝐲𝐨𝐮 💓",
  "𝐓𝐨𝐠𝐞𝐭𝐡𝐞𝐫 𝐢𝐬 𝐦𝐲 𝐟𝐚𝐯𝐨𝐫𝐢𝐭𝐞 𝐩𝐥𝐚𝐜𝐞 𝐭𝐨 𝐛𝐞 💞",
  "𝐘𝐨𝐮 𝐦𝐚𝐤𝐞 𝐦𝐲 𝐡𝐞𝐚𝐫𝐭 𝐬𝐦𝐢𝐥𝐞 ❤️",
  "𝐈𝐧 𝐚 𝐬𝐞𝐚 𝐨𝐟 𝐩𝐞𝐨𝐩𝐥𝐞, 𝐦𝐲 𝐞𝐲𝐞𝐬 𝐰𝐢𝐥𝐥 𝐚𝐥𝐰𝐚𝐲𝐬 𝐬𝐞𝐚𝐫𝐜𝐡 𝐟𝐨𝐫 𝐲𝐨𝐮 👀💕",
  "𝐘𝐨𝐮 𝐚𝐧𝐝 𝐦𝐞, 𝐟𝐨𝐫𝐞𝐯𝐞𝐫 𝐚𝐧𝐝 𝐚𝐥𝐰𝐚𝐲𝐬 🌟"
];

async function downloadTemplate() {
  if (!fs.existsSync(cacheDir)) {
    fs.mkdirSync(cacheDir, { recursive: true });
  }
  if (!fs.existsSync(templatePath)) {
    const response = await axios.get(templateUrl, { responseType: "arraybuffer" });
    fs.writeFileSync(templatePath, Buffer.from(response.data));
  }
}

async function getAvatar(uid) {
  const url = `https://graph.facebook.com/${uid}/picture?width=512&height=512&access_token=6628568379%7Cc1e620fa708a1d5696fb991c1bde5662`;
  const response = await axios.get(url, { responseType: "arraybuffer" });
  return Buffer.from(response.data);
}

async function makeCircularImage(buffer, size) {
  const image = await Jimp.read(buffer);
  image.resize({ w: size, h: size });

  const mask = new Jimp({ width: size, height: size, color: 0x00000000 });
  const center = size / 2;
  const radius = size / 2;

  for (let y = 0; y < size; y++) {
    for (let x = 0; x < size; x++) {
      const dist = Math.sqrt((x - center) ** 2 + (y - center) ** 2);
      if (dist <= radius) {
        mask.setPixelColor(0xFFFFFFFF, x, y);
      }
    }
  }

  image.mask(mask, 0, 0);
  return image;
}

function detectGender(name) {
  const lowerName = name.toLowerCase();
  if (femaleNames.some(n => lowerName.includes(n))) return "female";
  if (maleNames.some(n => lowerName.includes(n))) return "male";
  return "unknown";
}

async function getThreadMembers(api, threadID) {
  return new Promise((resolve) => {
    api.getThreadInfo(threadID, (err, info) => {
      if (err) return resolve([]);
      resolve(info.participantIDs || []);
    });
  });
}

async function getUserInfo(api, uid) {
  return new Promise((resolve) => {
    api.getUserInfo(uid, (err, info) => {
      if (err) return resolve({});
      resolve(info[uid] || {});
    });
  });
}

function isValidName(name) {
  if (!name || name.trim() === '') return false;
  const lower = name.toLowerCase();
  if (lower === 'facebook' || lower === 'facebook user' || lower.includes('facebook user')) return false;
  if (lower === 'unknown' || lower === 'user') return false;
  return true;
}

async function getProperName(api, uid, Users) {
  if (Users && Users.getNameUser) {
    return await Users.getNameUser(uid);
  }
  const info = await getUserInfo(api, uid);
  if (isValidName(info.name)) return info.name;
  if (isValidName(info.firstName)) return info.firstName;
  if (isValidName(info.alternateName)) return info.alternateName;
  return 'Jaan';
}

module.exports.run = async ({ api, event, Users }) => {
  const { threadID, messageID, senderID } = event;
  const mention = Object.keys(event.mentions);

  try {
    await downloadTemplate();

    let one = senderID;
    let two;
    let senderInfo = await getUserInfo(api, senderID);
    let senderGender = senderInfo.gender === 1 ? "female" : senderInfo.gender === 2 ? "male" : detectGender(senderInfo.name || "");

    if (mention[0]) {
      two = mention[0];
    } else {
      const members = await getThreadMembers(api, threadID);
      const filteredMembers = members.filter(m => m !== senderID);

      if (filteredMembers.length === 0) {
        return api.sendMessage("≿━━━━༺❀༻━━━━≾\n❌ 𝐍𝐨 𝐦𝐞𝐦𝐛𝐞𝐫𝐬 𝐟𝐨𝐮𝐧𝐝 𝐭𝐨 𝐩𝐚𝐢𝐫!\n≿━━━━༺❀༻━━━━≾", threadID, messageID);
      }

      let oppositeGenderMembers = [];
      for (const uid of filteredMembers) {
        const info = await getUserInfo(api, uid);
        const memberGender = info.gender === 1 ? "female" : info.gender === 2 ? "male" : detectGender(info.name || "");

        if (senderGender === "male" && memberGender === "female") {
          oppositeGenderMembers.push(uid);
        } else if (senderGender === "female" && memberGender === "male") {
          oppositeGenderMembers.push(uid);
        } else if (senderGender === "unknown" || memberGender === "unknown") {
          oppositeGenderMembers.push(uid);
        }
      }

      if (oppositeGenderMembers.length === 0) {
        oppositeGenderMembers = filteredMembers;
      }

      two = oppositeGenderMembers[Math.floor(Math.random() * oppositeGenderMembers.length)];
    }

    const avatarOne = await getAvatar(one);
    const avatarTwo = await getAvatar(two);

    const circleOne = await makeCircularImage(avatarOne, 280);
    const circleTwo = await makeCircularImage(avatarTwo, 280);

    const template = await Jimp.read(templatePath);

    template.composite(circleOne, 63, 80);
    template.composite(circleTwo, 525, 88);

    const outputPath = path.join(cacheDir, `pair_${one}_${two}_${Date.now()}.png`);
    await template.write(outputPath);

    let nameOne = await getProperName(api, one, Users);
    let nameTwo = await getProperName(api, two, Users);
    const randomMsg = romanticMessages[Math.floor(Math.random() * romanticMessages.length)];

    api.sendMessage(
      {
        body: `≿━━━━༺💝༻━━━━≾\n\n${randomMsg}\n\n👤 ${nameOne}\n💕 𝐏𝐀𝐈𝐑𝐄𝐃 𝐖𝐈𝐓𝐇 💕\n👤 ${nameTwo}\n\n≿━━━━༺💝༻━━━━≾`,
        attachment: fs.createReadStream(outputPath),
        mentions: [
          { tag: nameOne, id: one },
          { tag: nameTwo, id: two }
        ]
      },
      threadID,
      () => fs.unlinkSync(outputPath),
      messageID
    );

  } catch (error) {
    console.error("Pair command error:", error);
    api.sendMessage("≿━━━━༺❀༻━━━━≾\n❌ 𝐄𝐫𝐫𝐨𝐫 𝐜𝐫𝐞𝐚𝐭𝐢𝐧𝐠 𝐩𝐚𝐢𝐫!\n≿━━━━༺❀༻━━━━≾", threadID, messageID);
  }
};
