// lodash-pari/commands/pair4.js
const axios = require("axios");
const fs = require("fs-extra");
const path = require("path");
const { Jimp } = require("jimp");

module.exports.config = {
  name: "pair4",
  aliases: ["couple4"],
  version: "1.0.0",
  permission: 0,
  prefix: true,
  premium: false,
  category: "Love",
  credits: "KASHIF RAZA",
  description: "Create a romantic pair edit with profile pics (Costs 50$)",
  usage: "[@mention optional]",
  cooldowns: 5,
};

const COST = 50;
const cacheDir = path.join(__dirname, "cache", "canvas");
const templateUrl = "https://i.ibb.co/zTSMNfhW/bb84c4de0b9c.jpg";
const templatePath = path.join(cacheDir, "pair4_template.png");

const maleNames = ["ali", "ahmed", "muhammad", "hassan", "hussain", "kashif", "raza", "usman", "bilal", "hamza", "asad", "zain", "fahad", "faisal", "imran", "kamran", "adnan", "arslan", "waqas", "waseem", "irfan", "junaid", "khalid", "nadeem", "naveed", "omer", "qasim", "rizwan", "sajid", "salman", "shahid", "tariq", "umar", "yasir", "zahid"];
const femaleNames = ["fatima", "ayesha", "maria", "sana", "hira", "zara", "maryam", "khadija", "sara", "amina", "bushra", "farah", "iqra", "javeria", "kinza", "laiba", "maham", "nadia", "rabia", "saima", "tahira", "uzma", "zainab", "anam", "asma", "dua", "esha", "fiza", "huma", "iram"];

const romanticMessages = [
  "𝐓𝐮𝐦 𝐦𝐞𝐫𝐢 𝐳𝐢𝐧𝐝𝐚𝐠𝐢 𝐤𝐚 𝐬𝐛 𝐬𝐞 𝐤𝐡𝐨𝐨𝐛𝐬𝐮𝐫𝐚𝐭 𝐬𝐚𝐩𝐧𝐚 𝐡𝐨,\n𝐣𝐨 𝐡𝐚𝐫 𝐫𝐨𝐳 𝐦𝐞𝐫𝐢 𝐚𝐧𝐤𝐡𝐨𝐧 𝐦𝐞𝐢𝐧 𝐬𝐚𝐜𝐡 𝐡𝐨𝐭𝐚 𝐡𝐚𝐢 💕",
  "𝐌𝐨𝐡𝐚𝐛𝐛𝐚𝐭 𝐭𝐨 𝐛𝐚𝐡𝐮𝐭 𝐥𝐨𝐠 𝐤𝐚𝐫𝐭𝐞 𝐡𝐚𝐢𝐧,\n𝐩𝐚𝐫 𝐭𝐮𝐦𝐡𝐚𝐫𝐢 𝐣𝐚𝐢𝐬𝐢 𝐦𝐨𝐡𝐚𝐛𝐛𝐚𝐭 𝐬𝐢𝐫𝐟 𝐦𝐮𝐣𝐡𝐞 𝐦𝐢𝐥𝐢 𝐡𝐚𝐢 💖",
  "𝐓𝐮𝐦𝐡𝐚𝐫𝐢 𝐞𝐤 𝐦𝐮𝐬𝐤𝐚𝐧 𝐦𝐞𝐫𝐞 𝐝𝐢𝐥 𝐤𝐨 𝐬𝐮𝐤𝐨𝐨𝐧 𝐝𝐞𝐭𝐢 𝐡𝐚𝐢,\n𝐚𝐮𝐫 𝐭𝐮𝐦𝐡𝐚𝐫𝐢 𝐲𝐚𝐚𝐝𝐞𝐢𝐧 𝐦𝐮𝐣𝐡𝐞 𝐣𝐞𝐞𝐧𝐞 𝐤𝐚 𝐬𝐚𝐡𝐚𝐫𝐚 𝐝𝐞𝐭𝐢 𝐡𝐚𝐢𝐧 🌹",
  "𝐉𝐚𝐛 𝐛𝐡𝐢 𝐭𝐮𝐦 𝐩𝐚𝐚𝐬 𝐡𝐨𝐭𝐞 𝐡𝐨,\n𝐝𝐮𝐧𝐢𝐲𝐚 𝐤𝐢 𝐬𝐚𝐛𝐬𝐞 𝐤𝐡𝐮𝐬𝐡𝐧𝐚𝐬𝐞𝐞𝐛 𝐣𝐚𝐚𝐧 𝐦𝐚𝐢𝐧 𝐡𝐨𝐭𝐚 𝐡𝐨𝐨𝐧 💝",
  "𝐌𝐞𝐫𝐢 𝐳𝐢𝐧𝐝𝐚𝐠𝐢 𝐦𝐞𝐢𝐧 𝐭𝐮𝐦 𝐚𝐢𝐬𝐞 𝐚𝐚𝐲𝐞,\n𝐣𝐚𝐢𝐬𝐞 𝐛𝐚𝐫𝐢𝐬𝐡 𝐤𝐞 𝐛𝐚𝐚𝐝 𝐝𝐡𝐨𝐨𝐩 𝐤𝐚 𝐧𝐚𝐳𝐚𝐫𝐚 𝐡𝐨 ☀️💕"
];

async function downloadTemplate() {
  if (!fs.existsSync(cacheDir)) {
    fs.mkdirSync(cacheDir, { recursive: true });
  }
  if (!fs.existsSync(templatePath)) {
    const response = await axios.get(templateUrl, { responseType: "arraybuffer" });
    fs.writeFileSync(templatePath, Buffer.from(response.data));
  }
}

async function getAvatar(uid) {
  const url = `https://graph.facebook.com/${uid}/picture?width=512&height=512&access_token=6628568379%7Cc1e620fa708a1d5696fb991c1bde5662`;
  const response = await axios.get(url, { responseType: "arraybuffer" });
  return Buffer.from(response.data);
}

async function makeCircularImage(buffer, size) {
  const image = await Jimp.read(buffer);
  image.resize({ w: size, h: size });

  const mask = new Jimp({ width: size, height: size, color: 0x00000000 });
  const center = size / 2;
  const radius = size / 2;

  for (let y = 0; y < size; y++) {
    for (let x = 0; x < size; x++) {
      const dist = Math.sqrt((x - center) ** 2 + (y - center) ** 2);
      if (dist <= radius) {
        mask.setPixelColor(0xFFFFFFFF, x, y);
      }
    }
  }

  image.mask(mask, 0, 0);
  return image;
}

function detectGender(name) {
  const lowerName = name.toLowerCase();
  if (femaleNames.some(n => lowerName.includes(n))) return "female";
  if (maleNames.some(n => lowerName.includes(n))) return "male";
  return "unknown";
}

async function getThreadMembers(api, threadID) {
  return new Promise((resolve) => {
    api.getThreadInfo(threadID, (err, info) => {
      if (err) return resolve([]);
      resolve(info.participantIDs || []);
    });
  });
}

async function getUserInfo(api, uid) {
  return new Promise((resolve) => {
    api.getUserInfo(uid, (err, info) => {
      if (err) return resolve({});
      resolve(info[uid] || {});
    });
  });
}

function isValidName(name) {
  if (!name || name.trim() === '') return false;
  const lower = name.toLowerCase();
  if (lower === 'facebook' || lower === 'facebook user' || lower.includes('facebook user')) return false;
  if (lower === 'unknown' || lower === 'user') return false;
  return true;
}

async function getProperName(api, uid, Users, mentionName = null) {
  if (mentionName && isValidName(mentionName)) {
    const cleanName = mentionName.replace(/@/g, '').trim();
    if (isValidName(cleanName)) return cleanName;
  }
  
  if (Users && Users.getNameUser) {
    try {
      const name = await Users.getNameUser(uid);
      if (isValidName(name)) return name;
    } catch {}
  }
  
  const info = await getUserInfo(api, uid);
  if (isValidName(info.name)) return info.name;
  if (isValidName(info.firstName)) return info.firstName;
  if (isValidName(info.alternateName)) return info.alternateName;
  return 'Jaan';
}

function getLoveRatio() {
  return Math.floor(Math.random() * 41) + 60;
}

module.exports.run = async ({ api, event, Users, Currencies }) => {
  const { threadID, messageID, senderID } = event;
  const mention = Object.keys(event.mentions);

  try {
    const balance = Currencies.getBalance(senderID);
    if (balance < COST) {
      return api.sendMessage(`≿━━━━༺❀༻━━━━≾\n❌ 𝐀𝐚𝐩 𝐤𝐞 𝐩𝐚𝐚𝐬 𝐤𝐚𝐚𝐟𝐢 𝐛𝐚𝐥𝐚𝐧𝐜𝐞 𝐧𝐚𝐡𝐢 𝐡𝐚𝐢!\n\n💰 Required: $${COST}\n💵 Your Balance: $${balance}\n\n𝐃𝐚𝐢𝐥𝐲 𝐲𝐚 𝐖𝐨𝐫𝐤 𝐜𝐨𝐦𝐦𝐚𝐧𝐝 𝐬𝐞 𝐛𝐚𝐥𝐚𝐧𝐜𝐞 𝐥𝐞𝐢𝐧!\n≿━━━━༺❀༻━━━━≾`, threadID, messageID);
    }

    Currencies.removeBalance(senderID, COST);

    await downloadTemplate();

    let one = senderID;
    let two;
    let senderInfo = await getUserInfo(api, senderID);
    let senderGender = senderInfo.gender === 1 ? "female" : senderInfo.gender === 2 ? "male" : detectGender(senderInfo.name || "");

    if (mention[0]) {
      two = mention[0];
    } else {
      const members = await getThreadMembers(api, threadID);
      const filteredMembers = members.filter(m => m !== senderID);

      if (filteredMembers.length === 0) {
        Currencies.addBalance(senderID, COST);
        return api.sendMessage("≿━━━━༺❀༻━━━━≾\n❌ 𝐍𝐨 𝐦𝐞𝐦𝐛𝐞𝐫𝐬 𝐟𝐨𝐮𝐧𝐝 𝐭𝐨 𝐩𝐚𝐢𝐫!\n≿━━━━༺❀༻━━━━≾", threadID, messageID);
      }

      let oppositeGenderMembers = [];
      for (const uid of filteredMembers) {
        const info = await getUserInfo(api, uid);
        const memberGender = info.gender === 1 ? "female" : info.gender === 2 ? "male" : detectGender(info.name || "");

        if (senderGender === "male" && memberGender === "female") {
          oppositeGenderMembers.push(uid);
        } else if (senderGender === "female" && memberGender === "male") {
          oppositeGenderMembers.push(uid);
        } else if (senderGender === "unknown" || memberGender === "unknown") {
          oppositeGenderMembers.push(uid);
        }
      }

      if (oppositeGenderMembers.length === 0) {
        oppositeGenderMembers = filteredMembers;
      }

      two = oppositeGenderMembers[Math.floor(Math.random() * oppositeGenderMembers.length)];
    }

    const avatarOne = await getAvatar(one);
    const avatarTwo = await getAvatar(two);

    const circleOne = await makeCircularImage(avatarOne, 160);
    const circleTwo = await makeCircularImage(avatarTwo, 160);

    const template = await Jimp.read(templatePath);

    template.composite(circleOne, 85, 160);
    template.composite(circleTwo, 590, 160);

    const outputPath = path.join(cacheDir, `pair4_${one}_${two}_${Date.now()}.png`);
    await template.write(outputPath);

    let mentionNameTwo = mention[0] ? event.mentions[mention[0]] : null;
    let nameOne = await getProperName(api, one, Users);
    let nameTwo = await getProperName(api, two, Users, mentionNameTwo);
    const randomMsg = romanticMessages[Math.floor(Math.random() * romanticMessages.length)];
    const loveRatio = getLoveRatio();

    api.sendMessage(
      {
        body: `≿━━━━༺💝༻━━━━≾\n\n${randomMsg}\n\n👤 ${nameOne}\n💕 𝐏𝐀𝐈𝐑𝐄𝐃 𝐖𝐈𝐓𝐇 💕\n👤 ${nameTwo}\n\n❤️ 𝐋𝐨𝐯𝐞 𝐑𝐚𝐭𝐢𝐨: ${loveRatio}%\n💰 𝐂𝐨𝐬𝐭: $${COST}\n\n≿━━━━༺💝༻━━━━≾`,
        attachment: fs.createReadStream(outputPath),
        mentions: [
          { tag: nameOne, id: one },
          { tag: nameTwo, id: two }
        ]
      },
      threadID,
      () => fs.unlinkSync(outputPath),
      messageID
    );

  } catch (error) {
    console.error("Pair4 command error:", error);
    api.sendMessage("≿━━━━༺❀༻━━━━≾\n❌ 𝐄𝐫𝐫𝐨𝐫 𝐜𝐫𝐞𝐚𝐭𝐢𝐧𝐠 𝐩𝐚𝐢𝐫!\n≿━━━━༺❀༻━━━━≾", threadID, messageID);
  }
};
