// lodash-pari/commands/paste.js
const fs = require("fs");
const path = require("path");

const dataFolder = path.join(__dirname, "cache", "data");

const getUserJsonPath = (userID) => {
  return path.join(dataFolder, `copiedMessage_${userID}.json`);
};

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,
    name: "paste",
    aliases: [],
    version: "1.0.1",
    credits: "CHAND",
    description: "📤 Paste the copied message into this thread",
    category: "Utility",
    usage: "Use in any thread to paste last copied message",
    prefix: true
  },

  async run({ api, event, send }) {
    const { threadID, senderID } = event;
    const userJsonPath = getUserJsonPath(senderID);

    if (!fs.existsSync(userJsonPath)) {
      return send.reply("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ | 🅝𝐨 𝐜𝐨𝐩𝐢𝐞𝐝 𝐦𝐞𝐬𝐬𝐚𝐠𝐞 𝐟𝐨𝐮𝐧𝐝. 🅤𝐬𝐞 `copy` 𝐟𝐢𝐫𝐬𝐭!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━");
    }

    const data = JSON.parse(fs.readFileSync(userJsonPath, "utf-8"));
    
    if (data.timestamp && (Date.now() - data.timestamp) > 3600000) {
      data.attachments.forEach(attPath => {
        if (fs.existsSync(attPath)) {
          fs.unlinkSync(attPath);
        }
      });
      fs.unlinkSync(userJsonPath);
      return send.reply("━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ | 🅨𝐨𝐮𝐫 𝐜𝐨𝐩𝐢𝐞𝐝 𝐦𝐞𝐬𝐬𝐚𝐠𝐞 𝐡𝐚𝐬 𝐞𝐱𝐩𝐢𝐫𝐞𝐝 (1 𝐡𝐨𝐮𝐫 𝐥𝐢𝐦𝐢𝐭). 🅟𝐥𝐞𝐚𝐬𝐞 𝐜𝐨𝐩𝐲 𝐚𝐠𝐚𝐢𝐧!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━");
    }

    const attachments = [];

    for (const attPath of data.attachments) {
      if (fs.existsSync(attPath)) {
        attachments.push(fs.createReadStream(attPath));
      }
    }

    const message = {};
    if (data.body) message.body = data.body;
    if (attachments.length > 0) message.attachment = attachments;

    api.sendMessage(message, threadID);
  }
};
