// lodash-pari/commands/pending.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'pending',
    aliases: ['pendinglist', 'pendingreq'],
    description: 'List pending group requests - reply with number to approve',
    usage: 'pending - Lists all | Reply with number to approve',
    category: 'Utility',
    adminOnly: true,
    prefix: true
  },
  
  pendingData: new Map(),
  
  async run({ api, event, send, client, Threads }) {
    const { threadID, senderID } = event;
    
    const threads = Threads.getAll().filter(t => t.approved !== 1 && t.banned !== 1);
    
    if (threads.length === 0) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅣𝐡𝐞𝐫𝐞 𝐚𝐫𝐞 𝐧𝐨 𝐩𝐞𝐧𝐝𝐢𝐧𝐠 𝐠𝐫𝐨𝐮𝐩𝐬!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    let msg = `📋 PENDING GROUPS (${threads.length})\n`;
    msg += `═══════════════════════\n\n`;
    
    const pendingList = [];
    
    for (let i = 0; i < threads.length; i++) {
      const thread = threads[i];
      pendingList.push({
        index: i + 1,
        id: thread.id,
        name: thread.name || 'Unknown Group'
      });
      
      if (i < 20) {
        msg += `${i + 1}. ${thread.name || 'Unknown'}\n`;
        msg += `   ID: ${thread.id}\n\n`;
      }
    }
    
    if (threads.length > 20) {
      msg += `... aur ${threads.length - 20} more groups\n\n`;
    }
    
    msg += `═══════════════════════\n`;
    msg += `📌 Reply with number to approve\n`;
    msg += `📌 Reply "all" to approve all\n`;
    msg += `📌 Reply "1,3,5" for multiple`;
    
    this.pendingData.set(threadID, pendingList);
    
    const info = await send.reply(msg);
    
    if (client.replies && info?.messageID) {
      client.replies.set(info.messageID, {
        commandName: 'pending',
        author: senderID,
        data: { pendingList, threadID }
      });
      
      setTimeout(() => {
        if (client.replies) client.replies.delete(info.messageID);
        this.pendingData.delete(threadID);
      }, 300000);
    }
  },
  
  async handleReply({ api, event, send, client, Threads, data, config }) {
    const { body, senderID, threadID } = event;
    
    if (!body) return;
    
    const originalAuthor = data?.author;
    const isAdmin = config?.ADMINBOT?.includes(senderID);
    
    if (originalAuthor && senderID !== originalAuthor && !isAdmin) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅞𝐧𝐥𝐲 𝐭𝐡𝐞 𝐩𝐞𝐫𝐬𝐨𝐧 𝐰𝐡𝐨 𝐠𝐚𝐯𝐞 𝐭𝐡𝐞 𝐜𝐨𝐦𝐦𝐚𝐧𝐝 𝐨𝐫 𝐭𝐡𝐞 𝐚𝐝𝐦𝐢𝐧𝐢𝐬𝐭𝐫𝐚𝐭𝐨𝐫 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐭𝐡𝐢𝐬 𝐫𝐞𝐩𝐥𝐲.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const pendingList = data?.pendingList || this.pendingData.get(threadID);
    
    if (!pendingList || pendingList.length === 0) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅟𝐞𝐧𝐝𝐢𝐧𝐠 𝐝𝐚𝐭𝐚 𝐡𝐚𝐬 𝐞𝐱𝐩𝐢𝐫𝐞𝐝, 𝐫𝐮𝐧 𝐩𝐞𝐧𝐝𝐢𝐧𝐠 𝐚𝐠𝐚𝐢𝐧.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    const input = body.trim().toLowerCase();
    
    let toApprove = [];
    
    if (input === 'all') {
      toApprove = pendingList;
    } else if (input.includes(',')) {
      const nums = input.split(',').map(n => parseInt(n.trim())).filter(n => !isNaN(n));
      for (const num of nums) {
        const item = pendingList.find(p => p.index === num);
        if (item) toApprove.push(item);
      }
    } else {
      const num = parseInt(input);
      if (!isNaN(num)) {
        const item = pendingList.find(p => p.index === num);
        if (item) toApprove.push(item);
      }
    }
    
    if (toApprove.length === 0) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅘𝐧𝐯𝐚𝐥𝐢𝐝 𝐧𝐮𝐦𝐛𝐞𝐫. 🅛𝐢𝐬𝐭 𝐦𝐞𝐢𝐧 𝐬𝐞 𝐧𝐮𝐦𝐛𝐞𝐫 𝐜𝐡𝐨𝐨𝐬𝐞 𝐤𝐚𝐫𝐨.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    await send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n⏳ ${toApprove.length} group(s) 𝐢𝐬 𝐬𝐭𝐢𝐥𝐥 𝐛𝐞𝐢𝐧𝐠 𝐚𝐩𝐩𝐫𝐨𝐯𝐞𝐝...\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    
    let approved = 0;
    let failed = 0;
    let results = [];
    
    for (const item of toApprove) {
      try {
        Threads.update(item.id, { approved: 1 });
        
        try {
          await api.sendMessage('✅ Group approved! Bot is now active.', item.id);
        } catch {}
        
        approved++;
        results.push(`✅ ${item.name}`);
        
        await new Promise(r => setTimeout(r, 500));
      } catch (error) {
        failed++;
        results.push(`❌ ${item.name}`);
      }
    }
    
    let resultMsg = `📋 APPROVE RESULTS\n`;
    resultMsg += `═══════════════════════\n`;
    resultMsg += `✅ Approved: ${approved}\n`;
    resultMsg += `❌ Failed: ${failed}\n`;
    resultMsg += `───────────────────────\n`;
    resultMsg += results.slice(0, 10).join('\n');
    if (results.length > 10) {
      resultMsg += `\n... aur ${results.length - 10} more`;
    }
    
    this.pendingData.delete(threadID);
    
    return send.reply(resultMsg);
  }
};
