// lodash-pari/commands/rankup.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'rankup',
    aliases: ['rank', 'levelup'],
    description: 'Toggle rankup notifications on/off for this group',
    usage: 'rankup [on/off]',
    category: 'Group',
    groupOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, Threads }) {
    const { threadID, senderID } = event;
    
    const threadInfo = await api.getThreadInfo(threadID);
    const adminIDs = threadInfo.adminIDs.map(a => a.id);
    
    const isGroupAdmin = adminIDs.includes(senderID);
    
    const settings = Threads.getSettings(threadID);
    const action = args[0]?.toLowerCase();
    
    if (!action) {
      const currentStatus = settings.rankup !== false ? 'ON' : 'OFF';
      return send.reply(`━❮📊❯━━━❪✨❫━━━❮📊❯━\n🔒 🅡𝐚𝐧𝐤𝐮𝐩 🅢𝐭𝐚𝐭𝐮𝐬: ${currentStatus}\n🅤𝐬𝐚𝐠𝐞:\n• 𝐫𝐚𝐧𝐤𝐮𝐩 𝐨𝐧 - 🟢 Enable rankup notifications\n• 𝐫𝐚𝐧𝐤𝐮𝐩 𝐨𝐟𝐟 - 🔴 Disable rankup notifications\n━❮📊❯━━━❪✨❫━━━❮📊❯━`);
    }
    
    if (action === 'on' || action === 'enable') {
      Threads.setSetting(threadID, 'rankup', true);
      return send.reply('━❮✅❯━━━❪🎉❫━━━❮✅❯━\n🔔 🅡𝐚𝐧𝐤𝐮𝐩 𝐧𝐨𝐭𝐢𝐟𝐢𝐜𝐚𝐭𝐢𝐨𝐧𝐬 𝐞𝐧𝐚𝐛𝐥𝐞𝐝!\n\n🅤𝐬𝐞𝐫𝐬 𝐰𝐢𝐥𝐥 𝐧𝐨𝐰 𝐫𝐞𝐜𝐞𝐢𝐯𝐞 𝐥𝐞𝐯𝐞𝐥 𝐮𝐩 𝐚𝐧𝐧𝐨𝐮𝐧𝐜𝐞𝐦𝐞𝐧𝐭𝐬.\n━❮✅❯━━━❪🎉❫━━━❮✅❯━');
    }
    
    if (action === 'off' || action === 'disable') {
      Threads.setSetting(threadID, 'rankup', false);
      return send.reply('❌ Rankup notifications disabled.');
    }
    
    return send.reply('Invalid option. Use: rankup on/off');
  }
};
