// lodash-pari/commands/restart.js
const { spawn } = require('child_process');
const path = require('path');
const fs = require('fs-extra');

module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'restart',
    aliases: ['reboot'],
    description: 'Reload all bot commands and modules',
    usage: 'restart',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },

  async run({ api, event, send, config }) {
    await send.reply(`━❮🔄❯━━━❪⚙️❫━━━❮🔄❯━\n🔒 🅡𝐞𝐥𝐨𝐚𝐝𝐢𝐧𝐠 ${config.BOTNAME}...\n━❮🔄❯━━━❪⚙️❫━━━❮🔄❯━`);

    try {
      const result = await global.reloadChand();

      if (result.success) {
        await send.reply(`━❮✅❯━━━❪⚙️❫━━━❮✅❯━\n🅒𝐡𝐚𝐧𝐝 𝐦𝐨𝐝𝐮𝐥𝐞 𝐫𝐞𝐥𝐨𝐚𝐝𝐞𝐝 𝐬𝐮𝐜𝐜𝐞𝐬𝐬𝐟𝐮𝐥𝐥𝐲!\n━❮✅❯━━━❪⚙️❫━━━❮✅❯━`);
      } else {
        await send.reply(`━❮❌❯━━━❪⚙️❫━━━❮❌❯━\n🅡𝐞𝐥𝐨𝐚𝐝 𝐟𝐚𝐢𝐥𝐞𝐝: ${result.error}\n━❮❌❯━━━❪⚙️❫━━━❮❌❯━`);
      }
    } catch (error) {
      await send.reply(`━❮❌❯━━━❪⚙️❫━━━❮❌❯━\n🅡𝐞𝐥𝐨𝐚𝐝 𝐞𝐫𝐫𝐨𝐫: ${error.message}\n━❮❌❯━━━❪⚙️❫━━━❮❌❯━`);
    }
  }
};
