// lodash-pari/commands/screenshot.js
const axios = require('axios');
const fs = require('fs-extra');
const path = require('path');

module.exports = {
  config: {
cooldowns: 10,
premium: false,
permission: 0,

    name: "screenshot",
    aliases: ["ss", "webshot"],
    version: "1.0.0",
    credits: "RAZA",
    description: "Take screenshot of a website",
    category: "Utility",
    usage: "screenshot <URL> [type: windows/mobile]",
    prefix: true
  },

  async run({ api, event, args, send }) {
    const { threadID, messageID } = event;
    const url = args[0];
    const type = args[1]?.toLowerCase() || 'windows';

    if (!url) {
      return api.sendMessage("━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅤𝐬𝐚𝐠𝐞: 𝐬𝐜𝐫𝐞𝐞𝐧𝐬𝐡𝐨𝐭 <𝐔𝐑𝐋> [𝐰𝐢𝐧𝐝𝐨𝐰𝐬/𝐦𝐨𝐛𝐢𝐥𝐞]\n\n🅔𝐱𝐚𝐦𝐩𝐥𝐞: 𝐬𝐜𝐫𝐞𝐞𝐧𝐬𝐡𝐨𝐭 𝐡𝐭𝐭𝐩𝐬://𝐠𝐨𝐨𝐠𝐥𝐞.𝐜𝐨𝐦 𝐦𝐨𝐛𝐢𝐥𝐞\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━", threadID, messageID);
    }

    if (!url.startsWith('https://') && !url.startsWith('http://')) {
      return api.sendMessage("❌ URL must start with https:// or http://", threadID, messageID);
    }

    try {
      api.sendMessage(`📸 Taking screenshot (${type})...`, threadID);

      const cacheDir = path.join(__dirname, 'cache');
      if (!fs.existsSync(cacheDir)) fs.mkdirSync(cacheDir, { recursive: true });

      const imageUrl = await generateScreenshot(url, type);
      const imageRes = await axios.get(imageUrl, { responseType: 'arraybuffer' });
      const imagePath = path.join(cacheDir, `screenshot_${Date.now()}.png`);
      fs.writeFileSync(imagePath, Buffer.from(imageRes.data));

      return api.sendMessage({
        body: `📸 Website Screenshot\n\nURL: ${url}`,
        attachment: fs.createReadStream(imagePath)
      }, threadID, () => {
        if (fs.existsSync(imagePath)) fs.unlinkSync(imagePath);
      }, messageID);

    } catch (error) {
      console.error("Screenshot command error:", error);
      return api.sendMessage(`❌ Error: ${error.message}`, threadID, messageID);
    }
  }
};

async function generateScreenshot(url, type) {
  let width = 1920;
  let height = 1080;
  let scale = 1;

  if (type.toLowerCase() === 'mobile') {
    width = 375;
    height = 812;
    scale = 2;
  }

  const payload = {
    url: url.startsWith('https://') ? url : 'https://' + url,
    browserWidth: width,
    browserHeight: height,
    fullPage: false,
    deviceScaleFactor: scale,
    format: 'png'
  };

  const { data } = await axios.post('https://gcp.imagy.app/screenshot/createscreenshot', payload, {
    headers: {
      'content-type': 'application/json',
      referer: 'https://imagy.app/full-page-screenshot-taker/',
      'user-agent': 'Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36'
    }
  });

  if (!data.fileUrl) throw new Error('Failed to generate screenshot');
  return data.fileUrl;
}
