// lodash-pari/commands/setdatagroup.js
module.exports = {
  config: {
    name: "setdatagroup",
    aliases: ["sdgroup"],
    usage: "setdatagroup",
    version: "1.0",
    permission: 0,
    credits: "CHAND",
    premium: false,
    prefix: true,
    description: "Set new data of groups into database",
    category: "admin",
    cooldowns: 5
  },

  async run({ event, args, api, Threads }) { 
    const { threadID } = event;
    
    try {
      const inbox = await api.getThreadList(100, null, ['INBOX']);
      const list = [...inbox].filter(group => group.isSubscribed && group.isGroup);
      const lengthGroup = list.length;
      
      if (lengthGroup === 0) {
        return api.sendMessage('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅝𝐨 𝐠𝐫𝐨𝐮𝐩𝐬 𝐟𝐨𝐮𝐧𝐝.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━', threadID);
      }
      
      let updatedCount = 0;
      
      for (const groupInfo of list) {
        try {
          const threadInfo = await api.getThreadInfo(groupInfo.threadID);
          const groupName = threadInfo.threadName || threadInfo.name || 'Unknown Group';
          
          await Threads.setData(groupInfo.threadID, { 
            name: groupName,
            threadInfo: threadInfo
          });
          
          console.log(`Data updated for group ID: ${groupInfo.threadID} - Name: ${groupName}`);
          updatedCount++;
        } catch (e) {
          console.log(`Failed to update group ${groupInfo.threadID}: ${e.message}`);
        }
      }
      
      console.log(`Data updated for ${updatedCount} groups`);
      return api.sendMessage(`✅ Data updated for ${updatedCount} groups.`, threadID);
    } catch (error) {
      console.error('Error in setdatagroup:', error.message);
      return api.sendMessage(`❌ Error: ${error.message}`, threadID);
    }
  }
};
