// lodash-pari/commands/setdatauser.js
module.exports = {
  config: {
    name: "setdatauser",
    aliases: ["sduser"],
    usage: "setdatauser",
    version: "1.0",
    permission: 0,
    credits: "CHAND",
    prefix: true,
    premium: false,
    description: "Set new data of users into database",
    category: "admin",
    cooldowns: 5,
  },

  async run({ Users, event, args, api, Threads }) { 
    const { threadID } = event;
    
    try {
      let threadInfo;
      try {
        threadInfo = await Threads.getInfo(threadID);
      } catch (e) {
        threadInfo = await api.getThreadInfo(threadID);
      }
      
      const participantIDs = threadInfo.participantIDs || [];
      
      if (participantIDs.length === 0) {
        return api.sendMessage('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅝𝐨 𝐩𝐚𝐫𝐭𝐢𝐜𝐢𝐩𝐚𝐧𝐭𝐬 𝐟𝐨𝐮𝐧𝐝 𝐢𝐧 𝐭𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━', threadID);
      }
      
      let updatedCount = 0;
      
      for (const id of participantIDs) {
        try {
          const userInfo = await api.getUserInfo(id);
          const userName = userInfo[id]?.name || 'Unknown';
          
          await Users.setData(id, { name: userName });
          updatedCount++;
          
          console.log(`Data updated for user ID: ${id} - Name: ${userName}`);
        } catch (e) {
          console.log(`Failed to update user ${id}: ${e.message}`);
        }
      }
      
      console.log(`Updated data of ${updatedCount} users in group`);
      return api.sendMessage(`✅ Updated data of ${updatedCount} users in this group.`, threadID);
    } catch (error) {
      console.error('Error in setdatauser:', error.message);
      return api.sendMessage(`❌ Error: ${error.message}`, threadID);
    }
  }
};
