// lodash-pari/commands/setnicall.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'setnicall',
    aliases: ['nickall', 'allnick'],
    description: 'Set nickname for all members',
    usage: 'setnicall [nickname]',
    category: 'Group',
    groupOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { threadID, senderID } = event;
    
    const threadInfo = await api.getThreadInfo(threadID);
    const adminIDs = threadInfo.adminIDs.map(a => a.id);
    const botID = api.getCurrentUserID();
    
    const isGroupAdmin = adminIDs.includes(senderID);
    
    const nickname = args.join(' ');
    
    if (!nickname) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐧𝐢𝐜𝐤𝐧𝐚𝐦𝐞.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    const members = threadInfo.participantIDs || [];
    
    await send.reply(`━❮⚙️❯━━━❪✨❫━━━❮⚙️❯━\n🔒 🅢𝐞𝐭𝐭𝐢𝐧𝐠 𝐧𝐢𝐜𝐤𝐧𝐚𝐦𝐞 𝐟𝐨𝐫 ${members.length} 𝐦𝐞𝐦𝐛𝐞𝐫𝐬...\n━❮⚙️❯━━━❪✨❫━━━❮⚙️❯━`);
    
    let success = 0;
    let failed = 0;
    
    for (const uid of members) {
      try {
        await api.changeNickname(nickname, threadID, uid);
        success++;
        await new Promise(r => setTimeout(r, 500));
      } catch {
        failed++;
      }
    }
    
    return send.reply(`━❮✅❯━━━❪🎉❫━━━❮✅❯━\n🔒 🅝𝐢𝐜𝐤𝐧𝐚𝐦𝐞 🅒𝐡𝐚𝐧𝐠𝐞 🅒𝐨𝐦𝐩𝐥𝐞𝐭𝐞!\n─────────────────\n✅ 🅢𝐮𝐜𝐜𝐞𝐬𝐬: ${success}\n❌ 🅕𝐚𝐢𝐥𝐞𝐝: ${failed}\n━❮✅❯━━━❪🎉❫━━━❮✅❯━`);
  }
};
