// lodash-pari/commands/setname.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'setname',
    aliases: ['groupname', 'rename'],
    description: 'Change the group name',
    usage: 'setname [new name]',
    category: 'Group',
    groupOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config }) {
    const { threadID, senderID } = event;
    
    const threadInfo = await api.getThreadInfo(threadID);
    const adminIDs = threadInfo.adminIDs.map(a => a.id);
    
    const isGroupAdmin = adminIDs.includes(senderID);
    
    const newName = args.join(' ');
    
    if (!newName) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐧𝐞𝐰 𝐠𝐫𝐨𝐮𝐩 𝐧𝐚𝐦𝐞.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    try {
      await api.setTitle(newName, threadID);
      return send.reply(`━❮✅❯━━━❪🎉❫━━━❮✅❯━\n🔒 🅖𝐫𝐨𝐮𝐩 𝐧𝐚𝐦𝐞 𝐜𝐡𝐚𝐧𝐠𝐞𝐝 𝐭𝐨: ${newName}\n━❮✅❯━━━❪🎉❫━━━❮✅❯━`);
    } catch (error) {
      return send.reply('Failed to change group name.');
    }
  }
};
