// lodash-pari/commands/setprefix.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 1,
version: '1.0.0',
    name: 'setprefix',
    aliases: ['changeprefix'],
    description: 'Change the bot prefix',
    usage: 'setprefix [new prefix]',
    category: 'Utility',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send }) {
    const fs = require('fs-extra');
    const path = require('path');
    const configPath = path.join(__dirname, '../../../config.json');
    
    const newPrefix = args[0];
    
    if (!newPrefix) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐧𝐞𝐰 𝐩𝐫𝐞𝐟𝐢𝐱.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    if (newPrefix.length > 5) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐫𝐞𝐟𝐢𝐱 𝐜𝐚𝐧𝐧𝐨𝐭 𝐛𝐞 𝐥𝐨𝐧𝐠𝐞𝐫 𝐭𝐡𝐚𝐧 𝟓 𝐜𝐡𝐚𝐫𝐚𝐜𝐭𝐞𝐫𝐬.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    try {
      const configFile = fs.readFileSync(configPath, 'utf8');
      const config = JSON.parse(configFile);
      const oldPrefix = config.PREFIX;
      config.PREFIX = newPrefix;
      
      const configContent = JSON.stringify(config, null, 2);
      fs.writeFileSync(configPath, configContent);
      
      global.config = config;
      
      return send.reply(`━❮✅❯━━━❪🎉❫━━━❮✅❯━\n🔒 🅟𝐫𝐞𝐟𝐢𝐱 𝐜𝐡𝐚𝐧𝐠𝐞𝐝!\n─────────────────\n🔄 🅞𝐥𝐝: ${oldPrefix}\n✅ 🅝𝐞𝐰: ${newPrefix}\n━❮✅❯━━━❪🎉❫━━━❮✅❯━`);
    } catch (error) {
      return send.reply('❌ Error changing prefix: ' + error.message);
    }
  }
};
