// lodash-pari/commands/sticker.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'sticker',
    aliases: ['stickerid', 'sid'],
    description: 'Get sticker ID by replying to a sticker',
    usage: 'sticker (reply to sticker)',
    category: 'Utility',
    prefix: true
  },
  
  async run({ api, event, send }) {
    const { messageReply } = event;
    
    if (!messageReply) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐥𝐞𝐚𝐬𝐞 𝐫𝐞𝐩𝐥𝐲 𝐭𝐨 𝐚 𝐬𝐭𝐢𝐜𝐤𝐞𝐫 𝐰𝐢𝐭𝐡 𝐭𝐡𝐢𝐬 𝐜𝐨𝐦𝐦𝐚𝐧𝐝.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    const attachments = messageReply.attachments || [];
    
    if (attachments.length === 0) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅝𝐨 𝐬𝐭𝐢𝐜𝐤𝐞𝐫 𝐟𝐨𝐮𝐧𝐝 𝐢𝐧 𝐭𝐡𝐞 𝐫𝐞𝐩𝐥𝐢𝐞𝐝 𝐦𝐞𝐬𝐬𝐚𝐠𝐞.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    const sticker = attachments.find(a => a.type === 'sticker');
    
    if (!sticker) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅣𝐡𝐞 𝐫𝐞𝐩𝐥𝐢𝐞𝐝 𝐦𝐞𝐬𝐬𝐚𝐠𝐞 𝐝𝐨𝐞𝐬 𝐧𝐨𝐭 𝐜𝐨𝐧𝐭𝐚𝐢𝐧 𝐚 𝐬𝐭𝐢𝐜𝐤𝐞𝐫.\n\n🅟𝐥𝐞𝐚𝐬𝐞 𝐫𝐞𝐩𝐥𝐲 𝐭𝐨 𝐚 𝐬𝐭𝐢𝐜𝐤𝐞𝐫.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    const stickerID = sticker.stickerID || sticker.ID || sticker.id || 'Unknown';
    const packID = sticker.packID || 'Unknown';
    const spriteURI = sticker.spriteURI || sticker.url || 'N/A';
    const frameCount = sticker.frameCount || 1;
    const frameRate = sticker.frameRate || 0;
    
    return send.reply(`STICKER INFO
═══════════════════════
Sticker ID: ${stickerID}
Pack ID: ${packID}
Frames: ${frameCount}
Frame Rate: ${frameRate}
═══════════════════════
URL: ${spriteURI}
═══════════════════════

You can use this sticker ID to send this sticker programmatically.`);
  }
};
