// lodash-pari/commands/theme.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'theme',
    aliases: ['atheme', 'aitheme', 'themea', 'gentheme'],
    description: 'Generate and apply AI theme or change group theme',
    usage: 'theme [ai:prompt] or theme [color name]',
    category: 'Group Management',
    adminOnly: true,
    groupOnly: true,
    prefix: true
  },

  async run({ api, event, args, send }) {
    const { threadID } = event;

    if (!args.length) {
      return send.reply(`━❮🎨❯━━━❪✨❫━━━❮🎨❯━\n🔒 🅣𝐡𝐞𝐦𝐞 🅒𝐨𝐦𝐦𝐚𝐧𝐝 🅗𝐞𝐥𝐩\n━━━━━━━━━━━━━━━━━━━━━\n📌 🅐𝐈 🅣𝐡𝐞𝐦𝐞 🅖𝐞𝐧𝐞𝐫𝐚𝐭𝐢𝐨𝐧:\n.𝐭𝐡𝐞𝐦𝐞 𝐚𝐢:𝐝𝐞𝐬𝐜𝐫𝐢𝐛𝐞 𝐲𝐨𝐮𝐫 𝐭𝐡𝐞𝐦𝐞 𝐢𝐝𝐞𝐚\n🅔𝐱𝐚𝐦𝐩𝐥𝐞: .𝐭𝐡𝐞𝐦𝐞 𝐚𝐢:𝐝𝐚𝐫𝐤 𝐩𝐮𝐫𝐩𝐥𝐞 𝐰𝐢𝐭𝐡 𝐬𝐭𝐚𝐫𝐬\n━━━━━━━━━━━━━━━━━━━━━\n📌 🅒𝐨𝐥𝐨𝐫 🅣𝐡𝐞𝐦𝐞𝐬:\n.𝐜𝐨𝐥𝐨𝐫 [𝐜𝐨𝐥𝐨𝐫_𝐧𝐚𝐦𝐞]\n🅔𝐱𝐚𝐦𝐩𝐥𝐞: .𝐜𝐨𝐥𝐨𝐫 𝐛𝐥𝐮𝐞, .𝐜𝐨𝐥𝐨𝐫 𝐫𝐞𝐝\n━━━━━━━━━━━━━━━━━━━━━\n🎨 🅐𝐯𝐚𝐢𝐥𝐚𝐛𝐥𝐞 🅒𝐨𝐥𝐨𝐫𝐬:\n𝐃𝐞𝐟𝐚𝐮𝐥𝐭𝐁𝐥𝐮𝐞, 𝐑𝐞𝐝, 𝐓𝐞𝐚𝐥𝐁𝐥𝐮𝐞, 𝐘𝐞𝐥𝐥𝐨𝐰, 𝐎𝐫𝐚𝐧𝐠𝐞, 𝐆𝐫𝐞𝐞𝐧,\n𝐋𝐚𝐯𝐞𝐧𝐝𝐞𝐫𝐏𝐮𝐫𝐩𝐥𝐞, 𝐇𝐨𝐭𝐏𝐢𝐧𝐤, 𝐀𝐪𝐮𝐚𝐁𝐥𝐮𝐞, 𝐁𝐫𝐢𝐠𝐡𝐭𝐏𝐮𝐫𝐩𝐥𝐞,\n𝐂𝐨𝐫𝐚𝐥𝐏𝐢𝐧𝐤, 𝐀𝐪𝐮𝐚, 𝐌𝐚𝐧𝐠𝐨, 𝐁𝐞𝐫𝐫𝐲, 𝐂𝐢𝐭𝐫𝐮𝐬, 𝐂𝐚𝐧𝐝𝐲,\n𝐄𝐚𝐫𝐭𝐡, 𝐒𝐮𝐩𝐩𝐨𝐫𝐭, 𝐌𝐮𝐬𝐢𝐜, 𝐏𝐫𝐢𝐝𝐞, 𝐃𝐨𝐜𝐭𝐨𝐫𝐒𝐭𝐫𝐚𝐧𝐠𝐞, 𝐋𝐨𝐅𝐢,\n𝐒𝐤𝐲, 𝐋𝐮𝐧𝐚𝐫𝐍𝐞𝐰𝐘𝐞𝐚𝐫, 𝐂𝐞𝐥𝐞𝐛𝐫𝐚𝐭𝐢𝐨𝐧, 𝐂𝐡𝐢𝐥𝐥, 𝐒𝐭𝐫𝐚𝐧𝐠𝐞𝐫𝐓𝐡𝐢𝐧𝐠𝐬,\n𝐃𝐮𝐧𝐞, 𝐂𝐚𝐫𝐞, 𝐀𝐬𝐭𝐫𝐨𝐥𝐨𝐠𝐲, 𝐁𝐢𝐫𝐭𝐡𝐝𝐚𝐲, 𝐂𝐨𝐭𝐭𝐚𝐠𝐞𝐜𝐨𝐫𝐞, 𝐎𝐜𝐞𝐚𝐧,\n𝐋𝐨𝐯𝐞, 𝐓𝐢𝐞𝐃𝐲𝐞, 𝐌𝐨𝐧𝐨𝐜𝐡𝐫𝐨𝐦𝐞, 𝐑𝐨𝐜𝐤𝐞𝐭, 𝐔𝐧𝐢𝐜𝐨𝐫𝐧, 𝐓𝐫𝐨𝐩𝐢𝐜𝐚𝐥,\n𝐌𝐚𝐩𝐥𝐞, 𝐒𝐮𝐬𝐡𝐢, 𝐒𝐡𝐚𝐝𝐨𝐰, 𝐑𝐨𝐬𝐞, 𝐋𝐚𝐯𝐞𝐧𝐝𝐞𝐫, 𝐓𝐮𝐥𝐢𝐩, 𝐏𝐞𝐚𝐜𝐡,\n𝐇𝐨𝐧𝐞𝐲, 𝐊𝐢𝐰𝐢, 𝐆𝐫𝐚𝐩𝐞, 𝐀𝐮𝐭𝐮𝐦𝐧, 𝐂𝐲𝐛𝐞𝐫𝐩𝐮𝐧𝐤𝟐𝟎𝟕𝟕, & 𝐦𝐨𝐫𝐞!\n━❮🎨❯━━━❪✨❫━━━❮🎨❯━`);
    }

    const input = args.join(' ').toLowerCase();

    // Check if AI theme
    if (input.startsWith('ai:')) {
      const prompt = args.join(' ').substring(3).trim();
      
      if (!prompt) {
        return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐩𝐫𝐨𝐦𝐩𝐭 𝐟𝐨𝐫 🅐𝐈 𝐭𝐡𝐞𝐦𝐞 𝐠𝐞𝐧𝐞𝐫𝐚𝐭𝐢𝐨𝐧.\n\n🅤𝐬𝐚𝐠𝐞: .𝐭𝐡𝐞𝐦𝐞 𝐚𝐢:𝐲𝐨𝐮𝐫 𝐭𝐡𝐞𝐦𝐞 𝐝𝐞𝐬𝐜𝐫𝐢𝐩𝐭𝐢𝐨𝐧\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
      }

      try {
        send.reply('🎨 Generating AI theme... Please wait (this may take up to 30 seconds)');
        
        const theme = await api.createThemeAI(prompt);
        
        if (theme && theme.id) {
          // Apply the theme
          await api.changeThreadColor(theme.id, threadID);
          
          let response = `✅ AI Theme Applied Successfully!
━━━━━━━━━━━━━━━━━━━━━
🎨 Theme: ${theme.accessibility_label || 'Custom AI Theme'}`;
          
          if (theme.background_asset?.image?.url) {
            response += `\n🖼️ Image: ${theme.background_asset.image.url}`;
          }
          
          return send.reply(response);
        } else {
          return send.reply('❌ Failed to generate theme. Please try again with a different prompt.');
        }
      } catch (error) {
        return send.reply(`❌ AI Theme Generation Failed: ${error.message || 'Unknown error'}`);
      }
    }

    // Regular color theme
    try {
      return send.reply('💡 Use .color command for color themes instead.\n\nExample: .color blue');
    } catch (error) {
      return send.reply(`Error: ${error.message}`);
    }
  }
};
