// lodash-pari/commands/tiktok.js
const axios = require('axios');
const fs = require('fs-extra');
const path = require('path');

module.exports.config = {
    name: "tiktok",
    aliases: ["tt", "tik"],
    version: "1.0.0",
    permission: 0,
    prefix: true,
    premium: false,
    category: "Media",
    credits: "CHAND",
    description: "Download TikTok video",
    usage: ".tiktok [TikTok URL]",
    cooldowns: 10
};

const API_BASE = "https://api.kraza.qzz.io";

async function downloadTikTok(tiktokUrl) {
    try {
        const response = await axios.get(`${API_BASE}/download/tiktok`, {
            params: { url: tiktokUrl },
            timeout: 60000
        });
        
        if (response.data && response.data.status && response.data.result) {
            const videoUrl = response.data.result.video_nowm;
            const videoRes = await axios.get(videoUrl, {
                timeout: 60000,
                responseType: 'arraybuffer'
            });
            return { success: true, data: videoRes.data };
        }
        return null;
    } catch (err) {
        console.log("TikTok download failed:", err.message);
        return null;
    }
}

module.exports.run = async function ({ api, event, args }) {
    const url = args[0];
    
    if (!url) {
        return api.sendMessage("в”Ғвқ®вқҢвқҜв”Ғв”Ғв”ҒвқӘвҡ пёҸвқ«в”Ғв”Ғв”Ғвқ®вқҢвқҜв”Ғ\nрҹ”’ рҹ…ҹрқҗҘрқҗһрқҗҡрқҗ¬рқҗһ рқҗ©рқҗ«рқҗЁрқҗҜрқҗўрқҗқрқҗһ рқҗҡ рҹ…ЈрқҗўрқҗӨрқҗ“рқҗЁрқҗӨ рҹ…Өрқҗ‘рқҗӢ\n\nрҹ…Өрқҗ¬рқҗҡрқҗ рқҗһ: .рқҗӯрқҗўрқҗӨрқҗӯрқҗЁрқҗӨ [рқҗ”рқҗ‘рқҗӢ]\nв”Ғвқ®вқҢвқҜв”Ғв”Ғв”ҒвқӘвҡ пёҸвқ«в”Ғв”Ғв”Ғвқ®вқҢвқҜв”Ғ", event.threadID, event.messageID);
    }

    const tiktokRegex = /(?:https?:\/\/)?(?:www\.|vm\.|vt\.)?(?:tiktok\.com)\/.+/i;
    
    if (!tiktokRegex.test(url)) {
        return api.sendMessage("в”Ғвқ®вқҢвқҜв”Ғв”Ғв”ҒвқӘвҡ пёҸвқ«в”Ғв”Ғв”Ғвқ®вқҢвқҜв”Ғ\nрҹ”’ рҹ…ҳрқҗ§рқҗҜрқҗҡрқҗҘрқҗўрқҗқ рҹ…ЈрқҗўрқҗӨрқҗ“рқҗЁрқҗӨ рҹ…Өрқҗ‘рқҗӢ. рҹ…ҹрқҗҘрқҗһрқҗҡрқҗ¬рқҗһ рқҗ©рқҗ«рқҗЁрқҗҜрқҗўрқҗқрқҗһ рқҗҡ рқҗҜрқҗҡрқҗҘрқҗўрқҗқ рҹ…ЈрқҗўрқҗӨрқҗ“рқҗЁрқҗӨ рқҗҘрқҗўрқҗ§рқҗӨ.\nв”Ғвқ®вқҢвқҜв”Ғв”Ғв”ҒвқӘвҡ пёҸвқ«в”Ғв”Ғв”Ғвқ®вқҢвқҜв”Ғ", event.threadID, event.messageID);
    }

    const frames = [
        "рҹ©өв–°в–ұв–ұв–ұв–ұв–ұв–ұв–ұв–ұв–ұ 10%",
        "рҹ’ҷв–°в–°в–ұв–ұв–ұв–ұв–ұв–ұв–ұв–ұ 25%",
        "рҹ’ңв–°в–°в–°в–°в–ұв–ұв–ұв–ұв–ұв–ұ 45%",
        "рҹ’–в–°в–°в–°в–°в–°в–°в–ұв–ұв–ұв–ұ 70%",
        "рҹ’—в–°в–°в–°в–°в–°в–°в–°в–°в–°в–° 100% рҹҳҚ"
    ];

    const searchMsg = await api.sendMessage(`рҹҺө TikTok Downloader\n\n${frames[0]}`, event.threadID);

    try {
        await api.editMessage(`рҹҺө Fetching video...\n\n${frames[1]}`, searchMsg.messageID);
        await api.editMessage(`рҹҺө Downloading...\n\n${frames[2]}`, searchMsg.messageID);

        const downloadResult = await downloadTikTok(url);
        
        if (!downloadResult || !downloadResult.success) {
            api.unsendMessage(searchMsg.messageID);
            return api.sendMessage("вқҢ Download failed. Please check the URL and try again.", event.threadID, event.messageID);
        }

        await api.editMessage(`рҹҺө Processing...\n\n${frames[3]}`, searchMsg.messageID);

        const cacheDir = path.join(__dirname, "cache");
        await fs.ensureDir(cacheDir);

        const videoPath = path.join(cacheDir, `${Date.now()}_tiktok.mp4`);
        fs.writeFileSync(videoPath, Buffer.from(downloadResult.data));

        await api.editMessage(`рҹҺө Complete!\n\n${frames[4]}`, searchMsg.messageID);

        await api.sendMessage(
            {
                body: `рҹҺө TikTok Video Downloaded`,
                attachment: fs.createReadStream(videoPath)
            },
            event.threadID
        );

        setTimeout(() => {
            try {
                if (fs.existsSync(videoPath)) fs.unlinkSync(videoPath);
                api.unsendMessage(searchMsg.messageID);
            } catch (err) {
                console.log("Cleanup error:", err);
            }
        }, 15000);

    } catch (error) {
        console.error("TikTok command error:", error.message);
        try { api.unsendMessage(searchMsg.messageID); } catch(e) {}
        return api.sendMessage("вқҢ An error occurred. Please try again.", event.threadID, event.messageID);
    }
};
