// lodash-pari/commands/top.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'top',
    aliases: ['rich', 'leaderboard', 'lb'],
    description: 'View the richest users',
    usage: 'top [number]',
    category: 'Economy',
    prefix: true
  },
  
  async run({ api, event, args, send, Currencies, Users }) {
    const limit = Math.min(parseInt(args[0]) || 10, 20);
    
    const topUsers = Currencies.getTop(limit);
    
    if (topUsers.length === 0) {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅝𝐨 𝐮𝐬𝐞𝐫𝐬 𝐢𝐧 𝐭𝐡𝐞 𝐥𝐞𝐚𝐝𝐞𝐫𝐛𝐨𝐚𝐫𝐝 𝐲𝐞𝐭.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    let msg = `RICHEST USERS
─────────────────\n`;
    
    for (let i = 0; i < topUsers.length; i++) {
      const user = topUsers[i];
      let name = 'Unknown';
      try {
        const info = await api.getUserInfo(user.id);
        name = info[user.id]?.name || 'Unknown';
      } catch {}
      
      const medal = i === 0 ? '🥇' : i === 1 ? '🥈' : i === 2 ? '🥉' : `${i + 1}.`;
      msg += `${medal} ${name}\n   Total: $${user.total.toLocaleString()}\n`;
    }
    
    return send.reply(msg);
  }
};
