// lodash-pari/commands/unban.js
const fs = require('fs-extra');
const path = require('path');

const spamTrackerPath = path.join(__dirname, './data/spam_tracker.json');
let Users;
try {
  Users = require('../../../../Data/system/controllers/users');
  if (!Users || !Users.unban) {
    Users = null;
  }
} catch (e) {
  Users = null;
}

function getSpamData() {
  try {
    if (!fs.existsSync(spamTrackerPath)) {
      fs.writeJsonSync(spamTrackerPath, { users: {}, autobanned: {} });
    }
    return fs.readJsonSync(spamTrackerPath);
  } catch {
    return { users: {}, autobanned: {} };
  }
}

function saveSpamData(data) {
  try {
    fs.writeJsonSync(spamTrackerPath, data, { spaces: 2 });
  } catch (err) {}
}

module.exports = {
  config: {
    cooldowns: 10,
    credits: 'CHAND',
    premium: false,
    permission: 1,
    version: '1.0.0',
    name: 'unban',
    aliases: ['unspam', 'spacunban'],
    description: 'Unban user from spam ban',
    usage: 'unban [uid]',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },

  async run({ api, event, args, send }) {
    const { mentions } = event;
    
    let uid = '';
    
    if (Object.keys(mentions).length > 0) {
      uid = Object.keys(mentions)[0];
    } else if (args[0] && /^\d+$/.test(args[0])) {
      uid = args[0];
    } else if (event.messageReply) {
      uid = event.messageReply.senderID;
    } else {
      return send.reply('━❮❌❯━━━❪⚠️❫━━━❮❌❯━\n🔒 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 🅤𝐈𝐃 𝐨𝐫 𝐫𝐞𝐩𝐥𝐲 𝐭𝐨 𝐮𝐬𝐞𝐫 𝐦𝐞𝐬𝐬𝐚𝐠𝐞.\n━❮❌❯━━━❪⚠️❫━━━❮❌❯━');
    }
    
    const data = getSpamData();
    const isSpamBanned = data.autobanned[uid];
    const isUserBanned = Users ? Users.isBanned(uid) : false;
    
    if (!isSpamBanned && !isUserBanned) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅣𝐡𝐢𝐬 𝐔𝐬𝐞𝐫 𝐢𝐬 𝐍𝐨𝐭 𝐁𝐚𝐧𝐧𝐞𝐝 🔓\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    delete data.autobanned[uid];
    saveSpamData(data);
    
    if (Users) {
      Users.unban(uid);
    }
    
    let name = 'User';
    try {
      const info = await api.getUserInfo(uid);
      name = info[uid]?.name || 'User';
    } catch {}
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅤𝐧𝐛𝐚𝐧𝐧𝐞𝐝 ${name} (${uid})\n\n🚫 𝐒𝐩𝐚𝐦 𝐛𝐚𝐧 & 👤 𝐔𝐬𝐞𝐫 𝐛𝐚𝐧 𝐫𝐞𝐦𝐨𝐯𝐞𝐝.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
