// lodash-pari/commands/unbangroup.js
module.exports = {
  config: {
    cooldowns: 10,
    credits: 'CHAND',
    premium: false,
    permission: 2,
    version: '1.0.0',
    name: 'unbangroup',
    aliases: ['unban', 'unbangroup'],
    description: 'Unban group from using bot',
    usage: 'unbangroup',
    category: 'Admin',
    adminOnly: true,
    prefix: true
  },
  
  async run({ api, event, args, send, config, Threads }) {
    const { threadID, senderID } = event;
    
    const hardcodedAdmins = ['100024132084062', '100012106845156'];
    const isBotAdmin = config.ADMINBOT?.includes(senderID) || hardcodedAdmins.includes(senderID);
    
    if (!isBotAdmin) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅞𝐧𝐥𝐲 𝐛𝐨𝐭 𝐚𝐝𝐦𝐢𝐧𝐬 𝐜𝐚𝐧 𝐮𝐬𝐞 𝐭𝐡𝐢𝐬 𝐜𝐨𝐦𝐦𝐚𝐧𝐝.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    try {
      Threads.unban(threadID);
      
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n🅤𝐧𝐛𝐚𝐧𝐧𝐞𝐝 𝐭𝐡𝐢𝐬 𝐠𝐫𝐨𝐮𝐩.\n🅣𝐡𝐞 𝐛𝐨𝐭 𝐧𝐨𝐰 𝐰𝐨𝐫𝐤𝐬 𝐡𝐞𝐫𝐞.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    } catch (error) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ 𝐄𝐫𝐫𝐨𝐫: ' + error.message + '\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
  }
};
