// lodash-pari/commands/withdraw.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',

    name: 'withdraw',
    aliases: ['wd', 'take'],
    description: 'Withdraw money from bank',
    usage: 'withdraw [amount/all]',
    category: 'Economy',
    prefix: true
  },
  
  async run({ api, event, args, send, Currencies, Users }) {
    const { senderID } = event;
    
    const name = await Users.getNameUser(senderID);
    const bank = Currencies.getBank(senderID);
    
    if (bank <= 0) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ ${name}, 𝐲𝐨𝐮 𝐡𝐚𝐯𝐞 𝐧𝐨 𝐦𝐨𝐧𝐞𝐲 𝐢𝐧 𝐲𝐨𝐮𝐫 𝐛𝐚𝐧𝐤 𝐭𝐨 𝐰𝐢𝐭𝐡𝐝𝐫𝐚𝐰!\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    let amount;
    
    if (args[0]?.toLowerCase() === 'all') {
      amount = bank;
    } else {
      amount = parseInt(args[0]);
    }
    
    if (!amount || amount <= 0 || isNaN(amount)) {
      return send.reply('━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ 🅟𝐥𝐞𝐚𝐬𝐞 𝐩𝐫𝐨𝐯𝐢𝐝𝐞 𝐚 𝐯𝐚𝐥𝐢𝐝 𝐚𝐦𝐨𝐮𝐧𝐭 𝐭𝐨 𝐰𝐢𝐭𝐡𝐝𝐫𝐚𝐰.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━');
    }
    
    if (amount > bank) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n❌ ${name}, 𝐲𝐨𝐮 𝐝𝐨𝐧'𝐭 𝐡𝐚𝐯𝐞 𝐞𝐧𝐨𝐮𝐠𝐡 𝐢𝐧 𝐛𝐚𝐧𝐤!\n\n🏦 𝐁𝐚𝐧𝐤: $${bank.toLocaleString()}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    const success = Currencies.withdraw(senderID, amount);
    
    if (!success) {
      return send.reply('Failed to withdraw money.');
    }
    
    const newWallet = Currencies.getBalance(senderID);
    const newBank = Currencies.getBank(senderID);
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅦𝐈𝐓𝐇𝐃𝐑𝐀𝐖𝐀𝐋 𝐒𝐔𝐂𝐂𝐄𝐒𝐒𝐅𝐔𝐋!\n─────────────────\n👤 𝐔𝐬𝐞𝐫: ${name}\n💸 𝐖𝐢𝐭𝐡𝐝𝐫𝐚𝐰𝐧: $${amount.toLocaleString()}\n─────────────────\n👛 𝐖𝐚𝐥𝐥𝐞𝐭: $${newWallet.toLocaleString()}\n🏦 𝐁𝐚𝐧𝐤: $${newBank.toLocaleString()}\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
