// lodash-pari/commands/work.js
module.exports = {
  config: {
cooldowns: 10,
credits: 'CHAND',
premium: false,
permission: 0,
version: '1.0.0',
    name: 'work',
    aliases: ['job', 'earn'],
    description: 'Work to earn money',
    usage: 'work',
    category: 'Economy',
    prefix: true
  },
  
  async run({ api, event, send, Currencies, Users }) {
    const { senderID } = event;
    
    const result = Currencies.work(senderID);
    const name = await Users.getNameUser(senderID);
    
    if (!result.success) {
      return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n😴 ${name}, 𝐲𝐨𝐮 𝐧𝐞𝐞𝐝 𝐭𝐨 𝐫𝐞𝐬𝐭!\n\n⏳ 𝐘𝐨𝐮 𝐜𝐚𝐧 𝐰𝐨𝐫𝐤 𝐚𝐠𝐚𝐢𝐧 𝐢𝐧 ${result.remaining} 𝐦𝐢𝐧𝐮𝐭𝐞𝐬.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
    }
    
    const balance = Currencies.getBalance(senderID);
    
    return send.reply(`━❮🐉❯━━━❪💝❫━━━❮🐉❯━\n✅ 🅦𝐎𝐑𝐊 𝐂𝐎𝐌𝐏𝐋𝐄𝐓𝐄!\n─────────────────\n👤 𝐔𝐬𝐞𝐫: ${name}\n💼 𝐉𝐨𝐛: ${result.job}\n💰 𝐄𝐚𝐫𝐧𝐢𝐧𝐠𝐬: $${result.earnings.toLocaleString()}\n─────────────────\n💵 𝐍𝐞𝐰 𝐁𝐚𝐥𝐚𝐧𝐜𝐞: $${balance.toLocaleString()}\n⏳ 𝐘𝐨𝐮 𝐜𝐚𝐧 𝐰𝐨𝐫𝐤 𝐚𝐠𝐚𝐢𝐧 𝐢𝐧 𝟑𝟎 𝐦𝐢𝐧𝐮𝐭𝐞𝐬.\n━❮🐉❯━━━❪💝❫━━━❮🐉❯━`);
  }
};
